<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-GB" xml:lang="en-GB"><head>
    <meta charset="utf-8">
    <meta name="generator" content="quarto-1.5.56">

    <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


    <title>Local to global N input modelling</title>
    <style>
      code{white-space: pre-wrap;}
      span.smallcaps{font-variant: small-caps;}
      div.columns{display: flex; gap: min(4vw, 1.5em);}
      div.column{flex: auto; overflow-x: auto;}
      div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
      ul.task-list{list-style: none;}
      ul.task-list li input[type="checkbox"] {
        width: 0.8em;
        margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
        vertical-align: middle;
      }
      /* CSS for syntax highlighting */
      pre > code.sourceCode { white-space: pre; position: relative; }
      pre > code.sourceCode > span { line-height: 1.25; }
      pre > code.sourceCode > span:empty { height: 1.2em; }
      .sourceCode { overflow: visible; }
      code.sourceCode > span { color: inherit; text-decoration: inherit; }
      div.sourceCode { margin: 1em 0; }
      pre.sourceCode { margin: 0; }
      @media screen {
      div.sourceCode { overflow: auto; }
      }
      @media print {
      pre > code.sourceCode { white-space: pre-wrap; }
      pre > code.sourceCode > span { display: inline-block; text-indent: -5em; padding-left: 5em; }
      }
      pre.numberSource code
        { counter-reset: source-line 0; }
      pre.numberSource code > span
        { position: relative; left: -4em; counter-increment: source-line; }
      pre.numberSource code > span > a:first-child::before
        { content: counter(source-line);
          position: relative; left: -1em; text-align: right; vertical-align: baseline;
          border: none; display: inline-block;
          -webkit-touch-callout: none; -webkit-user-select: none;
          -khtml-user-select: none; -moz-user-select: none;
          -ms-user-select: none; user-select: none;
          padding: 0 4px; width: 4em;
        }
      pre.numberSource { margin-left: 3em;  padding-left: 4px; }
      div.sourceCode
        {   }
      @media screen {
      pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
      }
      /* CSS for citations */
      div.csl-bib-body { }
      div.csl-entry {
        clear: both;
        margin-bottom: 0em;
      }
      .hanging-indent div.csl-entry {
        margin-left:2em;
        text-indent:-2em;
      }
      div.csl-left-margin {
        min-width:2em;
        float:left;
      }
      div.csl-right-inline {
        margin-left:2em;
        padding-left:1em;
      }
      div.csl-indent {
        margin-left: 2em;
      }    </style>

    <style>
      body.hypothesis-enabled #quarto-embed-header {
        padding-right: 36px;
      }

      #quarto-embed-header {
        height: 3em;
        width: 100%;
        display: flex;
        justify-content: space-between;
        align-items: center;
        border-bottom: solid 1px;
      }

      #quarto-embed-header h6 {
        font-size: 1.1em;
        padding-top: 0.6em;
        margin-left: 1em;
        margin-right: 1em;
        font-weight: 400;
      }

      #quarto-embed-header a.quarto-back-link,
      #quarto-embed-header a.quarto-download-embed {
        font-size: 0.8em;
        margin-top: 1em;
        margin-bottom: 1em;
        margin-left: 1em;
        margin-right: 1em;
      }

      .quarto-back-container {
        padding-left: 0.5em;
        display: flex;
      }

      .headroom {
          will-change: transform;
          transition: transform 200ms linear;
      }

      .headroom--pinned {
          transform: translateY(0%);
      }

      .headroom--unpinned {
          transform: translateY(-100%);
      }      
    </style>

    <script>
    window.document.addEventListener("DOMContentLoaded", function () {

      var header = window.document.querySelector("#quarto-embed-header");
      const titleBannerEl = window.document.querySelector("body > #title-block-header");
      if (titleBannerEl) {
        titleBannerEl.style.paddingTop = header.clientHeight + "px";
      }
      const contentEl = window.document.getElementById('quarto-content');
      for (const child of contentEl.children) {
        child.style.paddingTop = header.clientHeight + "px";
        child.style.marginTop = "1em";
      }

      // Use the article root if the `back` call doesn't work. This isn't perfect
      // but should typically work
      window.quartoBackToArticle = () => {
        var currentUrl = window.location.href;
        window.history.back();
        setTimeout(() => {
            // if location was not changed in 100 ms, then there is no history back
            if(currentUrl === window.location.href){              
                // redirect to site root
                window.location.href = "index.html";
            }
        }, 100);
      }

      const headroom = new window.Headroom(header, {
        tolerance: 5,
        onPin: function () {
        },
        onUnpin: function () {
        },
      });
      headroom.init();
    });
    </script>

    
<script src="site_libs/manuscript-notebook/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="site_libs/quarto-contrib/glightbox/glightbox.min.js"></script>
<link href="site_libs/quarto-contrib/glightbox/glightbox.min.css" rel="stylesheet">
<link href="site_libs/quarto-contrib/glightbox/lightbox.css" rel="stylesheet">
      <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script> 
        <link rel="stylesheet" href="extra.css">
      <meta name="citation_title" content="Local to global N input modelling">
<meta name="citation_language" content="en-GB">
<meta name="citation_reference" content="citation_title=Dispersal and assimilation of an aquaculture waste subsidy in a low productivity coastal environment;,citation_abstract=To understand dispersal and assimilation of aquaculture waste subsidies in a naturally low-productivity environment, we applied a novel, rapid transmethylation technique to analyse sediment and biota fatty acid composition. This technique was initially validated at Atlantic salmon farms in Macquarie Harbour, Australia, where sediments were collected at farm and control locations. Subsequently, sediment, benthic polychaete and zooplankton were sampled at sites 0, 50, 250, 500 and 1000 m distant from multiple cages. Results demonstrated an acute deposition zone up to 50 m from cages and a diffuse zone extending 500 m from cages. Changes in sediment concentration of linoleic acid, oleic acid and total fatty acids were effective tracers of farm deposition. Bacterial biomarkers indicated that aquaculture waste stimulates bacterial productivity in sediments, with elevated biomarker concentrations also detected in benthic polychaetes. Overall, fatty acid analysis was a sensitive technique to characterize the benthic footprint of aquaculture influence.;,citation_author=Camille A. White;,citation_author=P. D. Nichols;,citation_author=D. J. Ross;,citation_author=Tim Dempster;,citation_publication_date=2017-07;,citation_cover_date=2017-07;,citation_year=2017;,citation_issue=1-2;,citation_doi=https://doi.org/ 10.1016/j.marpolbul.2017.05.042;,citation_issn=0025326X;,citation_volume=120;,citation_journal_title=Marine Pollution Bulletin;">
<meta name="citation_reference" content="citation_title=Future aquafeeds may compromise reproductive fitness in a marine invertebrate;,citation_abstract=Aquaculture of higher trophic level species is increasingly dependent on the use of terrestrial oil products. The input of terrestrially derived n-6 polyunsaturated fatty acids (PUFA) into marine environments has subsequently increased, with unknown consequences for recipient species. We exposed a sea urchin, Heliocidaris erythrogramma to three experimental diets for 78 days: a high n-3 PUFA marine imitation treatment, a high n-6 PUFA “future aquafeed” treatment and an intermediate “current aquafeed” treatment. Female urchins fed the high n-6 PUFA diet produced larvae with lower survival rates than all other treatments. Males fed the high n-6 PUFA diet produced no viable sperm. Fatty acid composition in reproductive material revealed comprehensive biosynthetic and dietary sparing capabilities in H.&nbsp;erythrogramma. Despite this, the ratio of n-6 PUFA to n-3 PUFA in reproductive tissue increased significantly with diet. We suggest alterations to this ratio is the likely mechanism of negative impact on larval development.;,citation_author=Camille A. White;,citation_author=Symon A. Dworjanyn;,citation_author=Peter D. Nichols;,citation_author=Benjamin Mos;,citation_author=Tim Dempster;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_doi=https://doi.org/ 10.1016/j.marenvres.2016.09.008;,citation_issn=18790291;,citation_volume=122;,citation_journal_title=Marine Environmental Research;">
<meta name="citation_reference" content="citation_title=Consumption of aquaculture waste affects the fatty acid metabolism of a benthic invertebrate;,citation_abstract=Trophic subsidies can drive widespread ecological change, thus knowledge of how keystone species respond to subsidies is important. Aquaculture of large carnivorous fish generates substantial waste as faeces and lost feed, providing a food source to mobile benthic invertebrates. We used a controlled feeding study combined with a field survey to better understand the interaction between salmon aquaculture and the sea urchin, Echinus acutus, a dominant mobile invertebrate in Norwegian fjords. We tested if diets affected urchin fatty acid composition by feeding them one of three diet treatments (“aquafeed”, “composite” and “natural”) for 10 weeks. To test if proximity to fish farms altered E. acutus fatty acid composition, populations were sampled at 10 locations in Hardangerfjord and Masfjord (Western Norway) from directly adjacent and up to 12 km from farms. Fatty acids were measured in gonads and eggs in the diet experiment and in gonads and gut contents from wild animals. Urchins directly assimilated aquaculture waste at farm sites, as evidenced by elevated linoleic acid (LA), oleic acid (OA) and ∑LA, OA in their tissues. The diet experiment highlighted the biosynthetic and selective dietary sparing capacity of E. acutus in both gonads and eggs, with evidence for the elongation and desaturation of eicosapentaenoic acid (EPA) and arachidonic acid (ARA) from C18 fatty acid precursors. Elevated biosynthesis of non-methylene interrupted (NMI) fatty acids, in particular 20:3Δ7,11,14 and 20:2 Δ5,11, were also linked to a high C18 fatty acid, low ≥ C20 long-chain polyunsaturated fatty acid (LC-PUFA) diet. Fatty acid composition of gonads of wild urchins indicated a highly variable diet. The study indicates that the generalist feeding ecology of E. acutus, coupled with extensive biosynthetic capacity, enables it to exploit aquaculture waste as an energyrich trophic subsidy.;,citation_author=Camille White;,citation_author=Raymond J. Bannister;,citation_author=Symon A. Dworjanyn;,citation_author=Vivian Husa;,citation_author=Peter D. Nichols;,citation_author=Tina Kutti;,citation_author=Tim Dempster;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_doi=https://doi.org/10.1016/j.scitotenv.2017.02.109;,citation_volume=586;,citation_journal_title=Science of The Total Environment;">
<meta name="citation_reference" content="citation_title=Aquaculture-derived trophic subsidy boosts populations of an ecosystem engineer;,citation_abstract=Environmental management of coastal aquaculture is focused on acute impacts of organic and nitrogenous wastes close to farms. However, the energy-rich trophic subsidy that aquaculture provides may create cascades with influences over broader spatial scales. In a fjord region with intensive fish farming, we tested whether an ecosystem engineer, the white urchin Gracilechinus acutus, was more abundant at aquaculture sites than control sites. Further, we tested whether diets influenced by aquaculture waste altered reproductive outputs compared with natural diets. Urchins formed barrens at aquaculture sites where they were 10 times more abundant (38 urchins m−2) than at control sites (4 urchins m−2). Urchins were on average 15 mm larger at control sites. In the laboratory, urchins fed aquafeed diets had 3 times larger gonad indices than urchins fed a natural diet. However, their reproduction was compromised. Eggs from females fed an aquafeed diet had 13% lower fertilisation success and 30% lower larval survival rates at 10 d compared with females fed a natural diet. A reproductive output model showed that enhanced numbers of 10 d old larvae produced by the dense aquaculture-associated aggregations of G. acutus will supersede any detrimental effects on reproduction, with larval outputs from aquaculture sites being on average 5 times greater than control sites. The results show that aquaculture waste can act as a trophic subsidy in fjord ecosystems, stimulating aggregations of urchins and promoting the formation of urchin barrens. Where finfish aquaculture is concentrated, combined effects on the wider environment may produce ecosystem-level consequences.;,citation_author=Camille White;,citation_author=Raymond J. Bannister;,citation_author=Symon A. Dworjanyn;,citation_author=Vivian Husa;,citation_author=Peter D. Nichols;,citation_author=Tim Dempster;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_doi=https://doi.org/10.3354/aei00270;,citation_volume=10;,citation_journal_title=Aquaculture Environment Interactions;">
<meta name="citation_reference" content="citation_title=Towards achieving circularity and sustainability in feeds for farmed blue foods;,citation_abstract=The aims of this review are to describe the role of “blue‐food production” (animals, plants and algae harvested from freshwater and marine environments) within a circular bioeconomy, discuss how such a framework can help the sustainability and resilience of aquaculture and to summarise key examples of novel nutrient sources that are emerging in the field of fed‐aquaculture species. Aquaculture now provides &amp;amp;amp;gt;50% of the global seafood supply, a share that is expected to increase to at least 60% within the next decade. Aquaculture is an important tool for reducing resource consumption in global protein production and increasing resilience to climate change and other global disruptions (i.e. pandemics, geo‐political instability). Importantly, blue foods also provide essential nutrition for a growing human population. Blue foods are helping to help the global goal of “zero hunger” (United Nation’s Sustainable Development Goal 2) while reducing the dependency on finite natural resources but further refinement and new solutions are needed to make the industry more “circular” and sustainable, particularly with respect to sourcing raw materials for aquafeeds. This review describes the feed resources that are available or may be created within a circular bioeconomy framework, their role within the framework and in aquaculture and ultimately, how these resources contribute to de‐risking and establishing a resilient aquaculture production chain.;,citation_author=Stefanie M. Colombo;,citation_author=Koushik Roy;,citation_author=Jan Mraz;,citation_author=Alex H. L. Wan;,citation_author=Simon J. Davies;,citation_author=Sean M. Tibbetts;,citation_author=Margareth Øverland;,citation_author=David S. Francis;,citation_author=Melissa M. Rocker;,citation_author=Laura Gasco;,citation_author=Marc Metian;,citation_author=Jesse T. Trushenski;,citation_author=Giovanni M. Turchini;,citation_publication_date=2022-11;,citation_cover_date=2022-11;,citation_year=2022;,citation_doi=https://doi.org/ 10.1111/raq.12766;,citation_journal_title=Reviews in Aquaculture;">
<meta name="citation_reference" content="citation_title=Insect feed in sustainable crustacean aquaculture;,citation_abstract=Aquaculture is a growing global food production sector that aims to meet the increasing demand for dietary protein. Crustaceans are an important and predominantly high-priced aquaculture segment that could support the transfer of sustainable new technologies to other sectors. Areas of interest include disease management and compound feeds, both of which have the potential to improve both the profitability and sustainability of aquaculture. Modern compound feeds are largely composed of fishmeal and/or terrestrial plant materials, the production of which is unsustainable, leading to the depletion of finite resources. Insects are a promising protein-rich alternative to fishmeal that can reduce the environmental footprint of aquafeeds and crustacean aquaculture. First research data have shown that insect meal has a favourable nutritional composition with positive health effects, and is environmentally sustainable with a strong economic potential, particularly supporting circular economy by valorising otherwise unused side-streams. In this article, we discuss the current state of crustacean aquaculture and highlight the benefits of insect meal compared to today’s compound aquafeeds in terms of health and growth-promoting properties as well as environmental benefits. We then consider the molecular mechanisms that confer immunity and disease resistance in crustacean aquaculture and show how insect feeds can support disease management and thus consumer health. Next, we assess the environmental sustainability of crustacean aquaculture and insect farming, the legal framework for insect-based feeds and consumer acceptance issues. Lastly, we identify research gaps, socioeconomic considerations and the potential of insect-based sustainable crustacean aquaculture for the global market.;,citation_author=Till Röthig;,citation_author=Annalena Barth;,citation_author=Martin Tschirner;,citation_author=Patrick Schubert;,citation_author=Marius Wenning;,citation_author=André Billion;,citation_author=Thomas Wilke;,citation_author=Andreas Vilcinskas;,citation_publication_date=2023-04;,citation_cover_date=2023-04;,citation_year=2023;,citation_issue=9;,citation_doi=https://doi.org/ 10.3920/jiff2022.0117;,citation_volume=9;,citation_journal_title=Journal of Insects as Food and Feed;">
<meta name="citation_reference" content="citation_title=Modeling of nutritional behavior and harvest profit margin in fattening pacu (&amp;amp;amp;lt;i&amp;gt;Piaractus mesopotamicus&amp;lt;/i&amp;gt;) fed on balanced digestible protein contents and raised in an intensive recirculating aquaculture system;,citation_abstract=Adjusting aquafeed to obtain sustainable growth and maximum profit margin of harvest is a worthy stride toward achieving the Sustainable Development Goals (SDGs) outlined by the United Nations. Therefore, this research aimed at modeling of nutritional behavior and harvest profit margin in fattening pacu (Piaractus mesopotamicus) fed on balanced digestible protein (BDP) contents and raised in an intensive recirculating aquaculture system (RAS). The feeding trial consisted of six treatments with three randomly arranged replicates (tanks) per treatment. A total of ten fattening pacus (490.60 ± 14.39 g, initial weight) were stocked per tank. Six experimental feeds containing the increasing contents (163, 201, 238, 272, 315, and 348 g kg-1 diet, dry matter) of BDP were manually fed to the fish for seven weeks three times a day until apparent satiation. The data obtained over the test diets were assessed through appropriate statistical procedures and mathematical models. Optimal levels of BDP were calculated for different nutritional and economic scenarios. Feed cost showed an ascending linear relationship with the BDP intake. Feed intake, feed efficiency, final body weight, weight gain, and specific growth rate (%) showed an inverted U-shaped quadratic behavior (P &amp;amp;lt; 0.05). Feed conversion ratio showed a U-shaped quadratic behavior (P < 0.05). Protein efficiency ratio significantly (P < 0.05) decreased linearly in response to the increasing BDP contents. Final weights of the eviscerated and sliced fish showed an ascending linear behavior, but the overall effect was insignificant (P &amp;gt; 0.05) between the treatments. Whole body moisture and crude protein content increased (P < 0.05) linearly, while the whole body lipid content decreased (P < 0.05) linearly over the increasing BDP intake. Whole body ash content showed a U-shaped quadratic behavior (P < 0.05). The nutritionally optimal BDP content corresponding to the maximum gain in body weight of fattening pacu was calculated as 292 g kg-1 diet (dry matter). To obtain sustainable growth and maximum profit margin of harvest when it will be sold in the form of whole-body, eviscerated, and sliced fish the economically functional BDP intakes were calculated as 249, 256, and 260 g kg-1 diet (dry matter), respectively. This research revealed that the optimal BDP intake greatly improve the nutritional performance of fattening pacu. Moreover, harvest profit margin and eventually the economically sustainable BDP intakes were found dependent on the way the fish harvest is marketed and the changes in total feed cost.;,citation_author=Kifayat Ullah Khan;,citation_author=Robert Mervyn Gous;,citation_author=Thiago Matias Torres Nascimento;,citation_author=Nilva Kazue Sakomura;,citation_author=Cleber Fernando Menegasso Mansano;,citation_author=Daniel Monge A. Queiroz;,citation_author=Andressa Tellechea Rodrigues;,citation_author=André Zuffo Boaratti;,citation_author=Rafael de Souza Romaneli;,citation_author=João Batista Kochenborger Fernandes;,citation_publication_date=2022-09;,citation_cover_date=2022-09;,citation_year=2022;,citation_doi=10.1016/j.anifeedsci.2022.115430;,citation_journal_title=Animal Feed Science and Technology;">
<meta name="citation_reference" content="citation_title=Complementarity of insect meal and poultry by-product meal as replacement for fishmeal can sustain the production performance of European perch (&amp;amp;amp;lt;i&amp;gt;Perca fluviatilis&amp;lt;/i&amp;gt;), reduce economic fish-in fish-out ratio and food-feed competition, and influence the environmental indices;,citation_author=Hung Q. Tran;,citation_author=Elena Wernicke von Siebenthal;,citation_author=Jean-Baptiste Luce;,citation_author=Tram Nguyen;,citation_author=Aleš Tomčala;,citation_author=Vlastimil Stejskal;,citation_author=Thomas Janssens;,citation_publication_date=2024-01;,citation_cover_date=2024-01;,citation_year=2024;,citation_doi=https://doi.org/ 10.1016/j.aquaculture.2023.740166;,citation_volume=579;">
<meta name="citation_reference" content="citation_title=Comparative life cycle assessment of heterotrophic microalgae &amp;amp;amp;lt;i&amp;gt;schizochytrium&amp;lt;/i&amp;gt; and fish oil in sustainable aquaculture feeds;,citation_abstract=Fish oil is primarily extracted from small marine pelagic fishes, reducing their availability for marine wildlife forage and artisanal fishing catches that support food security in lower income coastal nations. A primary use of fish oil is in feeds for aquaculture, the world’s fastest growing food sector. Efforts to transition fed aquaculture to sustainability includes replacing fish oil in aquafeeds with more environmentally responsible alternative ingredients. The heterotrophic marine microalga Schizochytrium sp., one of the first commercialized alternatives, lacks an open-access, systematic analysis of environmental impacts of substituting fish oil with heterotrophic microalgae from biorefineries. This study compared the “cradle to factory-gate” life cycle impacts of fish oil to whole-cell or extracted oil of Schizochytrium combined with canola oil. We conducted an attributional life cycle assessment using inventory data collected from published literature and patents and received feedback on commercial relevance of model assumptions from industry advisors. We examined sugar from a Brazilian sugarcane biorefinery and sucrose from U.S. sugar beets as feedstocks for heterotrophic cultivation of Schizochytrium; and compared life cycle impacts of extracting Schizochytrium oil using solvent-free microwave extraction to conventional solvent extraction. Results were that: cultivation processes had the largest overall effect for both products in both regions; whole-cell Schizochytrium combined with canola oil had significantly lower environmental impacts, in all assessed categories, than Schizochytrium oil blended with canola oil; and conventional solvent extraction had significantly lower environmental impacts compared to solvent-free microwave extraction except global warming potential. Schizochytrium products, compared to fish oil, had substantially lower biotic resource depletion and, in the case of whole cells combined with canola oil, had significantly lower global warming potential but higher impacts for all other categories, primarily because commercial Schizochytrium production used sugar feedstocks as carbon and energy sources. The mix of lower and higher environmental impacts of Schizochytrium products illustrates the importance of openly identifying environmental trade-offs to inform evidence-based decisions for commercial practices. Environmental impacts should also be weighed against potential human health benefits of maintaining omega-3 fatty acids and avoiding contaminants in fish flesh when considering alternatives to fish oil.;,citation_author=Brandi McKuin;,citation_author=Anne R. Kapuscinski;,citation_author=Pallab K. Sarker;,citation_author=Nicolette Cheek;,citation_author=Adele Colwell;,citation_author=Benjamin Schoffstall;,citation_author=Connor Greenwood;,citation_publication_date=2022-01;,citation_cover_date=2022-01;,citation_year=2022;,citation_issue=1;,citation_doi=https://doi.org/10.1525/elementa.2021.00098;,citation_volume=10;">
<meta name="citation_reference" content="citation_title=Defatted black soldier fly (&amp;amp;amp;lt;i&amp;gt;Hermetia illucens&amp;lt;/i&amp;gt;) in pikeperch (<i>Sander lucioperca</i>) diets: Effects on growth performance, nutrient digestibility, fillet quality, economic and environmental sustainability;,citation_abstract=The use of insect meal in aquafeed formulations has recently gained attention. Detailed knowledge about the inclusion levels for pikeperch ( Sander lucioperca ), a promising candidate for intensive aquaculture in Europe remains, however, fragmented. In the present study, 4 isoproteic (45% dry matter) and isoenergetic (21 MJ/kg) diets were formulated, including a control diet (H0) containing 30% fishmeal (FM) on an as fed basis and the other 3 diets in which FM protein was replaced by defatted black soldier fly meal ( Hemetia illucens ) (HIM) at 25%, 50%, and 100% (diet abbreviation H9, H18 and H36, corresponding to an inclusion level of 9%, 18% and 36%, respectively). The feeding trial was performed in triplicate groups of 50 juvenile pikeperch (mean weight, 68.7 g) fed with experimental diets for 84 d during which the growth performance, nutrient digestibility, fillet quality and economic and environmental sustainability of rearing pikeperch were evaluated. Our findings indicated that pikeperch fed H0, H9, and H18 groups displayed better results regarding growth performance indices, except for survival rate where no significant difference among groups was recorded ( P = 0.642). A significantly lower organ-somatic index, including hepatosomatic, viscerosomatic and perivisceral fat index, was found in fish fed H18 groups than other groups ( P &amp;amp;lt; 0.05). Inclusion of HIM affected the digestibility of the nutrients and resulted in an almost linear reduction in the apparent digestibility coefficient of dry matter and protein. Concerning the fillet quality, dietary HIM negatively affected the protein and ash contents of the fish fillets, while the crude fat remained unchanged. Dietary HIM did not significantly modify total saturated, monounsaturated and polyunsaturated fatty acids in the fillets of fed pikeperch ( P &amp;gt; 0.05) but did reduce total n-3 fatty acids ( P = 0.001) and increased total n-6 ( P < 0.001). Increasing inclusion levels of HIM reduced the environmental impacts associated with fish-in-to-fish out ratio but entailed heavy burdens on energy use and eutrophication. Low and moderate inclusion levels of HIM did not negatively affect land use and water use compared to an HIM-free diet ( P > 0.05). The addition of HIM at a level as low as 9% elicited a similar carbon footprint to that of the control diet. The economic conversion ratio and economic profit index were negatively affected at increased insect meal inclusion levels. This study has shown that the incorporation of HIM in feed formulations for pikeperch is feasible at inclusion levels of 18% without adverse effects on growth performance parameters. The feasibility also highlighted the environmental benefits associated with land use and marine resources required to produce farmed fish.;,citation_author=Vlastimil Stejskal;,citation_author=Hung Q. Tran;,citation_author=Markéta Prokesová;,citation_author=Mahyar Zare;,citation_author=Tatyana Gebauer;,citation_author=Tomas Policar;,citation_author=Christian Caimi;,citation_author=Francesco Gai;,citation_author=Laura Gasco;,citation_publication_date=2022-09;,citation_cover_date=2022-09;,citation_year=2022;,citation_doi=https://doi.org/10.1016/j.aninu.2022.06.022;,citation_journal_title=Animal Nutrition;">
<meta name="citation_reference" content="citation_title=Interventions for improving the productivity and environmental performance of global aquaculture for future food security;,citation_abstract=Aquatic foods are increasingly being recognized as having an important role to play in an environmentally sustainable and nutritionally sufficient food system. Proposals for increasing aquatic food production often center around species, environments, and ambitious hi-tech solutions that mainly will benefit the 16% of the global population living in high-income countries. Meanwhile, most aquaculture species and systems suffer from large performance gaps, meaning that targeted interventions and investments could significantly boost aquatic food supply and access to nutritious foods without a concomitant increase in environmental footprints. Here we contend that the dialogue around aquatic foods should pay greater attention to identifying and implementing interventions to improve the productivity and environmental performance of low-value commodity species that have been relatively overlooked in this regard to date. We detail a range of available technical and institutional intervention options and evaluate their potential for increasing the output and environmental performance of global aquaculture.;,citation_author=Patrik J. G. Henriksson;,citation_author=Max Troell;,citation_author=Lauren Katherine Banks;,citation_author=Ben Belton;,citation_author=Malcolm Beveridge;,citation_author=Dane H. Klinger;,citation_author=Nathan Pelletier;,citation_author=Michael Phillips;,citation_author=Nhuong Tran;,citation_publication_date=2021-09;,citation_cover_date=2021-09;,citation_year=2021;,citation_issue=9;,citation_doi=https://doi.org/ 10.1016/j.oneear.2021.08.009;,citation_volume=4;,citation_journal_title=One Earth;">
<meta name="citation_reference" content="citation_title=Environmental consequences of using insect meal as an ingredient in aquafeeds: A systematic view;,citation_abstract=We retrieved data from various studies to investigate the consequences of insect meal production and insect meal-based diets with respect to their environmental impact, including global warming potential, energy use, land use, water use, acidification, eutrophication as well as to economic fish-in fish-out ratio and solid waste output production. Analysis indicated that insect meals’ production exerted positive effects on land use but was associated with greater energy use and a larger carbon footprint compared to conventional protein sources. Substitution of silkworm meal (Bombyx mori) meals for fishmeal in aquatic animal diets significantly reduced solid phosphorus waste compared to insect-free diets. In contrast, the inclusion of black soldier fly (Hermetia illucens), housefly (Musca domestica), mealworm (Tenebrio molitor) and grasshopper (Zonocerus variegatus) has led, in comparison to insect-free diet, to greater solid nitrogen waste. Reducing the proportion of fishmeal and, to a lesser extent fish oil, by various insect meals in aquatic diet formulations significantly reduces economic fish-in fish-out, indicating less marine forage fish required per unit fish yield. The simulated data showed environmental benefit associated with land use of insect-containing aquafeeds compared to insect meal-free feeds, especially insect species of M. domestica and T. molitor. In all, this study suggested a trade-off of using insect meal as an aquafeed ingredient regarding environmental consequence. Since insect meal has excellent potential to supply protein for aquafeeds in the coming years, improvement in insect meal production systems and nutritional composition will be essential to make insect meal a sustainable aquafeed ingredient.;,citation_author=Hung Q. Tran;,citation_author=Hien Van Doan;,citation_author=Vlastimil Stejskal;,citation_publication_date=2021-07;,citation_cover_date=2021-07;,citation_year=2021;,citation_doi=https://doi.org/ 10.1111/raq.12595;,citation_journal_title=Reviews in Aquaculture;">
<meta name="citation_reference" content="citation_title=Microalgae in aquafeeds for a sustainable aquaculture industry;,citation_abstract=Due to the rapid global expansion of the aquaculture industry, access to key feedstuffs (fishmeal and fish oil) is becoming increasingly limited because of the finite resources available for wild fish harvesting. This has resulted in other sources of feedstuffs being investigated, namely plant origin substitutes for fishmeal and fish oil for aquafeed. Conventional land-based crops have been favored for some applications as substitutes for a portion of the fishmeal, but they can result in changes in the nutritional quality of the fish produced. Microalgae can be regarded as a promising alternative that can replace fishmeal and fish oil and ensure sustainability standards in aquaculture. They have a potential for use in aquaculture as they are sources of protein, lipid, vitamins, minerals, pigments, etc. This comprehensive review summarizes the most important and recent developments of microalgae use as supplement or feed additive to replace fishmeal and fish oil for use in aquaculture. It also reflects the microalgal nutritional quality and digestibility of microalgae-based aquafeed. Simultaneously, safety and regulatory aspects of microalgae feed applications, major challenges on the use microalgae in aquafeed in commercial production, and future research and development perspective are also presented in a critical manner. This review will serve as a useful guide to present current status of knowledge and highlight key areas for future development of a microalgae-based aquafeed industry and overall development of a sustainable aquaculture industry.;,citation_author=Mahfuzur Rahman Shah;,citation_author=Giovanni Antonio Lutzu;,citation_author=Asraful Alam;,citation_author=Pallab K. Sarker;,citation_author=M. A. Kabir Chowdhury;,citation_author=Ali Parsaeimehr;,citation_author=Yuanmei Liang;,citation_author=Maurycy Daroch;,citation_publication_date=2018-02;,citation_cover_date=2018-02;,citation_year=2018;,citation_issue=1;,citation_doi=https://doi.org/ 10.1007/s10811-017-1234-z;,citation_volume=30;,citation_journal_title=Journal of Applied Phycology;">
<meta name="citation_reference" content="citation_title=The environmental cost of animal source foods;,citation_abstract=We reviewed 148 assessments of animal source food ( ASF ) production for livestock, aquaculture, and capture fisheries that measured four metrics of environmental impact (energy use, greenhouse‐gas emissions, release of nutrients, and acidifying compounds) and standardized these per unit of protein production. We also examined additional literature on freshwater demand, pesticide use, and antibiotic use. There are up to 100‐fold differences in impacts between specific products and, in some cases, for the same product, depending on the production method being used. The lowest impact production methods were small pelagic fisheries and mollusk aquaculture, whereas the highest impact production methods were beef production and catfish aquaculture. Many production methods have not been evaluated, limiting our analysis to the range of studies that have been published. Regulatory restrictions on ASF production methods, as well as consumer guidance, should consider the relative environmental impact of these systems, since, currently, there appears to be little relationship between regulatory restrictions and impact in most developed countries.;,citation_author=Ray Hilborn;,citation_author=Jeannette A. Banobi;,citation_author=Stephen J. Hall;,citation_author=Teressa Pucylowski;,citation_author=Timothy E. Walsworth;,citation_publication_date=2018-08;,citation_cover_date=2018-08;,citation_year=2018;,citation_issue=6;,citation_doi=https://doi.org/ 10.1002/fee.1822;,citation_volume=16;,citation_journal_title=Frontiers in Ecology and the Environment;">
<meta name="citation_reference" content="citation_title=Global fishmeal and fish-oil supply: Inputs, outputs and markets.;,citation_abstract=Recent data on fishmeal and fish-oil supply are presented identifying key producer countries and raw material sources and distinguishing between whole fish and by-products. The conversion of these raw materials into marine ingredients is discussed and global volumes presented. This is followed by a summary of the main countries using these marine ingredients over recent years. Uses of fishmeal and fish-oil by market segment are then presented. From this, a global mass balance of inputs and outputs is derived which allows the calculation of the input-to-output ratios (fish in:fish out; FIFO) for the main aquaculture production types to be made. Current areas of focus by the industry include the need to demonstrate sustainable practice, more strategic use of marine ingredients, greater use of fishery and land-animal by-products as well as vegetable substitutes, and novel sources of essential omega-3 fats, notably the long-chain polyunsaturated fatty acids, eicosapentaenoic (EPA) and docosahexaenoic (DHA) acids. Implications are drawn for future supply prospects of fishmeal and fish-oil and their future role in aquaculture, agriculture and human health.;,citation_author=Christopher J. Shepherd;,citation_author=A. J. Jackson;,citation_publication_date=2013-10;,citation_cover_date=2013-10;,citation_year=2013;,citation_issue=4;,citation_doi=https://doi.org/ 10.1111/jfb.12224;,citation_volume=83;,citation_journal_title=Journal of Fish Biology;">
<meta name="citation_reference" content="citation_title=Insect larvae meal as an alternative source of nutrients in the diet of Atlantic salmon (&amp;amp;amp;lt;i&amp;gt;Salmo salar&amp;lt;/i&amp;gt;) postsmolt;,citation_abstract=Two insect meal (IM) products were tested as ingredient in diets for Atlantic salmon weighing ~250g. The modern control diet contained 200gkg−1 fish meal (FM100), which was stepwise replaced by insect meal A (IM A) as a 25% (A25), 50% (A50) or 100% (A100) FM replacement or insect meal B (IM B) as a 25% (B25) or 100% (B100) FM replacement. Selected nutrient parameters and contaminants were measured in the diets. After 15weeks, fish length and weight were recorded and visceral and hepatic indices calculated. Fatty acid and amino acid digestibility was calculated. Morphology of the liver, kidney, mid- and hind intestine was evaluated. Finally, a sensory testing of the fillets was conducted. A25, A50 and A100 performed equally well as the FM100 diet. The feed intake decreased moderately with increasing IM A inclusion; however, food conversion ratio decreased, resulting in an equal net growth of the fish. Histology did not show any differences between any of the dietary groups, and sensory testing of fillets from FM100, A100 and B25 did not reveal any significant differences in odour, flavour/taste or texture between groups. IM A was suitable to replace up to 100% of dietary FM, whilst IM B did not perform equally well.;,citation_author=Erik-Jan Lock;,citation_author=Tarique Arsiwalla;,citation_author=Rune Waagbø;,citation_publication_date=2016-12;,citation_cover_date=2016-12;,citation_year=2016;,citation_issue=6;,citation_doi=https://doi.org/ 10.1111/anu.12343;,citation_volume=22;,citation_journal_title=Aquaculture Nutrition;">
<meta name="citation_reference" content="citation_title=Utilisation of feed resources in production of Atlantic salmon (&amp;amp;amp;lt;i&amp;gt;Salmo salar&amp;lt;/i&amp;gt;) in Norway;,citation_abstract=In 1990, 90% of the ingredients in Norwegian salmon feed were of marine origin, whereas in 2013 only around 30%. The contents of fish meal and fish oil in the salmon feed were 18% and 11%, respectively, in 2013. Between 2010 and 2013, salmon production in Norway increased by 30%, but due to a lower inclusion of marine ingredientsinthediet,thetotalamountofmarineingredientsusedforsalmonfeedproductionwasreducedfrom 544,000 to 466,000 tonnes. Norwegian salmon farming consumed 1.63 million tonnes of feed ingredients in 2012, containing close to 40 million GJ of energy, 580,000 tonnes of protein and 530,000 tonnes of lipid. 1.26 million tonnes of salmon was produced. Assuming an edible yield of 65%, 820,000 tonnes of salmon fillet, containing 9.44 million GJ, and 156,000 tonnes of protein were produced. The retentions of protein and energy inthe edibleproductin2012were 27% and24%,respectively. Ofthe43,000 tonnes ofEPAand DHAinthesalmon feed in 2012, around 11,000 tonnes were retained in the edible part of salmon. The retentions of EPA and DHA were 46% in whole salmon and 26% in fillets, respectively. The fish in/fish out ratio (FIFO) measures the amount of fish meal and fish oil that is used to produce one weight equivalent of farmed fish back to wild fish weight equiva- lents, and the forage fish dependency ratio (FFDR) is the amount of wild caught fish used to produce the amount of fish meal and fish oil required to produce 1 kg of salmon. From 1990 to 2013, the forage fish dependency ratio for fish meal decreased from 4.4 to 0.7 in Norwegian salmon farming. However, weight-to-weight ratios such as FIFO and FFDR do not account for the different nutrient contents in the salmon product and in the forage fish used for fish meal and fish oil production. Marine nutrient dependency ratios express the amount of marine oil and protein required to produce 1 kg of salmon oil and protein. In 2013, 0.7 kg of marine protein was used to produce 1 kg of salmon protein, so the Norwegian farmed salmon is thus a net producer of marine protein.;,citation_author=Trine Ytrestøyl;,citation_author=Turid Synnøve Aas;,citation_author=Torbjørn Åsgård;,citation_publication_date=2015-11;,citation_cover_date=2015-11;,citation_year=2015;,citation_doi=https://doi.org/ 10.1016/j.aquaculture.2015.06.023;,citation_volume=448;,citation_journal_title=Aquaculture;">
<meta name="citation_reference" content="citation_title=Potential of insect-based diets for Atlantic salmon (&amp;amp;amp;lt;i&amp;gt;Salmo salar&amp;lt;/i&amp;gt;);,citation_abstract=In the present study, we aimed to assess the effect of dietary insect meal (IM) and insect oil (IO) on growth performance, body composition and nutrient digestibility of freshwater reared Atlantic salmon. The IM and IO were produced from black soldier fly larvae ( Hermetia illucens , L.; BSF) that had been grown on (1) media containing organic waste streams, or on (2) media partially containing seaweed ( Ascophyllum nodosum ). The feeding trial of the current study followed a factorial 2×3 way-ANOVA experimental design with six dietary groups of Atlantic salmon fed diets with insect-derived ingredients for 8weeks. A typical industrial diet, with protein from fish meal and soy protein concentrate (SPC) (50:50) and lipids from fish oil and vegetable oil (33:66), was fed to a positive control group. Five experimental diets were formulated, where 85% of the dietary protein was replaced by IM and/or all the vegetable oil was replaced by IO (IM from insects grown on media 1, IO from insects grown on either media 1 (IO1) or media 2 (IO2)). Replacing the dietary fish meal and SPC with insect protein significantly reduced the apparent digestibility coefficients (ADC) of protein, lipid and all amino acids investigated, though remained highly digestible. There were, however, only small differences due to protein or lipid source in growth performance, and no effects of insect ingredients on feed intake or feed conversion ratio. Inclusion of IM-based diets significantly increased both hepatosomatic index and visceral somatic index of Atlantic salmon. Proteinase activity in the intestine was not affected by dietary inclusion of BSF larvae meal, while leucine aminopeptidase activity was lower in fish fed with insect ingredients than the control group. Whole-body protein, lipid, amino acids and minerals contents were not affected by protein or lipid source. In general, this study showed that protein meal and oil from BSF larvae hold a great potential as a source of nutrients for Atlantic salmon.;,citation_author=Ikram Belghit;,citation_author=Nina S. Liland;,citation_author=Rune Waagbø;,citation_author=Irene Biancarosa;,citation_author=Nicole Francesca Pelusio;,citation_author=Yanxian Li;,citation_author=Åshild Krogdahl;,citation_author=Erik-Jan Lock;,citation_publication_date=2018-04;,citation_cover_date=2018-04;,citation_year=2018;,citation_doi=https://doi.org/ 10.1016/j.aquaculture.2018.03.016;,citation_volume=491;,citation_journal_title=Aquaculture;">
<meta name="citation_reference" content="citation_title=The sustainability conundrum of fishmeal substitution by plant ingredients in Shrimp Feeds;,citation_abstract=Aquaculture is central in meeting expanding global demands for shrimp consumption. Consequently, increasing feed use is mainly responsible for the overall environmental impact of aquaculture production. Significant amounts of fishmeal are included in shrimp diets, causing dependency on finite marine resources. Driven by economic incentives, terrestrial plant ingredients are widely viewed as sustainable alternatives. Incremental fishmeal substitution by plant ingredients in shrimp feed was modeled and effects on marine and terrestrial resources such as fish, land, freshwater, nitrogen, and phosphorus were assessed. We find that complete substitution of 20–30% fishmeal totals could lead to increasing demand for freshwater (up to 63%), land (up to 81%), and phosphorus (up to 83%), while other substitution rates lead to proportionally lower impacts. These findings suggest additional pressures on essential agricultural resources with associated socio-economic and environmental effects as a trade-off to pressures on finite marine resources. Even though the production of shrimp feed (or aquafeed in general) utilizes only a small percentage of the global crop production, the findings indicate that the sustainability of substituting fishmeal by plant ingredients should not be taken for granted, especially since aquaculture has been one of the fastest growing food sectors. Therefore, the importance of utilizing by-products and novel ingredients such as microbial biomass, algae, and insect meals in mitigating the use of marine and terrestrial resources is discussed.;,citation_author=Wesley Malcorps;,citation_author=Björn Kok;,citation_author=Mike Land;,citation_author=Maarten Fritz;,citation_author=Davy Doren;,citation_author=Kurt Servin;,citation_author=Paul Heijden;,citation_author=Roy Palmer;,citation_author=Neil Alexander Auchterlonie;,citation_author=Max Rietkerk;,citation_author=Maria João Santos;,citation_author=Simon J. Davies;,citation_publication_date=2019-02;,citation_cover_date=2019-02;,citation_year=2019;,citation_issue=4;,citation_doi=10.3390/su11041212;,citation_volume=11;,citation_journal_title=Sustainability;">
<meta name="citation_reference" content="citation_title=Most fish destined for fishmeal production are food-grade fish;,citation_abstract=Marine fisheries target and catch fish both for direct human consumption (DHC) as well as for fishmeal and fish oil, and other products. We derived the fractions used for each for 1950–2010 by fishing country, and thus provide a factual foundation for discussions of the optimal use of fisheries resources. From 1950 to 2010, 27% (~20 million tonnes annually) of globally reconstructed marine fisheries landings were destined for uses other than DHC. Importantly, 90% of fish destined for uses other than DHC are food-grade or prime food-grade fish, while fish without a ready market for DHC make up a much smaller proportion. These findings have implications for how we are using fish to feed ourselves or, more appropriately, how we are not using fish to feed ourselves.;,citation_author=Tim Cashion;,citation_author=Frédéric Le Manach;,citation_author=Dirk Zeller;,citation_author=Daniel Pauly;,citation_publication_date=2017-09;,citation_cover_date=2017-09;,citation_year=2017;,citation_issue=5;,citation_doi=https://doi.org/ 10.1111/faf.12209;,citation_volume=18;,citation_journal_title=Fish and Fisheries;">
<meta name="citation_reference" content="citation_title=Global overview on the use of fish meal and fish oil in industrially compounded aquafeeds: Trends and future prospects;,citation_abstract=The finfish and crustacean aquaculture sector is still highly dependent upon marine capture fisheries for sourcing key dietary nutrient inputs, including fish meal and fish oil. This dependency is particularly strong within compound aquafeeds for farmed carnivorous finfish species and marine shrimp. Results are presented concerning the responses received from a global survey conducted between December 2006 and October 2007 concerning the use of fish meal and fish oil within compound aquafeeds using a questionnaire sent to over 800 feed manufacturers, farmers, researchers, fishery specialists, and other stakeholders in over 50 countries. On the basis of the responses received, it is estimated that in 2006 the aquaculture sector consumed 3724 thousand tonnes of fish meal (68.2% total global fish meal production in 2006) and 835 thousand tonnes of fish oil (88.5% total reported fish oil production in 2006), or the equivalent of 16.6 million tonnes of small pelagic forage fish (using a wet fish to fish meal processing yield of 22.5% and wet fish to fish oil processing yield of 5%) with an overall fish-in fish-out ratio of 0.70. At a species-group level, calculation of small pelagic forage fish input per unit of farmed fish or crustacean output showed steadily decreasing fish-in fish-out ratios for all cultivated species from 1995 to 2006, with decreases being most dramatic for carnivorous fish species such as salmon (decreasing from 7.5 to 4.9 from 1995 to 2006), trout (decreasing from 6.0 to 3.4), eel (decreasing from 5.2 to 3.5), marine fish (decreasing from 3.0 to 2.2) and to a lesser extent shrimp (decreasing by 1.9 to 1.4 from 1995 to 2006. Net fish producing species in 2006 (with fish-in fish-out ratios below 1), included herbivorous and omnivorous finfish and crustacean species, including non-filter feeding Chinese carp (0.2), milkfish (0.2), tilapia (0.4), catfish (0.5), and freshwater crustaceans (0.6). On the basis of increasing global fish meal and fish oil costs, it is predicted that dietary fish meal and fish oil inclusion levels within compound aquafeeds will decrease in the long term, with fish meal and fish oil usage increasingly being targeted for use as a high value specialty feed ingredient for use within higher value starter, finisher and broodstock feeds, and by so doing extending supply of these much sought after and limited feed ingredient commodities.;,citation_author=Albert G. J. Tacon;,citation_author=Marc Metian;,citation_publication_date=2008-12;,citation_cover_date=2008-12;,citation_year=2008;,citation_issue=1;,citation_doi=https://doi.org/10.1016/j.aquaculture.2008.08.015;,citation_volume=285;,citation_journal_title=Aquaculture;">
<meta name="citation_reference" content="citation_title=Environmental impacts of plant-based salmonid diets at feed and farm scales;,citation_abstract=The environmental consequences of replacing fish meal and fish oil with plant-based sources in salmonid feeds were investigated using Life Cycle Assessment (LCA). Two scenarios of Atlantic salmon (Salmo salar) and rainbow trout (Oncorhynchus mykiss) farming were compared. The first scenario used a Standard Diet (STD) with high levels of fish meal and fish oil, and the second a Low Marine-Fishery-Capture Diet (LFD) in which fish meal and fish oil were replaced by plant protein and oil sources. Environmental impacts assessed were: acidification, eutrophication, climate change, terrestrial ecotoxicity, net primary production use (biotic resource use), water use, land occupation and total cumulative energy demand. The assessment confirmed the substantial contribution of feed to the environmental burdens of fish production and the LFD scenario led to a significant decrease in biotic resource use compared to the STD scenario with the same total energy demand. Environmental impacts of feeds depended highly on the geographic origins of feed ingredients from fishery (e.g., fish oil from Norway or Peru) and from terrestrial agricultural crop species (e.g., palm oil or rapeseed oil). This study demonstrated the importance of a multicriteria method to give stakeholders the most accurate information on the potential consequences of replacing fishery products with plant-based sources in aquafeeds.;,citation_author=Joachim Boissy;,citation_author=Joël Aubin;,citation_author=Abdeljalil Drissi;,citation_author=Hayo M. G. Werf;,citation_author=Gordon J. Bell;,citation_author=Sadasivam Kaushik;,citation_publication_date=2011-11;,citation_cover_date=2011-11;,citation_year=2011;,citation_doi=https://doi.org/ 10.1016/j.aquaculture.2011.08.033;,citation_volume=321;,citation_journal_title=Aquaculture;">
<meta name="citation_reference" content="citation_title=Nutritional Attributes, Substitutability, Scalability, and Environmental Intensity of an Illustrative Subset of Current and Future Protein Sources for Aquaculture Feeds: Joint Consideration of Potential Synergies and Trade-offs;,citation_abstract=Aquaculture is anticipated to play an increasingly important role in global food security because it may represent one of the best opportunities to increase the availability of healthy animal protein in the context of resource and environmental constraints. However, the growth and sustainability of the aquaculture industry faces important bottlenecks with respect to feed resources, which may be derived from diverse sources. Here, using a small but representative subset of potential aquafeed inputs (which we selected to highlight a range of relevant attributes), we review a core suite of considerations that need to be accommodated in concert in order to overcome key bottlenecks to the continued development and expansion of the aquaculture industry. Specifically, we evaluate the nutritional attributes, substitutability, scalability, and resource and environmental intensity of each input. On this basis, we illustrate a range of potential synergies and trade-offs within and across attributes that are characte...;,citation_author=Nathan Pelletier;,citation_author=Dane H. Klinger;,citation_author=Neil A. Sims;,citation_author=Janice Renee Yoshioka;,citation_author=Janice Renee Yoshioka;,citation_author=John N. Kittinger;,citation_publication_date=2018-04;,citation_cover_date=2018-04;,citation_year=2018;,citation_issue=10;,citation_doi=https://doi.org/ 10.1021/acs.est.7b05468;,citation_volume=52;,citation_journal_title=Environmental Science &amp;amp;amp; Technology;">
<meta name="citation_reference" content="citation_title=The Future of Aquatic Protein: Implications for Protein Sources in Aquaculture Diets;,citation_abstract=Approximately 70% of the aquatic-based production of animals is fed aquaculture, whereby animals are provided with high-protein aquafeeds. Currently, aquafeeds are reliant on fish meal and fish oil sourced from wild-captured forage fish. However, increasing use of forage fish is unsustainable and, because an additional 37.4 million tons of aquafeeds will be required by 2025, alternative protein sources are needed. Beyond plant-based ingredients, fishery and aquaculture byproducts and insect meals have the greatest potential to supply the protein required by aquafeeds over the next 10–20 years. Food waste also has potential through the biotransformation and/or bioconversion of raw waste materials, whereas microbial and macroalgal biomass have limitations regarding their scalability and protein content, respectively. In this review, we describe the considerable scope for improved efficiency in fed aquaculture and discuss the development and optimization of alternative protein sources for aquafeeds to ensure a socially and environmentally sustainable future for the aquaculture industry.;,citation_author=Katheline Hua;,citation_author=Jennifer M. Cobcroft;,citation_author=Andrew J. Cole;,citation_author=Kelly Condon;,citation_author=Dean R. Jerry;,citation_author=Arnold Mangott;,citation_author=Christina Praeger;,citation_author=Matthew J. Vucko;,citation_author=Chaoshu Zeng;,citation_author=Kyall R. Zenger;,citation_author=Jan M. Strugnell;,citation_publication_date=2019-11;,citation_cover_date=2019-11;,citation_year=2019;,citation_issue=3;,citation_doi=https://doi.org/ 10.1016/j.oneear.2019.10.018;,citation_volume=1;">
<meta name="citation_reference" content="citation_title=From Cradle to Plate: Analysing the Life Cycle Sustainability of Fish Feed Composition;,citation_abstract=Sustainability is defined as balancing environmental, economic and social factors, and various methodologies and tools are available to assess sustainability across sectors and scales. The demand for food has increased due to the increase in the population and the consumption of aquatic food in the world has increased significantly and is growing by an average of 3 % per year, while the population is growing by 1.6 % per year. As the aquaculture industry expands worldwide, it is important to consider the environmental impact of the industry and choose environmentally friendly alternatives to reduce its impact. The aim of this study is to assess the composition of five different fish feeds based on environmental, economic and social parameters using life cycle analysis (LCA), life cycle costing (LCC), social life cycle analysis (S-LCA) as well as technical considerations. The impact of alternatives to the main feed ingredients was analysed separately, while the development of fish feed focused on the protein source used in the feed and the oil used, as halieutic resources are used as raw material in their production, and alternatives are being considered. The best alternatives, considering all four dimensions, were the alternatives where fishmeal protein was partially replaced by Black Solder fly and Yellow Mealworm protein, as the proportion replaced is different for each alternative. By evaluating fish feed along several dimensions, the aim is to improve fish welfare while reducing the environmental impact of feed production.;,citation_author=Beate Zlaugotne;,citation_author=J. Pubule;,citation_publication_date=2024-10;,citation_cover_date=2024-10;,citation_year=2024;,citation_doi=https://doi.org/ 10.2478/rtuect-2024-0053;,citation_journal_title=Environmental and Climate Technologies;">
<meta name="citation_reference" content="citation_title=Life cycle assessment of aquafeed ingredients;,citation_abstract=This study performs an exploratory comparative evaluation of various animal and vegetable protein and lipid sources, used as feed in the aquaculture industry. The ingredients studied include fishmeal (FM) and fish oil (FO) from fisheries by-products, meal and fat by-products from poultry slaughter, FM and FO from Peruvian anchovy capture, and soybean meal and oil. The boundaries studied include the production or capture, the ingredient processing unit and the transport to the unit that processes the ingredients into aquafeeds in Portugal.;,citation_author=Catarina Basto Silva;,citation_author=Luisa M. P. Valente;,citation_author=Elisabete Matos;,citation_author=Miguel Brandão;,citation_author=Belmira Neto;,citation_publication_date=2018-05;,citation_cover_date=2018-05;,citation_year=2018;,citation_issue=5;,citation_doi=https://doi.org/ 10.1007/s11367-017-1414-8;,citation_volume=23;,citation_journal_title=International Journal of Life Cycle Assessment;">
<meta name="citation_reference" content="citation_title=Designing environmentally efficient aquafeeds through the use of multicriteria decision support tools;,citation_abstract=Aquaculture is the fastest growing food production system and the sector accounts for more than half of all fish consumed. Its potential as a sustainable food source has been recognised within the EU Farm to Fork strategy and by the targeting of EU Green Deal research funds. Aquafeed is the primary source of environmental cost in farmed finfish and shrimp life cycle assessments (LCA) and thus, emerging ingredients have a key role to play in increasing its sustainability. This paper proposes that the development and &amp;amp;amp;quot;design&amp;quot; of feed compositions can be disrupted by innovative approaches that simultaneously optimise – via multi-criteria analysis – ingredient selection through leveraging economic, nutritional, life cycle cost and environmental datasets. This would be supported by the ongoing collection and curation of these same resources. Further, integration of these disparate resources can enable advanced ingredient traceability. The outcome of such an approach would greatly contribute to increasingly efficient aquafeed production.;,citation_author=Ronan Cooney;,citation_author=Alex H. L. Wan;,citation_author=Fearghal O’Donncha;,citation_author=Eoghan Clifford;,citation_publication_date=2021-05;,citation_cover_date=2021-05;,citation_year=2021;,citation_doi=https://doi.org/ 10.1016/j.coesh.2021.100276;,citation_volume=23;,citation_journal_title=Current Opinion in Environmental Science &amp;amp; Health;">
<meta name="citation_reference" content="citation_title=Understanding and modelling the aquafeed extrusion process for Atlantic salmon feed with an empirical model;,citation_abstract=An empirical model was applied to quantitatively model the effects of extrusion variables and recipe compositions on quality of feed pellets. Lab extrusion trials were performed with an Atlantic salmon feed recipe with using a central component design (CCD). Comparing with a coated commercial feed made with the same recipe, the lab extrusion pellets had similar bulk density, hardness, and durability properties. A Response Surface Method (RSM) was also used to regress the extrusion trial data and compared with the empirical model results. The respective Absolute Average Deviation (AAD%) values of the empirical model and RSM regression were 4.1% and 1.8% for pellet bulk density, 7.4% and 4.9% for pellet hardness, 10.4% and 10.0% for pellet oil absorption, 13.7% and 8.3% for pellet water stability, and 11.3% and 5.5% for pellet durability. Using the model coefficients obtained from the extrusion results, the empirical model can partly extrapolate the pellet qualities of an additional extruded recipe, which has a slightly adjusted wheat gluten and wheat. The model performance was evaluated based on. R2, RSME (root-mean-square deviation), AAD% of the empirical model was better than the RSM model to predict the effects of extrusion variables on pellet quality parameters. The effects of the extrusion process variables and recipe composition on pellet quality parameters are predicted by the empirical model. Its demonstrates that the empirical model is a new method to understand extrusion trial results in different extrusion systems and recipes without changing its mathematical form of expression, which gives a common reference to compare an extruded recipe in different systems and has practical engineering applications.;,citation_author=Hongyan Cheng;,citation_author=Yiqing Guo;,citation_author=S.J. Højberg;,citation_author=Kyla Meagan Zatti;,citation_author=Aberham Hailu Feyissa;,citation_publication_date=2023-10;,citation_cover_date=2023-10;,citation_year=2023;,citation_doi=https://doi.org/ 10.1016/j.anifeedsci.2023.115752;,citation_volume=304;,citation_journal_title=Animal Feed Science and Technology;">
<meta name="citation_reference" content="citation_title=Fishmeal partial substitution within aquafeed formulations: Life cycle assessment of four alternative protein sources;,citation_abstract=Fed aquaculture has long been based on fishmeal (FM) as the main protein source for carnivorous species, but when its demand and price began increasing, both researchers and the industry started to search for alternative protein sources to meet the challenge of securing aquafeed. Consequently, this study has recourse to life cycle assessment (LCA) methodology to identify any critical points and improvement strategies in the current production of four partial substitutes for FM namely dried microalgae biomass from Tetraselmis suecica (DMB_TETRA) and Tisochrysis lutea (DMB_TISO); insect meal (IM) from Hermetia illucens larvae; and poultry by-product meal (PBM). System boundaries are from the cradle to the mill gate, thus including the production phase (and the related upstream activities) and the subsequent biomass processing into unpackaged dried meal. One tonne of protein content was chosen as functional unit. The inventory was based on foreground data provided by the industry and complemented by background data sourced from the Ecoinvent v 3.4 and Agribalyse® v 1.3 LCI databases. The environmental effects were assessed considering five impacts namely global warming (kg CO2 eq.), acidification (kg SO2 eq.) and eutrophication (kg PO43− eq.) estimated via the CML-IA method, plus cumulative energy use (MJ) and water use (m3m−2month−1). Two alternative scenarios per production chain were considered in order to increase the robustness of the results. The performance-based ranking indicated the PBM and IM scenarios as the most sustainable options. Both microalgal systems scored the worst performances in four impacts out of five, with eutrophication impact as the only exception. The nutrients provided to poultry and insects (i.e. the feed) as well as to microalgae (the carbon source and the fertilizers) were the main contributors to impacts, together with energy consumption. Despite being cultivated with identical technologies, the two microalgae showed different performances due to their different annual yields and to different consumable goods, water and energy consumptions. The results provided a ranking of these four partial FM substitutes and allowed to make useful considerations on how to improve their environmental sustainability. To this regard, the impacts of IM and DMB production could be reduced by improving nutrient efficiency and reducing energy needs. On the other hand, PBM production is already optimized and is not expected to change substantially in the future years.;,citation_author=Silvia Maiolo;,citation_author=Giuliana Parisi;,citation_author=Natascia Biondi;,citation_author=Fernando Lunelli;,citation_author=Emilio Tibaldi;,citation_author=Roberto Pastres;,citation_publication_date=2020-08;,citation_cover_date=2020-08;,citation_year=2020;,citation_issue=8;,citation_doi=https://doi.org/ 10.1007/s11367-020-01759-z;,citation_volume=25;,citation_journal_title=International Journal of Life Cycle Assessment;">
<meta name="citation_reference" content="citation_title=Trends in Global Aquaculture and Aquafeed Production: 2000–2017;,citation_abstract=This research analyzes global aquaculture and aquafeed production trends from 2000 to 2017, highlighting the dominance of fish and crustacean production, particularly within developing countries. Key findings reveal that aquaculture production has surpassed capture fisheries, driven by significant growth rates in various species categories, including freshwater fish and marine shrimp, while also noting the economic implications and production value across different aquatic species.;,citation_author=Albert G. J. Tacon;,citation_publication_date=2020-01;,citation_cover_date=2020-01;,citation_year=2020;,citation_issue=1;,citation_doi=https://doi.org/ 10.1080/23308249.2019.1649634;,citation_volume=28;,citation_journal_title=Reviews in Fisheries Science &amp;amp;amp; Aquaculture;">
<meta name="citation_reference" content="citation_title=Fish as feed: Using economic allocation to quantify the Fish in : Fish out ratio of major fed aquaculture species;,citation_abstract=Efficiency assessments of marine ingredient use in aquaculture are required to fully understand their contribution to global seafood supply and their impacts on all UN Sustainable Development Goals. Fish In: Fish Out (FIFO) ratios have become the principal metric used to ensure aquaculture does not negatively impact wild fish stocks. However, several approaches have been advocated to calculate the FIFO ratio and there have been criticisms that the different approaches employed lead to over- or under- estimates of the dependence of aquaculture on marine ingredients. Critically, FIFO does not align with Life Cycle Assessment as a measure of other environmental impacts. In this paper we present an alternative method to calculate the FIFO ratio based on the principle of economic allocation (economic Fish In: Fish Out – eFIFO) as commonly used in Life Cycle Assessments. Economic allocation acts as a proxy for the nutritional value of ingredients and places higher importance on the more limiting co-products generated and their relative demand. Substitution of marine ingredients by alternate feed ingredients has significantly reduced the amount of fishmeal and fish oil in aquafeed formulations for most farmed fish species, resulting in a continually decreasing FIFO ratio. Results show that most aquaculture species groups assessed in this study are net producers of fish, while salmon and trout aquaculture are net neutral, producing as much fish biomass as is consumed. Overall, global fed-aquaculture currently produces three to four times as much fish as it consumes. Tracking historical prices of fish oil against fishmeal, the relative higher price of fish oil leads to relatively higher allocation of fish to fish oil compared to fishmeal. This leads to relatively higher eFIFO for species with high fish oil requirements.;,citation_author=Björn Kok;,citation_author=Wesley Malcorps;,citation_author=Michael F. Tlusty;,citation_author=Mahmoud Eltholth;,citation_author=Neil A Auchterlonie;,citation_author=David Little;,citation_author=Robert Harmsen;,citation_author=Richard Newton;,citation_author=Simon J. Davies;,citation_publication_date=2020-11;,citation_cover_date=2020-11;,citation_year=2020;,citation_fulltext_html_url=https://doi.org/10.1016/j.aquaculture.2020.735474;,citation_doi=https://doi.org/10.1016/j.aquaculture.2020.735474;,citation_volume=528;,citation_journal_title=Aquaculture;">
<meta name="citation_reference" content="citation_title=Avoiding the ecological limits of forage fish for fed aquaculture;,citation_abstract=Aquaculture is supporting demand and surpassing wild-caught seafood. Yet, most fed aquaculture species (finfish and crustacea) rely on wild-captured forage fish for essential fatty acids and micronutrients, an important but limited resource. As the fastest growing food sector in the world, fed aquaculture demand will eventually surpass ecological supply of forage fish, but when and how best to avoid this ecological boundary is unclear. Using global production data, feed use trends, and human consumption patterns, we show how combined actions of fisheries reform, reduced feed use by non-carnivorous aquaculture and agricultural species, and greater consistent inclusion of fish by-products in China-based production can circumvent forage fish limits by mid-century. However, we also demonstrate that the efficacies of such actions are diminished if global diets shift to more seafood-heavy (that is, pescatarian) diets and are further constrained by possible ecosystem-based fisheries regulations in the future. Long-term, nutrient-equivalent alternative feed sources are essential for more rapid and certain aquaculture sustainability.;,citation_author=Halley E. Froehlich;,citation_author=Nis S. Jacobsen;,citation_author=Timothy E. Essington;,citation_author=Tyler Clavelle;,citation_author=Benjamin S. Halpern;,citation_author=Chelsea M. Williams;,citation_author=Daniel J. Pondella;,citation_author=Douglas Dodds;,citation_author=Andrew J. Plater;,citation_author=Fengfeng Le;,citation_author=Jun Bo;,citation_publication_date=2018-06;,citation_cover_date=2018-06;,citation_year=2018;,citation_issue=6;,citation_doi=https://doi.org/10.1038/s41893-018-0077-1;,citation_volume=1;">
<meta name="citation_reference" content="citation_title=Fish oil replacement in finfish nutrition;,citation_abstract=Unsustainable fishing practices have placed a heavy emphasis on aquaculture to meet the global shortfalls in the supply of fish and seafood, which are commonly accepted as the primary source of health-promoting essential omega-3 (n-3 highly unsaturated fatty acids). However, dietary fish oil is required for the production of omega-3-rich farmed fish and this commodity, in a vicious circle, is at present derived solely from wild fisheries. Decreasing global availability coupled with the highly variable price of this resource has forced the aquaculture industry to investigate the possibilities of alternative dietary lipid sources. This review attempts to compile all principal information available regarding the effects of fish oil replacement for the diets of farmed finfish, analysing the findings using a comparative approach among different cultured fish species. The review initially focuses on the present situation with regard to the production, availability and main nutritional characteristics of fish oil and the principal alternative lipid sources (such as vegetable oils and animal fats). Following this, the effects of fish oil replacement in finfish nutrition on feed quality, fish performance, feed efficiency, fish lipid metabolism, final eating quality and related economic aspects are presented and discussed.;,citation_author=Giovanni M. Turchini;,citation_author=Bente E. Torstensen;,citation_author=Wing-Keong Ng;,citation_publication_date=2009-03;,citation_cover_date=2009-03;,citation_year=2009;,citation_issue=1;,citation_doi=https://doi.org/ 10.1111/j.1753-5131.2008.01001.x;,citation_volume=1;,citation_journal_title=Reviews in Aquaculture;">
<meta name="citation_reference" content="citation_title=Does aquaculture add resilience to the global food system;,citation_abstract=Aquaculture is the fastest growing food sector and continues to expand alongside terrestrial crop and livestock production. Using portfolio theory as a conceptual framework, we explore how current interconnections between the aquaculture, crop, livestock, and fisheries sectors act as an impediment to, or an opportunity for, enhanced resilience in the global food system given increased resource scarcity and climate change. Aquaculture can potentially enhance resilience through improved resource use efficiencies and increased diversification of farmed species, locales of production, and feeding strategies. However, aquaculture’s reliance on terrestrial crops and wild fish for feeds, its dependence on freshwater and land for culture sites, and its broad array of environmental impacts diminishes its ability to add resilience. Feeds for livestock and farmed fish that are fed rely largely on the same crops, although the fraction destined for aquaculture is presently small (∼4%). As demand for high-value fed aquaculture products grows, competition for these crops will also rise, as will the demand for wild fish as feed inputs. Many of these crops and forage fish are also consumed directly by humans and provide essential nutrition for low-income households. Their rising use in aquafeeds has the potential to increase price levels and volatility, worsening food insecurity among the most vulnerable populations. Although the diversification of global food production systems that includes aquaculture offers promise for enhanced resilience, such promise will not be realized if government policies fail to provide adequate incentives for resource efficiency, equity, and environmental protection.;,citation_author=Max Troell;,citation_author=Rosamond L. Naylor;,citation_author=Marc Métian;,citation_author=Malcolm C. M. Beveridge;,citation_author=Peter Tyedmers;,citation_author=Carl Folke;,citation_author=Kenneth J. Arrow;,citation_author=Scott Barrett;,citation_author=Anne-Sophie Crépin;,citation_author=Paul R. Ehrlich;,citation_author=Åsa Gren;,citation_author=Nils Kautsky;,citation_author=Simon A. Levin;,citation_author=Karine Nyborg;,citation_author=Henrik Österblom;,citation_author=Stephen Polasky;,citation_author=Marten Scheffer;,citation_author=Brian Walker;,citation_author=Tasos Xepapadeas;,citation_author=Aart Zeeuw;,citation_publication_date=2014-09;,citation_cover_date=2014-09;,citation_year=2014;,citation_issue=37;,citation_doi=https://doi.org/ 10.1073/pnas.1404067111;,citation_volume=111;,citation_journal_title=Proceedings of the National Academy of Sciences of the United States of America;">
<meta name="citation_reference" content="citation_title=Direct and indirect effects of climate and fishing on changes in coastal ecosystem services: A historical perspective from the North Sea;,citation_abstract=Humanity depends on the marine environment for a range of vital ecosystem services, at global (e.g. climate regulation), regional (e.g. commercial fisheries) and local scales (e.g. coastal defence and recreation). At the same time, marine ecosystems have been exploited for centuries, and many systems today are under stress from multiple sources. Recent studies have shown how both climate change and fishing have caused long-term changes in the marine environment. However, there is still poor understanding of how these changes influence change in coastal ecosystem services. In this paper, an integrated modelling approach is used to assess how the final delivery of marine ecosystem services to coastal communities is influenced by the direct and indirect effects of changes in ecosystem processes brought about by climate and human impacts, using fisheries of the North Sea region as a case study. Partial least squares path analysis is used to explore the relationships between drivers of change, marine ecosystem processes and services (landings). A simple conceptual model with four variables—climate, fishing effort, ecosystem process and ecosystem services—is applied to the English North Sea using historic ecological, climatic and fisheries time series spanning 1924–2010 to identify the multiple pathways that might exist. As expected, direct and indirect links between fishing effort, ecosystem processes and service provision were significant. However, links between climate and ecosystem processes were weak. This paper highlights how path analysis can be used for analysing long-term temporal links between ecosystem processes and services following a simplified pathway.;,citation_author=Samiya A. Selim;,citation_author=Julia L. Blanchard;,citation_author=Jacob Bedford;,citation_author=Thomas J. Webb;,citation_publication_date=2016-02;,citation_cover_date=2016-02;,citation_year=2016;,citation_issue=2;,citation_doi=https://doi.org/ 10.1007/s10113-014-0635-7;,citation_volume=16;,citation_journal_title=Regional Environmental Change;">
<meta name="citation_reference" content="citation_title=Global adoption of novel aquaculture feeds could substantially reduce forage fish demand by 2030;,citation_abstract=With the global supply of forage fish at a plateau, fed aquaculture must continue to reduce dependence on fishmeal and oil in feeds to ensure sustainable sector growth. The use of novel aquaculture feed ingredients is growing, but their contributions to scalable and sustainable aquafeed solutions are unclear. Here, we show that global adoption of novel aquafeeds could substantially reduce aquaculture’s forage fish demand by 2030, maintaining feed efficiencies and omega-3 fatty acid profiles. We combine production data, scenario modelling and a decade of experimental data on forage fish replacement using microalgae, macroalgae, bacteria, yeast and insects to illustrate how reducing future fish oil demand, particularly in high-value species such as salmonids, will be key for the sustainability of fed aquaculture. However, considerable uncertainties remain surrounding novel feed efficacy across different life-cycle stages and taxa, and various social, environmental, economic and regulatory challenges will dictate their widespread use. Yet, we demonstrate how even limited adoption of novel feeds could aid sustainable aquaculture growth, which will become increasingly important for food security. Novel aquaculture feeds are rapidly developing, but their contributions to sustainable industry growth are unknown. Cottrell et al. model feed efficiency and fatty acid profiles, showing that replacing forage fish with novel feed ingredients could strengthen aquaculture’s role in global food security.;,citation_author=Richard S. Cottrell;,citation_author=Julia L. Blanchard;,citation_author=Benjamin S. Halpern;,citation_author=Marc Metian;,citation_author=Halley E. Froehlich;,citation_publication_date=2020-05;,citation_cover_date=2020-05;,citation_year=2020;,citation_issue=5;,citation_doi=10.1038/s43016-020-0078-x;,citation_volume=1;,citation_journal_title=Nature Food;">
<meta name="citation_reference" content="citation_title=Linked sustainability challenges and trade-offs among fisheries, aquaculture and agriculture.;,citation_abstract=Fisheries and aquaculture make a crucial contribution to global food security, nutrition and livelihoods. However, the UN Sustainable Development Goals separate marine and terrestrial food production sectors and ecosystems. To sustainably meet increasing global demands for fish, the interlinkages among goals within and across fisheries, aquaculture and agriculture sectors must be recognized and addressed along with their changing nature. Here, we assess and highlight development challenges for fisheries-dependent countries based on analyses of interactions and trade-offs between goals focusing on food, biodiversity and climate change. We demonstrate that some countries are likely to face double jeopardies in both fisheries and agriculture sectors under climate change. The strategies to mitigate these risks will be context-dependent, and will need to directly address the trade-offs among Sustainable Development Goals, such as halting biodiversity loss and reducing poverty. Countries with low adaptive capacity but increasing demand for food require greater support and capacity building to transition towards reconciling trade-offs. Necessary actions are context-dependent and include effective governance, improved management and conservation, maximizing societal and environmental benefits from trade, increased equitability of distribution and innovation in food production, including continued development of low input and low impact aquaculture.;,citation_author=Julia L. Blanchard;,citation_author=Reg Watson;,citation_author=Elizabeth A. Fulton;,citation_author=Richard S. Cottrell;,citation_author=Kirsty L. Nash;,citation_author=Andrea Bryndum-Buchholz;,citation_author=Matthias Büchner;,citation_author=David A. Carozza;,citation_author=William W. L. Cheung;,citation_author=Joshua Elliott;,citation_author=Lindsay Davidson;,citation_author=Nicholas K. Dulvy;,citation_author=John P. Dunne;,citation_author=Tyler D. Eddy;,citation_author=Eric D. Galbraith;,citation_author=Heike K. Lotze;,citation_author=Olivier Maury;,citation_author=Christoph Müller;,citation_author=Derek P. Tittensor;,citation_author=Simon Jennings;,citation_publication_date=2017-08;,citation_cover_date=2017-08;,citation_year=2017;,citation_issue=9;,citation_doi=https://doi.org/ 10.1038/s41559-017-0258-8;,citation_volume=1;,citation_journal_title=Nature Ecology and Evolution;">
<meta name="citation_reference" content="citation_title=Modelling the effects of ration on individual growth of &amp;amp;amp;lt;i&amp;gt;Oncorhynchus tshawytscha&amp;lt;/i&amp;gt; under controlled conditions;,citation_abstract=Fed aquaculture is one of the fastest-growing and most valuable food production industries in the world. The efficiency with which farmed fish convert feed into biomass influences both environmental impact and economic revenue. Salmonid species, such as king salmon (Oncorhynchus tshawytscha), exhibit high levels of plasticity in vital rates such as feed intake and growth rates. Accurate estimations of individual variability in vital rates are important for production management. The use of mean trait values to evaluate feeding and growth performance can mask individual-level differences that potentially contribute to inefficiencies. Here, the authors apply a cohort integral projection model (IPM) framework to investigate individual variation in growth performance of 1625 individually tagged king salmon fed one of three distinct rations of 60%, 80%, and 100% satiation and tracked over a duration of 276 days. To capture the observed sigmoidal growth of individuals, they compared a nonlinear mixed-effects (logistic) model to a linear model used within the IPM framework. Ration significantly influenced several aspects of growth, both at the individual and at the cohort level. Mean final body mass and mean growth rate increased with ration; however, variance in body mass and feed intake also increased significantly over time. Trends in mean body mass and individual body mass variation were captured by both logistic and linear models, suggesting the linear model to be suitable for use in the IPM. The authors also observed that higher rations resulted in a decreasing proportion of individuals reaching the cohort’s mean body mass or larger by the end of the experiment. This suggests that, in the present experiment, feeding to satiation did not produce the desired effects of efficient, fast, and uniform growth in juvenile king salmon. Although monitoring individuals through time is challenging in commercial aquaculture settings, recent technological advances combined with an IPM approach could provide new scope for tracking growth performance in experimental and farmed populations. Using the IPM framework might allow the exploration of other size-dependent processes affecting vital rate functions, such as competition and mortality.;,citation_author=Alexandra Johne;,citation_author=Chris Carter;,citation_author=Simon Wotherspoon;,citation_author=Scott Hadley;,citation_author=Jane E. Symonds;,citation_author=Seumas P. Walker;,citation_author=Julia L. Blanchard;,citation_publication_date=2023-07;,citation_cover_date=2023-07;,citation_year=2023;,citation_issue=5;,citation_doi=https://doi.org/ 10.1111/jfb.15499;,citation_volume=103;,citation_journal_title=Journal of Fish Biology;">
<meta name="citation_reference" content="citation_title=Environmental footprints of farmed chicken and salmon bridge the land and sea;,citation_abstract=Food production, particularly of fed animals, is a leading cause of environmental degradation globally.1,2 Understanding where and how much environmental pressure different fed animal products exert is critical to designing effective food policies that promote sustainability.3 Here, we assess and compare the environmental footprint of farming industrial broiler chickens and farmed salmonids (salmon, marine trout, and Arctic char) to identify opportunities to reduce environmental pressures. We map cumulative environmental pressures (greenhouse gas emissions, nutrient pollution, freshwater use, and spatial disturbance), with particular focus on dynamics across the land and sea. We found that farming broiler chickens disturbs 9 times more area than farming salmon (∼924,000 vs. ∼103,500 km2) but yields 55 times greater production. The footprints of both sectors are extensive, but 95% of cumulative pressures are concentrated into &amp;amp;amp;lt;5% of total area. Surprisingly, the location of these pressures is similar (85.5% spatial overlap between chicken and salmon pressures), primarily due to shared feed ingredients. Environmental pressures from feed ingredients account for &amp;gt;78% and &amp;gt;69% of cumulative pressures of broiler chicken and farmed salmon production, respectively, and could represent a key leverage point to reduce environmental footprints. The environmental efficiency (cumulative pressures per tonne of production) also differs geographically, with areas of high efficiency revealing further potential to promote sustainability. The propagation of environmental pressures across the land and sea underscores the importance of integrating food policies across realms and sectors to advance food system sustainability.;,citation_author=Caitlin D. Kuempel;,citation_author=Melanie Frazier;,citation_author=Juliette Verstaen;,citation_author=Paul-Eric Rayner;,citation_author=Julia L. Blanchard;,citation_author=Richard S. Cottrell;,citation_author=Halley E. Froehlich;,citation_author=Jessica A. Gephart;,citation_author=Nis S. Jacobsen;,citation_author=Peter B. McIntyre;,citation_author=Marc Métian;,citation_author=Daniel Moran;,citation_author=Kirsty L. Nash;,citation_author=Johannes Többen;,citation_author=David Williams;,citation_author=Benjamin S. Halpern;,citation_publication_date=2023-03;,citation_cover_date=2023-03;,citation_year=2023;,citation_issue=5;,citation_doi=https://doi.org/ 10.1016/j.cub.2023.01.037;,citation_volume=33;,citation_journal_title=Current Biology;">
<meta name="citation_reference" content="citation_title=Vulnerability of blue foods to human-induced environmental change;,citation_abstract=Global aquatic or “blue” foods, essential to over 3.2 billion people, face challenges of maintaining supply in a changing environment while adhering to safety and sustainability standards. Despite the growing concerns over their environmental impacts, limited attention has been paid to how blue food production is influenced by anthropogenic environmental changes. Here we assess the vulnerability of global blue food systems to predominant environmental disturbances and predict the spatial impacts. Over 90% of global blue food production faces substantial risks from environmental change, with the major producers in Asia and the United States facing the greatest threats. Capture fisheries generally demonstrate higher vulnerability than aquaculture in marine environments, while the opposite is true in freshwater environments. While threats to production quantity are widespread across marine and inland systems, food safety risks are concentrated within a few countries. Identifying and supporting mitigation and adaptation measures in response to environmental stressors is particularly important in developing countries in Asia, Latin America and Africa where risks are high and national response capacities are low. These findings lay groundwork for future work to map environmental threats and opportunities, aiding strategic planning and policy development for resilient and sustainable blue food production under changing conditions.;,citation_author=Ling Cao;,citation_author=Benjamin S. Halpern;,citation_author=Max Troell;,citation_author=Richard L. Short;,citation_author=Cong Zeng;,citation_author=Ziyu Jiang;,citation_author=Y. Liu;,citation_author=Chengxuan Zou;,citation_author=Chunyu Liu;,citation_author=Shurong Liu;,citation_author=Xiangwei Liu;,citation_author=William W. L. Cheung;,citation_author=Richard S. Cottrell;,citation_author=Fabrice DeClerck;,citation_author=Stefan Gelcich;,citation_author=Jessica A. Gephart;,citation_author=Dakoury Godo-Solo;,citation_author=Jessie Ihilani Kaull;,citation_author=Fiorenza Micheli;,citation_author=Rosamond L. Naylor;,citation_author=Hanna J. Payne;,citation_author=Elizabeth R. Selig;,citation_author=U. Rashid Sumaila;,citation_author=Michelle Tigchelaar;,citation_publication_date=2023-06;,citation_cover_date=2023-06;,citation_year=2023;,citation_issue=10;,citation_doi=https://doi.org/ 10.1038/s41893-023-01156-y;,citation_volume=6;,citation_journal_title=Nature Sustainability;">
<meta name="citation_reference" content="citation_title=An R package for simulating growth and organic wastage in aquaculture farms in response to environmental conditions and husbandry practices;,citation_abstract=A new R software package, RAC, is presented. RAC allows to simulate the rearing cycle of 4 species, finfish and shellfish, highly important in terms of production in the Mediterranean Sea. The package works both at the scale of the individual and of the farmed population. Mathematical models included in RAC were all validated in previous works, and account for growth and metabolism, based on input data characterizing the forcing functions—water temperature, and food quality/quantity. The package provides a demo dataset of forcings for each species, as well as a typical set of husbandry parameters for Mediterranean conditions. The present work illustrates RAC main features, and its current capabilities/limitations. Three test cases are presented as a proof of concept of RAC applicability, and to demonstrate its potential for integrating different open products nowadays provided by remote sensing and operational oceanography.;,citation_author=Damiano Baldan;,citation_author=Erika M. D. Porporato;,citation_author=Roberto Pastres;,citation_author=Daniele Brigolin;,citation_publication_date=2018-05;,citation_cover_date=2018-05;,citation_year=2018;,citation_fulltext_html_url=https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0195732;,citation_issue=5;,citation_doi=https://doi.org/10.1371/journal.pone.0195732;,citation_issn=1932-6203;,citation_volume=13;,citation_journal_title=PLOS ONE;">
<meta name="citation_reference" content="citation_title=The origins of ingredients matter more than ingredient types for the environmental footprint of aquaculture feeds;,citation_abstract=Reduced dependence on wild-caught fish in favour of crop-based ingredients for feeds has been crucial for sustaining growth in fed aquaculture. However, the net environmental implications of displacing feed provisioning burdens from marine to terrestrial systems remain understudied with little attention to multiple environmental pressures or the importance of where ingredients are sourced from. Here we model the change in environmental footprint (a cumulative and spatial measure of greenhouse gas emissions, habitat disturbance, excess nutrient, and water consumption pressures) of shifting dependence from largely fish-based to plant-based ingredients in feeds for Atlantic salmon farming. We show that average differences exist between feeds in their cumulative and individual environmental pressures, but more importantly, the locations where feed raw materials are produced and processed drives far more variability in footprint within a feed than the typical variation between feeds. We demonstrate that responsible sourcing will be critical for sustainable feed production across all farming systems as the next generation of ingredients are developed.;,citation_author=Richard S. Cottrell;,citation_author=Benjamin S. Halpern;,citation_author=Marceau Cormery;,citation_author=Helen A. Hamilton;,citation_author=Gage Clawson;,citation_author=Beth Penrose;,citation_author=Louise Adams;,citation_author=John P. Bowman;,citation_author=Duncan D. Cameron;,citation_author=Chris G. Carter;,citation_author=Tom Fox-Smith;,citation_author=Scott Hadley;,citation_author=Alexandra Johne;,citation_author=Catriona Macleod;,citation_author=Sowdamini Sesha Prasad;,citation_author=Xintong Qiu;,citation_author=Camille A. White;,citation_author=Xin Zhan;,citation_author=Julia L. Blanchard;">
<meta name="citation_reference" content="citation_title=Multi-scale integrated evaluation of the sustainability of large-scale use of alternative feeds in salmon aquaculture;,citation_abstract=The steady increase in production volume of salmon aquaculture has sharpened concerns about its sustainability. In particular the production of salmon feed is a reason for concern given its reliance on scarce natural resources, such as wild fish captures. Multi-scale integrated analysis is put forward as a tool to anticipate the environmental and socio-economic impacts of large-scale implementation of alternative salmon feeds, considering both plant and insect sources as potential replacements of fish meal and fish oil. The proposed accounting framework, based on relational analysis across hierarchical levels, describes the patterns of required inputs using biophysical and economic variables. It also considers the inputs used by external systems for the production of imported feed, thus providing a coherent assessment of the sustainability of the production system in terms of feasibility, viability, and desirability. The analytical tool-kit is illustrated in conceptual terms and then applied to the Norwegian salmon aquaculture, both in diagnostic (describing the actual situation) and anticipatory mode (examining feed scenarios). Results are used in an exercise of quantitative story-telling to check the quality of the narratives currently shaping policy discussions on aquaculture. Quantitative story-telling is a heuristic approach aimed at checking the robustness of knowledge claims in face of uncertainty. It is concluded that rearing insects in the salmon feed production chain enlarges the option space of feed sources by opening up the possibility of using locally-produced seaweed and organic waste, but also raises the level of uncertainty with regard to the possible insurgence of negative side effects.;,citation_author=Juan J. Cadillo-Benalcazar;,citation_author=Mario Giampietro;,citation_author=Sandra G. F. Bukkens;,citation_author=Roger Strand;,citation_publication_date=2020-03;,citation_cover_date=2020-03;,citation_year=2020;,citation_doi=https://doi.org/ 10.1016/j.jclepro.2019.119210;,citation_issn=0959-6526;,citation_volume=248;,citation_journal_title=Journal of Cleaner Production;">
<meta name="citation_reference" content="citation_title=Continued Transitions from Fish Meal and Oil in Aquafeeds Require Close Attention to Biodiversity Trade-Offs;,citation_abstract=Feed constitutes a considerable portion of the environmental impact embedded in aquaculture production, making it crucial for enhancing aquaculture sustainability. While previous research has evaluated the environmental pressures of feed production, no spatially explicit global biodiversity assessment of animal feeds exists. Here, we assess biodiversity impacts on 54,628 marine and terrestrial species for two simplified but plausible Atlantic salmon feeds. We find widespread impact on both marine (~89%) and terrestrial (~71%) species, though the average magnitude of impact is small. Despite minimising wild-sourced fishmeal and oil by necessity, increased agricultural dependence for feed provisioning appears to have disproportionately increased impacts on terrestrial taxa. Our results provide key information for sourcing aquafeed to minimise impacts and optimise sustainability. As the aquaculture industry expands to feed billions more, a standardised and integrative approach to assess and potentially forecast the effects of feed on global biodiversity is essential for informed decision-making.;,citation_author=Gage Clawson;,citation_author=Julia L. Blanchard;,citation_author=Marceau Cormery;,citation_author=Beth Fulton;,citation_author=Benjamin S. Halpern;,citation_author=Helen A. Hamilton;,citation_author=Casey C. O’Hara;,citation_author=Richard S. Cottrell;,citation_publication_date=2024-11;,citation_cover_date=2024-11;,citation_year=2024;,citation_doi=https://doi.org/ 10.2139/ssrn.5030822;,citation_publisher=Social Science Research Network;">
<meta name="citation_reference" content="citation_title=Perspectives on aquaculture’s contribution to the Sustainable Development Goals for improved human and planetary health;,citation_abstract=The diverse aquaculture sector makes important contributions toward achieving the Sustainable Development Goals (SDGs)/Agenda 2030, and can increasingly do so in the future. Its important role for food security, nutrition, livelihoods, economies, and cultures is not clearly visible in the Agenda 21 declaration. This may partly reflect the state of development of policies for aquaculture compared with its terrestrial counterpart, agriculture, and possibly also because aquaculture production has historically originated from a few key hotspot regions/countries. This review highlights the need for better integration of aquaculture in global food system dialogues. Unpacking aquaculture’s diverse functions and generation of values at multiple spatiotemporal scales enables better understanding of aquaculture’s present and future potential contribution to the SDGs. Aquaculture is a unique sector that encompasses all aquatic ecosystems (freshwater, brackish/estuarine, and marine) and is also tightly interconnected with terrestrial ecosystems through, for example, feed resources and other dependencies. Understanding environmental, social, and economic characteristics of the multifaceted nature of aquaculture provides for more context-specific solutions for addressing both opportunities and challenges for its future development. This review includes a rapid literature survey based on how aquaculture links to the specific SDG indicators. A conceptual framework is developed for communicating the importance of context specificity related to SDG outcomes from different types of aquaculture. The uniqueness of aquaculture’s contributions compared with other food production systems are discussed, including understanding of species/systems diversity, the role of emerging aquaculture, and its interconnectedness with supporting systems. A selection of case studies is presented to illustrate: (1) the diversity of the aquaculture sector and what role this diversity can play for contributions to the SDGs, (2) examples of methodologies for identification of aquaculture’s contribution to the SDGs, and (3) trade-offs between farming systems’ contribution to meeting the SDGs. It becomes clear that decision-making around resource allocation and trade-offs between aquaculture and other aquatic resource users needs review of a wide range of established and emergent systems. The review ends by highlighting knowledge gaps and pathways for transformation that will allow further strengthening of aquaculture’s role for contributing to the SDGs. This includes identification and building on already existing monitoring that can enable capturing SDG-relevant aquaculture statistics at a national level and discussion of how a cohesive and comprehensive aquaculture strategy, framed to meet the SDGs, may help countries to prioritize actions for improving well-being.;,citation_author=Max Troell;,citation_author=Barry Costa-Pierce;,citation_author=Selina Stead;,citation_author=Richard S. Cottrell;,citation_author=Cecile Brugere;,citation_author=Anna K. Farmery;,citation_author=David C. Little;,citation_author=Åsa Strand;,citation_author=Roger Pullin;,citation_author=Doris Soto;,citation_author=Malcolm Beveridge;,citation_author=Khalid Salie;,citation_author=Jorge Dresdner;,citation_author=Patricia Moraes-Valenti;,citation_author=Julia Blanchard;,citation_author=Philip James;,citation_author=Rodrigue Yossa;,citation_author=Edward Allison;,citation_author=Christopher Devaney;,citation_author=Uwe Barg;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1111/jwas.12946;,citation_issue=2;,citation_doi=https://doi.org/ 10.1111/jwas.12946;,citation_issn=1749-7345;,citation_volume=54;,citation_journal_title=Journal of the World Aquaculture Society;">
<meta name="citation_reference" content="citation_title=San Pedro Shelf California: Testing the Pearson-Rosenberg model (PRM);,citation_abstract=Based on quarterly sampling (n = 260) over four years (1985–1989) from California’s San Pedro Shelf, the Pearson-Rosenberg Model (PRM) or organic enrichment was tested for the Orange County ocean outfall. The null hypothesis was that test species, abundance, and biomass curves (SAB) from the shelf closely resemble those from the model. Principal areas of agreement between test curves and model curves include increased abundance and biomass approaching an ocean outfall. Major departures from the PRM include: (1) no sharp decline in SAB curves to azoic conditions, (2) displacement of SAB curves away from the outfall, and (3) opportunistic species did not exclude or eliminate rare species. Moreover, the role of local dominant species (bivalve—Parvilucina tenuisculpta, ostracod—Euphilomedes carcharodonta polychaetous annelid—Capitella capitata) can greatly influence SAB curves within the model. Bioenhancement should not necessarily be viewed as a diagnostic feature of a polluted site. Since the PRM was originally developed for semi-enclosed, low energy depositional habitats with long residence times (fjords, sea lochs), open ocean, high energy, erosional habitats (coast and shelf) may not be the most appropriate sites to apply this model. Uncritical application of the PRM to the Orange County ocean outfall may lead to unnecessary and costly decisions.;,citation_author=Don Maurer;,citation_author=George Robertson;,citation_author=Thomas Gerlinger;,citation_publication_date=1993-01;,citation_cover_date=1993-01;,citation_year=1993;,citation_issue=4;,citation_doi=https://doi.org/ 10.1016/0141-1136(93)90099-L;,citation_issn=0141-1136;,citation_volume=35;,citation_journal_title=Marine Environmental Research;">
<meta name="citation_reference" content="citation_title=Macrobenthos community succession in the northern Gulf of Mexico hypoxic regions: Testing the Pearson-Rosenberg model;,citation_abstract=The Pearson and Rosenberg (P-R) conceptual model of macrobenthic succession was used to assess the impact of hypoxia (dissolved oxygen concentration ≤2 mg L–1) on the macrobenthic community on the continental shelf of the northern Gulf of Mexico. The stress-response relationship between bottom-water oxygen concentration and the macrobenthic community was used to determine the ecological condition of the benthic habitat. In lieu of testing the effect of decreasing influence of organic enrichment on the macrobenthic community as a function of distance from the source of organic pollution, we tested the effect of bottom-water dissolved oxygen concentration as a proxy for distance from the source. To include the seasonal response in the long-term cumulative effect of recurring hypoxia on the macrobenthic community, samples were collected during early spring (April 2009), late summer (September 2009), and midsummer (August 2010). The cumulative effect of annual hypoxia on the macrobenthos at each site was apparent from the species diversity, abundance, and biomass. High diversity values at site H7 and low diversity values at site A6 indicated the existence of a westward-diminishing hypoxic gradient on the Louisiana/Texas shelf, interrupted by the Atchafalaya River discharge. Unsurprisingly, the macrobenthos collected during midsummer were generally the most stressed community, with the exception of site E4. Assessment of successional stages of the benthic communities of the four sites with multivariate analyses of nonmetric multidimensional scaling, distance-based redundancy analysis, and principal component analysis identified the influence of bottom-water hypoxia. The study demonstrated the applicability of the P-R model in predicting successional change in the macrobenthic community on the Louisiana continental shelf. Site H7 was in the advanced phase of stage II, sites E4 and D5 were in the intermediate phase of stage II, and site A6 was in the intermediate phase of stage I in the P-R successional continuum.;,citation_author=Shivakumar Shivarudrappa;,citation_author=Kevin Briggs;,citation_publication_date=2017-01;,citation_cover_date=2017-01;,citation_year=2017;,citation_fulltext_html_url=https://elischolar.library.yale.edu/journal_of_marine_research/425;,citation_issue=1;,citation_doi=https://doi.org/ 10.1357/002224017821219036;,citation_volume=75;,citation_journal_title=Journal of Marine Research;">
<meta name="citation_reference" content="citation_title=Animal-sediment relationships: Evaluating the “Pearson–Rosenberg paradigm” in Mediterranean coastal lagoons;,citation_abstract=We investigated the applicability of the Pearson–Rosenberg (P–R) conceptual model describing a generalized pattern of response of benthic communities in relation to organic enrichment to Mediterranean Sea coastal lagoons. Consistent with P–R model predictions, benthic diversity and abundance showed two different peaks at low (&amp;amp;amp;gt;2.5–5mgg−1) and high (&amp;gt;25–30mgg−1) total organic carbon (TOC) ranges, respectively. We identified TOC thresholds indicating that risks of reduced benthic diversity should be relatively low at TOC values&amp;lt;about 10mgg−1, high at TOC values>about 28mgg−1, and intermediate at values in-between. Predictive ability within these ranges was high based on results of re-sampling simulation. While not a direct measure of causality, it is anticipated that these TOC thresholds should serve as a general screening-level indicator for evaluating the likelihood of reduced sediment quality and associated bioeffects in such eutrophic systems of the Mediterranean Sea.;,citation_author=Paolo Magni;,citation_author=Davide Tagliapietra;,citation_author=Claudio Lardicci;,citation_author=Len Balthis;,citation_author=Alberto Castelli;,citation_author=Serena Como;,citation_author=Gretel Frangipane;,citation_author=Gianmarco Giordani;,citation_author=Jeffrey L. Hyland;,citation_author=Ferruccio Maltagliati;,citation_author=Giuseppe Pessa;,citation_author=Andrea F. Rismondo;,citation_author=Mariella Tataranni;,citation_author=Paolo Tomassetti;,citation_author=Pierluigi Viaroli;,citation_publication_date=2009-04;,citation_cover_date=2009-04;,citation_year=2009;,citation_issue=4;,citation_doi=https://doi.org/10.1016/j.marpolbul.2008.12.009;,citation_issn=0025-326X;,citation_volume=58;,citation_journal_title=Marine Pollution Bulletin;">
<meta name="citation_reference" content="citation_title=Efficiency of feed utilization in Atlantic salmon (&amp;amp;amp;lt;i&amp;gt;Salmo salar&amp;lt;/i&amp;gt; L.) fed diets with increasing substitution of fish meal with vegetable proteins;,citation_abstract=Atlantic salmon smolts were fed for 15-week diets where 28% and 55% of the fish meal, respectively, were replaced by a mixture of full-fat soybean meal and maize gluten meal in comparison with a diet where 89% of the protein came from fish meal and the remainder from full-fat soybean meal and maize gluten meal. The diets were equal in gross energy, crude protein, lipids, carbohydrates, lysine and methionine plus cystine. Apparent digestibilities of energy and protein in fish meal, full-fat soybean meal and maize gluten meal were determined as 93.5% (19.5 MJ digestible energy (DE) kg−1) and 89.3%, 69.3% (14.4 MJ DE kg−1) and 76.9%, and 78.6% (16.4 MJ DE kg−1) and 87.0%, respectively. Tested by ANOVA and orthogonal polynomials, a linear reduction that approached significance and with no significant deviation from linearity was found in growth, SGR and TGC with increasing substitution of fish meal. Substitution of fish meal caused a significant linear reduction in the condition factor, while feed consumption was not affected. The effect of substitution on feed conversion ratio (FCR=g feed g−1 growth) showed a near significant deviation from linearity, FCR being significantly higher on Diet 3 than Diet 1 and slightly lower on Diet 2 than Diet 1. Apparent protein digestibility of the test diets decreased significantly linearly with increasing substitution, while no significant effects were found on digestibility of lipids and gross energy. Carbohydrate digestibility was low with no significant differences between diets. Fish meal substitution had no significant effect on carcass content of moisture, protein, ash or energy, while content of fat was nearly significantly linearly reduced. Substitution caused a nominal insignificant linear reduction in protein accretion but had no effect on digestible protein consumption per gram of protein accretion. Protein efficiency ratio (PER=weight gain/protein consumption) decreased significantly curvilinearly and net protein value (NPV=carcass protein gain/protein consumption) significantly linearly with increasing substitution. Energy utilization for growth decreased significantly linearly with increasing substitution. There was a nominal linear reduction, which approached significance in energy accretion with increasing substitution. Substitution impaired energy utilization for energy accretion (i.e. increased MJ DE consumed/MJ accredited).;,citation_author=Johannes Opstvedt;,citation_author=Anders Aksnes;,citation_author=Britt Hope;,citation_author=Ian H. Pike;,citation_publication_date=2003-05;,citation_cover_date=2003-05;,citation_year=2003;,citation_issue=1-4;,citation_doi=https://doi.org/ 10.1016/S0044-8486(03)00026-7;,citation_issn=00448486;,citation_volume=221;,citation_journal_title=Aquaculture;">
<meta name="citation_reference" content="citation_title=Global change in marine aquaculture production potential under climate change;,citation_abstract=Climate change is an immediate and future threat to food security globally. The consequences for fisheries and agriculture production potential are well studied, yet the possible outcomes for aquaculture (that is, aquatic farming)—one of the fastest growing food sectors on the planet—remain a major gap in scientific understanding. With over one-third of aquaculture produced in marine waters and this proportion increasing, it is critical to anticipate new opportunities and challenges in marine production under climate change. Here, we model and map the effect of warming ocean conditions (Representative Concentration Pathway scenario 8.5) on marine aquaculture production potential over the next century, based on thermal tolerance and growth data of 180 cultured finfish and bivalve species. We find heterogeneous patterns of gains and losses, but an overall greater probability of declines worldwide. Accounting for multiple drivers of species growth, including shifts in temperature, chlorophyll and ocean acidification, reveals potentially greater declines in bivalve aquaculture compared with finfish production. This study addresses a missing component in food security research and sustainable development planning by identifying regions that will face potentially greater climate change challenges and resilience with regards to marine aquaculture in the coming decades. Understanding the scale and magnitude of future increases and reductions in aquaculture potential is critical for designing effective and efficient use and protection of the oceans, and ultimately for feeding the planet sustainably.;,citation_author=Halley E. Froehlich;,citation_author=Rebecca R. Gentry;,citation_author=Benjamin S. Halpern;,citation_publication_date=2018-11;,citation_cover_date=2018-11;,citation_year=2018;,citation_issue=11;,citation_doi=https://doi.org/ 10.1038/s41559-018-0669-1;,citation_issn=2397-334X;,citation_volume=2;,citation_journal_title=Nature Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Mapping the global potential for marine aquaculture;,citation_abstract=Marine aquaculture presents an opportunity for increasing seafood production in the face of growing demand for marine protein and limited scope for expanding wild fishery harvests. However, the global capacity for increased aquaculture production from the ocean and the relative productivity potential across countries are unknown. Here, we map the biological production potential for marine aquaculture across the globe using an innovative approach that draws from physiology, allometry and growth theory. Even after applying substantial constraints based on existing ocean uses and limitations, we find vast areas in nearly every coastal country that are suitable for aquaculture. The development potential far exceeds the space required to meet foreseeable seafood demand; indeed, the current total landings of all wild-capture fisheries could be produced using less than 0.015% of the global ocean area. This analysis demonstrates that suitable space is unlikely to limit marine aquaculture development and highlights the role that other factors, such as economics and governance, play in shaping growth trajectories. We suggest that the vast amount of space suitable for marine aquaculture presents an opportunity for countries to develop aquaculture in a way that aligns with their economic, environmental and social objectives.;,citation_author=Rebecca R. Gentry;,citation_author=Halley E. Froehlich;,citation_author=Dietmar Grimm;,citation_author=Peter Kareiva;,citation_author=Michael Parke;,citation_author=Michael Rust;,citation_author=Steven D. Gaines;,citation_author=Benjamin S. Halpern;,citation_publication_date=2017-09;,citation_cover_date=2017-09;,citation_year=2017;,citation_issue=9;,citation_doi=https://doi.org/ 10.1038/s41559-017-0257-9;,citation_issn=2397-334X;,citation_volume=1;,citation_journal_title=Nature Ecology &amp;amp;amp; Evolution;">
<meta name="citation_reference" content="citation_title=Estimating stocking weights for Atlantic salmon to grow to market size at novel aquaculture sites with extreme temperatures;,citation_abstract=Land-based hatcheries are now capable of growing large Atlantic salmon (Salmo salar) post-smolts (approximately 150 – 1000 g), which means that marine net-pens can be stocked with substantially larger fish compared to traditional stocking sizes (&amp;amp;amp;lt; 150 g). This stocking strategy typically aims to reduce the time required for fish to grow to market size in the marine environment and limit risks (e.g., exposure to pathogens and diseases, opportunities for escapes). This study investigates another potential application of this strategy: the use of novel sites in areas previously considered unsuitable for aquaculture due to seasonally cold temperatures. The thermal-unit growth coefficient (TGC) model was applied to estimate the stocking weight needed to reach a harvest size of 5.5 kg, based on observed degree days for three sites. High resolution, depth-partitioned temperature time series from coastal locations in Atlantic Canada were used to represent a short, medium, and long growing season, as constrained by seasonal temperature extremes. Growing days for model inputs were defined as temperatures &amp;gt; 4 °C and trending up for stocking, &amp;lt; 18 °C to account for heat stress, and &amp;gt; -0.7 °C to avoid superchill conditions. Different TGC values were applied to simulate remedial, average, and elite growth performance. There was a range of model stocking weight estimates for each site (1.5 – 2.5 kg, 0.94 – 2.8 kg, and < 0.1 – 0.52 kg, for the short, medium, and long season sites, respectively). Results were sensitive to the number of degree days, heat stress threshold, and TGC value. At the two sites where season length was constrained by superchill, fish with a stocking weight of approximately 1.5 kg could grow to market size in shallow water depths (< 15 m), assuming elite growth performance. This investigation suggests that with appropriate growth performance assumptions and high-resolution temperature data, large post-smolt stocking strategies could enable the use of novel sites in coastal areas previously considered unsuitable for aquaculture.;,citation_author=Danielle P. Dempsey;,citation_author=Gregor K. Reid;,citation_author=Leah Lewis-McCrea;,citation_author=Toby Balch;,citation_author=Roland Cusack;,citation_author=André Dumas;,citation_author=Jack Rensel;,citation_publication_date=2023-05;,citation_cover_date=2023-05;,citation_year=2023;,citation_doi=https://doi.org/ 10.3389/fmars.2023.1094247;,citation_issn=2296-7745;,citation_volume=10;,citation_journal_title=Frontiers in Marine Science;">
<meta name="citation_reference" content="citation_title=Towards better survival: Modeling drivers for daily mortality in Norwegian Atlantic salmon farming;,citation_abstract=Mortality in the production of farmed salmonids is a major constraint to the sustainability of this form of animal husbandry. We have developed a model for the daily mortality in salmon farming over a full production cycle from stocking to harvest, considering different environmental and production factors. These factors included sea temperature, salinity, day within year, fish weight at stocking, stocking day, four types of lice treatments and the possible occurrence of pancreas disease (PD). We considered a generalized additive model following full production cycles, allowing for non-linear descriptions of how relevant factors relate to the daily mortality. We saw a high overall mortality rate immediately after stocking, which decreased the first three months in the cycle and thereafter increased. We found that the total mortality could be reduced by 21% if avoiding all lice treatments, and similarly reduced by 20% if no PD infections occurred. If avoiding jointly PD and all lice treatments, the accumulated mortality could be reduced by 34%. A single thermal or hydrogen peroxide treatment was associated with a mortality of around 1.6% and 1.3%, respectively. This modeling approach gave a unique opportunity to model how different factors interact on the overall global mortality and can easily be extended by other factors, such as additional fish diseases.;,citation_author=Ingunn Fride Tvete;,citation_author=Magne Aldrin;,citation_author=Britt Bang Jensen;,citation_publication_date=2023-01;,citation_cover_date=2023-01;,citation_year=2023;,citation_doi=https://doi.org/ 10.1016/j.prevetmed.2022.105798;,citation_issn=0167-5877;,citation_volume=210;,citation_journal_title=Preventive Veterinary Medicine;">
<meta name="citation_reference" content="citation_title=Modelling Compensatory Growth;,citation_abstract=1. We formulate a simple, physiologically based model to investigate the phenomenon of compensatory growth. 2. The model is based upon two key assumptions. First, that an individual partitions net assimilate between two tissue types: those which can and those which cannot be remobilized once laid down. Second, that the individual modulates its behaviour and physiology in response to the instantaneous ratio of mobilizable to non-mobilizable tissues. 3. The model is parameterized for salmonids from published studies of their energetics. Data from aquaculture studies of fish growth under conditions of fluctuating food are used to test the model. Using a common parameter set, the model successfully reproduces the growth patterns observed in 16 different feeding regimes.;,citation_author=Niall. Broekhuizen;,citation_author=William. S. C. Gurney;,citation_author=A. Jones;,citation_author=A. D. Bryant;,citation_publication_date=1994;,citation_cover_date=1994;,citation_year=1994;,citation_issue=6;,citation_doi=https://doi.org/10.2307/2390237;,citation_issn=0269-8463;,citation_volume=8;,citation_journal_title=Functional Ecology;">
<meta name="citation_reference" content="citation_title=Energy equivalents of oxygen consumption in animal energetics;,citation_abstract=The energy equivalents given in the literature are reviewed and criticised. New equivalents are calculated for protein respiration.The energy equivalent for converting rate of oxygen consumption into rate of heat production (Q ox cal mg-1 oxygen consumed) is 3.53 cal mg-1 for carbohydrate oxidation, 3.28 cal mg-1 (range 3.22–3.32) for fat oxidation. Q ox values for the respiration of standard protein are the same at 3.25 cal mg-1 for ureotelic and uricotelic animals, and about 2% less at 3.20 cal mg-1 for ammoniotelic animals. The energy equivalent for converting rate of oxygen consumption into rate of energy loss in excreta (Q ex cal mg-1) varies considerably with different excretory products. Values for standard protein are 0.62 cal mg-1 for ammonioteles, 0.58 cal mg-1 for ureoteles, and 0.94 cal mg-1 for uricoteles.The various factors affecting both Q ox and Q ex are discussed, and examples of the estimation of general energy equivalents are given.;,citation_author=J. M. Elliott;,citation_author=W. Davison;,citation_publication_date=1975-09;,citation_cover_date=1975-09;,citation_year=1975;,citation_issue=3;,citation_doi=https://doi.org/10.1007/BF00345305;,citation_issn=1432-1939;,citation_volume=19;,citation_journal_title=Oecologia;">
<meta name="citation_reference" content="citation_title=Using meta-analysis to derive a respiration model for Atlantic salmon (&amp;amp;amp;lt;i&amp;gt;Salmo salar&amp;lt;/i&amp;gt;) to assess bioenergetics requirements of juveniles in two Canadian rivers;,citation_abstract=Standard metabolic rates (SMRs) for Atlantic salmon (Salmo salar) have been calculated independently for different life stages and populations, but the absence of a comprehensive SMR model limits its application for modelling the energy use or life stage-specific growth. Atlantic salmon respiration data were compiled from a meta-analysis of 26 publications, and exponential or optimal relationships were fitted to the metadata to estimate respiration equation parameters and generate confidence intervals dependent on temperature and body mass. While model parameters were significant for both models, mass-corrected standard metabolic rates (g O2·day−1) increased as a function of water temperature (°C) and decreased beyond ∼16 °C following an optimal relationship (AICoptimal = –9185.50 versus AICexponential = –8948.95; ΔAIC = 236.55). Juvenile Atlantic salmon growth (cohorts 1 and 2) from bioenergetics simulations did not vary between Little Southwest Miramichi and Northwest Miramichi rivers; however, variation between simulations using the different respiration models (i.e., exponential versus optimal) led to differences in the way fish allocate energy throughout the year. Results from this analysis will inform conservation efforts for the species throughout its current range and predict the energetic requirements at juvenile life stages.;,citation_author=Camille J. Macnaughton;,citation_author=David Deslauriers;,citation_author=Erinn L. Ipsen;,citation_author=Emily Corey;,citation_author=Eva C. Enders;,citation_publication_date=2019-12;,citation_cover_date=2019-12;,citation_year=2019;,citation_issue=12;,citation_doi=https://doi.org/ 10.1139/cjfas-2018-0436;,citation_issn=0706-652X;,citation_volume=76;,citation_journal_title=Canadian Journal of Fisheries and Aquatic Sciences;">
<meta name="citation_reference" content="citation_title=Thermal performance of juvenile Atlantic Salmon, &amp;amp;amp;lt;i&amp;gt;Salmo salar&amp;lt;/i&amp;gt; L.;,citation_abstract=1 Experimental data for maximum growth and food consumption of Atlantic Salmon (Salmo salar L.) parr from five Norwegian rivers situated between 59 and 70°N were analysed and modelled. The growth and feeding models were also applied to groups of Atlantic Salmon growing and feeding at rates below the maximum. The data were fitted to the Ratkowsky model, originally developed for bacterial growth. 2 The rates of growth and food consumption varied significantly among populations but the variation appeared unrelated to thermal conditions in the river of population origins. No correlation was found between the thermal conditions and limits for growth, thermal growth optima or maximum growth, and hypotheses of population-specific thermal adaptation were not supported. Estimated optimum temperatures for growth were between 16 and 20 °C. 3 Model parameter estimates differed among growth-groups in that maximum growth and the performance breadth decreased from fast to slow growing individuals. The optimum temperature for growth did not change with growth rate. 4 The model for food consumption (expressed in energy terms) peaked at 19–21 °C, which is only slightly higher than the optimal temperature for growth. Growth appeared directly related to food consumption. Consumption was initiated ≈2 °C below the lower temperature for growth and terminated ≈1·5 °C above the upper critical temperature for growth. Model parameter estimates for consumption differed among growth-groups in a manner similar to the growth models. 5 By combining the growth and consumption models, growth efficiencies were estimated. The maximum efficiencies were high, 42–58%, and higher in rivers offering hostile than benign feeding and growth opportunities.;,citation_author=Bror Jonsson;,citation_author=Torbjørn Forseth;,citation_author=Arne. J. Jensen;,citation_author=Tor. F. Næsje;,citation_publication_date=2001;,citation_cover_date=2001;,citation_year=2001;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1046/j.0269-8463.2001.00572.x;,citation_issue=6;,citation_doi=https://doi.org/10.1046/j.0269-8463.2001.00572.x;,citation_issn=1365-2435;,citation_volume=15;,citation_journal_title=Functional Ecology;">
<meta name="citation_reference" content="citation_title=Unusual aerobic performance at high temperatures in juvenile Chinook salmon, &amp;amp;amp;lt;i&amp;gt;Oncorhynchus tshawytscha&amp;lt;/i&amp;gt;;,citation_abstract=Understanding how the current warming trends affect fish populations is crucial for effective conservation and management. To help define suitable thermal habitat for juvenile Chinook salmon, the thermal performance of juvenile Chinook salmon acclimated to either 15 or 19°C was tested across a range of environmentally relevant acute temperature changes (from 12 to 26°C). Swim tunnel respirometers were used to measure routine oxygen uptake as a measure of routine metabolic rate (RMR) and oxygen uptake when swimming maximally as a measure of maximal metabolic rate (MMR) at each test temperature. We estimated absolute aerobic scope (AAS&nbsp;=&nbsp;MMR&nbsp;−&nbsp;RMR), the capacity to supply oxygen beyond routine needs, as well as factorial aerobic scope (FAS&nbsp;=&nbsp;MMR/RMR). All fish swam at a test temperature of 23°C regardless of acclimation temperature, but some mortality occurred at 25°C during MMR measurements. Overall, RMR and MMR increased with acute warming, but aerobic capacity was unaffected by test temperatures up to 23°C in both acclimation groups. The mean AAS for fish acclimated and tested at 15°C (7.06&nbsp;±&nbsp;1.76 mg O2 kg−1 h−1) was similar to that measured for fish acclimated and tested at 19°C (8.80&nbsp;±&nbsp;1.42 mg O2 kg−1 h−1). Over the entire acute test temperature range, while MMR and AAS were similar for the two acclimation groups, RMR was significantly lower and FAS consequently higher at the lower test temperatures for the fish acclimated at 19°C. Thus, this stock of juvenile Chinook salmon shows an impressive aerobic capacity when acutely warmed to temperatures close to their upper thermal tolerance limit, regardless of the acclimation temperature. These results are compared with those for other salmonids, and the implications of our findings for informing management actions are discussed.;,citation_author=Jamilynn B. Poletto;,citation_author=Dennis E. Cocherell;,citation_author=Sarah E. Baird;,citation_author=Trinh X. Nguyen;,citation_author=Valentina Cabrera-Stagno;,citation_author=Anthony P. Farrell;,citation_author=Nann A. Fangue;,citation_publication_date=2017-01;,citation_cover_date=2017-01;,citation_year=2017;,citation_issue=1;,citation_doi=https://doi.org/ 10.1093/conphys/cow067;,citation_issn=2051-1434;,citation_volume=5;,citation_journal_title=Conservation Physiology;">
<meta name="citation_reference" content="citation_title=Lowering temperature to 1&nbsp;°C results in physiological changes, stress and mortality in cultured Atlantic Salmon (&amp;amp;amp;lt;i&amp;gt;Salmo salar&amp;lt;/i&amp;gt;);,citation_abstract=Atlantic salmon (Salmo salar) reared in sea-cages in Atlantic Canada and Iceland can be exposed to temperatures as low as 0–1&nbsp;°C during the winter months, and this can result in large-scale mortalities. Despite this threat to salmon aquaculture, most of our knowledge of this species’ thermal biology is at temperatures above 10&nbsp;°C. In this laboratory experiment, Atlantic salmon post-smolts (175&nbsp;±&nbsp;24&nbsp;g) were exposed to a temperature decline of 1&nbsp;°C/week (from 8 to 1&nbsp;°C) with an additional week at 1&nbsp;°C (to simulate temperature changes in sea-cages in Atlantic Canada during the winter months), or a constant temperature of 8&nbsp;°C (control), and sampled weekly. Growth was reduced by 58% in the cold-exposed group and was associated with a decrease in appetite beginning at 6&nbsp;°C. Starting at 4–5&nbsp;°C, an osmoregulatory disturbance ensued characterized by higher plasma [Cl−], lower plasma [K+] and increased osmolality. Hsp70 and hsp90 transcript expression also increased as temperature was reduced from 5 to 1&nbsp;°C. However, plasma cortisol levels did not increase (to ∼18&nbsp;ng/ml) until the fish reached 1&nbsp;°C, and serpinh1 and cirbp expression were not good biomarkers of decreasing temperature. Symptoms suggestive of liver dysfunction (changes in plasma [aspartate aminotransferase] and an increase in liver size) were also evident at 1&nbsp;°C. Finally, at 1&nbsp;°C, 5% of the fish died, and these mortalities were typically associated with snout ulceration/erosion and fin rot. In summary, Atlantic salmon held at temperatures &amp;amp;lt;5&nbsp;°C suffer from decreased growth performance, osmoregulatory disturbance, liver dysfunction, stress, opportunistic infections, and increased mortality.;,citation_author=Émile Vadboncoeur;,citation_author=Charlotte Nelson;,citation_author=Jennifer R. Hall;,citation_author=Kathy A. Clow;,citation_author=Rebeccah M. Sandrelli;,citation_author=Colin J. Brauner;,citation_author=Andrew K. Swanson;,citation_author=A. Kurt Gamperl;,citation_publication_date=2023-04;,citation_cover_date=2023-04;,citation_year=2023;,citation_doi=https://doi.org/ 10.1016/j.aquaculture.2023.739313;,citation_issn=0044-8486;,citation_volume=568;,citation_journal_title=Aquaculture;">
<meta name="citation_reference" content="citation_title=Development of a nutritional model to define the energy and protein requirements of cobia, &amp;amp;amp;lt;i&amp;gt;Rachycentron canadum&amp;lt;/i&amp;gt;;,citation_abstract=This study assessed the protein and energy requirements of Cobia (Rachycentron canadum) using a bio-energetic factorial approach. Using a series of inter-related studies, several parameters were defined to enable the construction of a bio-energetic factorial model for this species. The studies included two controlled laboratory experiments and also extensive field-data collection from commercial and research farms in Vietnam. The devised model includes parameters for both maintenance and protein demands; the effect of fish live-weight on maintenance protein (LW0.697), lipid (LW0.972), and energy demands (LW0.815); the efficiencies of protein, lipid and energy utilisation at various protein, lipid and energy intake levels; and the variability in whole body composition with varying live-weight. The protein utilisation efficiencies (0.456·[protein intake]−0.445), lipid utilisation efficiencies (1.292·[lipid intake]−1.120) and energy utilisation efficiencies (0.651·[energy intake]−48.41) were similar to other carnivorous fish species. However, the maintenance requirements for both energy (74.3kJ/kgBW0.8/d− at 28°C) and protein (0.99g/kgBW0.7/d at 27.9°C) were about double to other species. Using this modelling approach it was possible to iteratively derive optimal dietary protein and energy specifications for this species.;,citation_author=Nguyen Van Tien;,citation_author=Dam T. M. Chinh;,citation_author=Tran T. M. Huong;,citation_author=Truong H. Phuong;,citation_author=Simon Irvin;,citation_author=Brett Glencross;,citation_publication_date=2016-10;,citation_cover_date=2016-10;,citation_year=2016;,citation_doi=https://doi.org/ 10.1016/j.aquaculture.2016.05.037;,citation_issn=0044-8486;,citation_volume=463;,citation_journal_title=Aquaculture;">
<meta name="citation_reference" content="citation_title=The effect of temperature and fish size on growth, feed intake, food conversion efficiency and stomach evacuation rate of Atlantic salmon post-smolts;,citation_abstract=The present paper describes the growth properties of Atlantic salmon (Salmo salar) post-smolts reared at 6, 10, 14 and 18&nbsp;°C for 12&nbsp;weeks following transfer to seawater. Growth rate, feed intake, feed conversion efficiency (FCE), and stomach evacuation rate were significantly influenced by temperature and fish size. Highest growth rate was seen in the 14&nbsp;°C group (1.53% d−1), no differences in growth were seen between the 10 and 18&nbsp;°C groups (1.35% d−1, 1.29% d−1), and lowest growth observed at 6&nbsp;°C (0.78% d−1) group. Optimal temperature for growth (ToptG) increased with fish size, whereas optimal temperature for feed conversion efficiency (ToptFCE) decreased as fish size increased. The results suggest an optimum temperature for growth of 12.8&nbsp;°C for 70–150&nbsp;g to 14.0&nbsp;°C for 150–300&nbsp;g post-smolts. Optimum temperature for FCE dropped from 13.4&nbsp;°C to 11.0&nbsp;°C for the same size classes. A wide parabolic regression curve between growth, feed conversion efficiency and temperature indicates high temperature tolerance of Atlantic salmon in this size range studied. Highest stomach evacuation rate was seen in the 18&nbsp;°C group, where the proportion of meal remaining in the stomach decreased from 100 to less than 5% after 24&nbsp;h of starvation. No differences in stomach evacuation rate were recorded between the 10 and 14&nbsp;°C groups, whereas a significant delay in stomach evacuation was seen in the 6&nbsp;°C group. Overall, these findings may have important consequences for optimization of commercial production of Atlantic salmon post-smolts.;,citation_author=Sigurd O. Handeland;,citation_author=Albert K. Imsland;,citation_author=Sigurd O. Stefansson;,citation_publication_date=2008-10;,citation_cover_date=2008-10;,citation_year=2008;,citation_issue=1;,citation_doi=https://doi.org/ 10.1016/j.aquaculture.2008.06.042;,citation_issn=0044-8486;,citation_volume=283;,citation_journal_title=Aquaculture;">
<meta name="citation_reference" content="citation_title=Energy allocation among developmental stages, age groups, and types of Atlantic salmon (&amp;amp;amp;lt;i&amp;gt;Salmo salar&amp;lt;/i&amp;gt;) spawners;,citation_abstract=Relationships between energy density (kJ·100 g–1) and body length (mm), which varied between life-history stages of Atlantic salmon (Salmo salar), were described by power functions. The variation was chiefly caused by differences in lipid concentrations. Energy densities of maturing and rematuring parr were higher than those of similarly aged smolts. Monthly specific growth rate was higher in maturing parr than in salmon at sea. Mature parr males allocated relatively more energy to gonadal development and less to soma development than anadromous males. Spawning expenditure increased from 34 to 53% for 55- to 72-cm-long females. In males, similar losses were from 47 to 49% of their total energy content at river entry. More energy was used in reproduction in a long river than in a short one. Male allocation to gonads relative to soma was highest in the shorter river. Energy allocation pattern varied among developmental stages. Very energy-rich male parr rematured in subsequent years without going to sea; less energy-rich ones smolted and migrated to sea.;,citation_author=Nina Jonsson;,citation_author=Bror Jonsson;,citation_publication_date=2003-05;,citation_cover_date=2003-05;,citation_year=2003;,citation_issue=5;,citation_doi=https://doi.org/ 10.1139/f03-042;,citation_issn=0706-652X;,citation_volume=60;,citation_journal_title=Canadian Journal of Fisheries and Aquatic Sciences;">
<meta name="citation_reference" content="citation_title=Does waste feed from salmon farming affect the quality of saithe (&amp;amp;amp;lt;i&amp;gt;Pollachius virens&amp;lt;/i&amp;gt; L.) attracted to fish farms?;,citation_abstract=Salmon farms attract large amounts of wild fish, which prey on uneaten feed pellets. The modified diet of the wild fish aggregating at salmon farms may reduce the flesh quality of the fish, influencing the local fisheries. We compared the quality of saithe (Pollachius virens) captured near (farm associated—FA) or more than 5 km away (un-associated—UA) from salmon farms in Norway. The fish were captured during summer, autumn and spring using two commercial fishing methods (jigging and bottom nets). Overall, the fillet quality of FA saithe was good, although it was clearly reduced for almost 10% of the catch. Moreover, the quality of the FA saithe was significantly reduced compared with UA saithe, but the differences were small. Our results also showed that fish caught with jigging had better quality than fish caught with nets, and that fish that died in the nets were of lower quality than fish that were alive after hauling. There was no clear variation among seasons in fillet quality. Although no major and overall differences in quality were found between FA and UA saithe, reduced quality for even a modest proportion of the fish may influence the value of the total catch.;,citation_author=Ingebrigt Uglem;,citation_author=Kilian Toledo-Guedes;,citation_author=Pablo Sanchez-Jerez;,citation_author=Eva M. Ulvan;,citation_author=Tor Evensen;,citation_author=Bjørn S. Sæther;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://onlinelibrary.wiley.com/doi/abs/10.1111/are.14519;,citation_issue=4;,citation_doi=https://doi.org/ 10.1111/are.14519;,citation_issn=1365-2109;,citation_volume=51;,citation_journal_title=Aquaculture Research;">
<meta name="citation_reference" content="citation_title=Feed Ingredients for Sustainable Aquaculture;,citation_abstract=Due to the precarious status of global fisheries the aquaculture sector has come under pressure to move away from its addiction to fishmeal (FM) and fish oil (FO) as feed ingredients, toward more sustainable alternatives. This chapter provides a brief overview of animal, microbial and plant-based feedstuffs that have been examined as FM/FO substitutes. Other than classic rendered meat products, attention is given to insect meals and issues surrounding their safety. Single celled products, including fungi and yeasts, bacteria, and microalgae are examined as sources of protein, lipid, pigments and enzymes. Plant-based proteins and lipid sources are also examined. Feed additives such as exogenous enzymes (phytases, lipases, proteases and carbohydrases) are evaluated as potential aquafeed ingredients as too are pigments, chemoattractants and palatants. Discussion is provided on pre-, pro- and synbiotics. Examples of the application of these various ingredients are considered with reference to over 50 species of cultivated organism.;,citation_author=Ewen McLean;,citation_editor=Pasquale Ferranti;,citation_publication_date=2023-01;,citation_cover_date=2023-01;,citation_year=2023;,citation_fulltext_html_url=https://doi.org/10.1016/b978-0-12-823960-5.00085-8;,citation_isbn=978-0-12-824166-0;,citation_inbook_title=Sustainable Food Science - A Comprehensive Approach;">
<meta name="citation_reference" content="citation_title=R: A language and environment for statistical computing;,citation_author=R Core Team;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_fulltext_html_url=https://www.R-project.org;">
<meta name="citation_reference" content="citation_title=Quarto;,citation_author=J. J. Allaire;,citation_author=Charles Teague;,citation_author=Carlos Scheidegger;,citation_author=Yihui Xie;,citation_author=Christophe Dervieux;,citation_publication_date=2024-02;,citation_cover_date=2024-02;,citation_year=2024;,citation_fulltext_html_url=https://github.com/quarto-dev/quarto-cli;,citation_doi=10.5281/zenodo.5960048;">
<meta name="citation_reference" content="citation_title=acronyms;,citation_author=Remy Chaput;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://github.com/rchaput/acronyms;">
<meta name="citation_reference" content="citation_title=Tinytex: Helper functions to install and maintain TeX live, and compile LaTeX documents;,citation_author=Yihui Xie;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_fulltext_html_url=https://cran.r-project.org/web/packages/tinytex;">
<meta name="citation_reference" content="citation_title=Ecoflux: Functions for ecological flux studies including sap flux and soil/stem c efflux.;,citation_author=Joe Shannon;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;">
<meta name="citation_reference" content="citation_title=The targets r package: A dynamic make-like function-oriented pipeline toolkit for reproducibility and high-performance computing;,citation_author=William Michael Landau;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://doi.org/10.21105/joss.02959;,citation_issue=57;,citation_volume=6;,citation_journal_title=Journal of Open Source Software;">
<meta name="citation_reference" content="citation_title=Welcome to the tidyverse;,citation_author=Hadley Wickham;,citation_author=Mara Averick;,citation_author=Jennifer Bryan;,citation_author=Winston Chang;,citation_author=Lucy D’Agostino McGowan;,citation_author=Romain François;,citation_author=Garrett Grolemund;,citation_author=Alex Hayes;,citation_author=Lionel Henry;,citation_author=Jim Hester;,citation_author=Max Kuhn;,citation_author=Thomas Lin Pedersen;,citation_author=Evan Miller;,citation_author=Stephan Milton Bache;,citation_author=Kirill Müller;,citation_author=Jeroen Ooms;,citation_author=David Robinson;,citation_author=Dana Paige Seidel;,citation_author=Vitalie Spinu;,citation_author=Kohske Takahashi;,citation_author=Davis Vaughan;,citation_author=Claus Wilke;,citation_author=Kara Woo;,citation_author=Hiroaki Yutani;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=43;,citation_doi=10.21105/joss.01686;,citation_volume=4;,citation_journal_title=Journal of Open Source Software;">
<meta name="citation_reference" content="citation_title=Renv: Project environments;,citation_author=Kevin Ushey;,citation_author=Hadley Wickham;,citation_publication_date=2025;,citation_cover_date=2025;,citation_year=2025;,citation_fulltext_html_url=https://rstudio.github.io/renv/;">
<meta name="citation_reference" content="citation_title=Broom: Convert statistical objects into tidy tibbles;,citation_author=David Robinson;,citation_author=Alex Hayes;,citation_author=Simon Couch;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=broom;">
<meta name="citation_reference" content="citation_title=Cli: Helpers for developing command line interfaces;,citation_author=Gábor Csárdi;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=cli;">
<meta name="citation_reference" content="citation_title=Conflicted: An alternative conflict resolution strategy;,citation_author=Hadley Wickham;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=conflicted;">
<meta name="citation_reference" content="citation_title=Cowplot: Streamlined plot theme and plot annotations for ’ggplot2’;,citation_author=Claus O. Wilke;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=cowplot;">
<meta name="citation_reference" content="citation_title=Dbplyr: A ’dplyr’ back end for databases;,citation_author=Hadley Wickham;,citation_author=Maximilian Girlich;,citation_author=Edgar Ruiz;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=dbplyr;">
<meta name="citation_reference" content="citation_title=Dplyr: A grammar of data manipulation;,citation_author=Hadley Wickham;,citation_author=Romain François;,citation_author=Lionel Henry;,citation_author=Kirill Müller;,citation_author=Davis Vaughan;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=dplyr;">
<meta name="citation_reference" content="citation_title=Dtplyr: Data table back-end for ’dplyr’;,citation_author=Hadley Wickham;,citation_author=Maximilian Girlich;,citation_author=Mark Fairbanks;,citation_author=Ryan Dickerson;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=dtplyr;">
<meta name="citation_reference" content="citation_title=Forcats: Tools for working with categorical variables (factors);,citation_author=Hadley Wickham;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=forcats;">
<meta name="citation_reference" content="citation_title=Formatdown: Formatting numbers in ’rmarkdown’ documents;,citation_author=Richard Layton;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=formatdown;">
<meta name="citation_reference" content="citation_title=ggplot2: Elegant graphics for data analysis;,citation_author=Hadley Wickham;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_fulltext_html_url=https://ggplot2.tidyverse.org;,citation_isbn=978-3-319-24277-4;">
<meta name="citation_reference" content="citation_title=Googledrive: An interface to google drive;,citation_author=Lucy D’Agostino McGowan;,citation_author=Jennifer Bryan;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=googledrive;">
<meta name="citation_reference" content="citation_title=googlesheets4: Access google sheets using the sheets API V4;,citation_author=Jennifer Bryan;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=googlesheets4;">
<meta name="citation_reference" content="citation_title=Haven: Import and export ’SPSS’, ’stata’ and ’SAS’ files;,citation_author=Hadley Wickham;,citation_author=Evan Miller;,citation_author=Danny Smith;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=haven;">
<meta name="citation_reference" content="citation_title=Hms: Pretty time of day;,citation_author=Kirill Müller;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=hms;">
<meta name="citation_reference" content="citation_title=Httr: Tools for working with URLs and HTTP;,citation_author=Hadley Wickham;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=httr;">
<meta name="citation_reference" content="citation_title=The jsonlite package: A practical and consistent mapping between JSON data and r objects;,citation_author=Jeroen Ooms;,citation_publication_date=2014;,citation_cover_date=2014;,citation_year=2014;,citation_fulltext_html_url=https://arxiv.org/abs/1403.2805;,citation_journal_title=arXiv:1403.2805 [stat.CO];">
<meta name="citation_reference" content="citation_title=Dates and times made easy with lubridate;,citation_author=Garrett Grolemund;,citation_author=Hadley Wickham;,citation_publication_date=2011;,citation_cover_date=2011;,citation_year=2011;,citation_fulltext_html_url=https://www.jstatsoft.org/v40/i03/;,citation_issue=3;,citation_volume=40;,citation_journal_title=Journal of Statistical Software;">
<meta name="citation_reference" content="citation_title=Magrittr: A forward-pipe operator for r;,citation_author=Stefan Milton Bache;,citation_author=Hadley Wickham;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://CRAN.R-project.org/package=magrittr;">
<meta name="citation_reference" content="citation_title=Modelr: Modelling functions that work with the pipe;,citation_author=Hadley Wickham;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=modelr;">
<meta name="citation_reference" content="citation_title=Ozmaps: Australia maps;,citation_author=Michael Sumner;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://github.com/mdsumner/ozmaps;">
<meta name="citation_reference" content="citation_title=Pillar: Coloured formatting for columns;,citation_author=Kirill Müller;,citation_author=Hadley Wickham;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=pillar;">
<meta name="citation_reference" content="citation_title=Purrr: Functional programming tools;,citation_author=Hadley Wickham;,citation_author=Lionel Henry;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=purrr;">
<meta name="citation_reference" content="citation_title=Ragg: Graphic devices based on AGG;,citation_author=Thomas Lin Pedersen;,citation_author=Maxim Shemanarev;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=ragg;">
<meta name="citation_reference" content="citation_title=Readr: Read rectangular text data;,citation_author=Hadley Wickham;,citation_author=Jim Hester;,citation_author=Jennifer Bryan;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=readr;">
<meta name="citation_reference" content="citation_title=Readxl: Read excel files;,citation_author=Hadley Wickham;,citation_author=Jennifer Bryan;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=readxl;">
<meta name="citation_reference" content="citation_title=Reprex: Prepare reproducible example code via the clipboard;,citation_author=Jennifer Bryan;,citation_author=Jim Hester;,citation_author=David Robinson;,citation_author=Hadley Wickham;,citation_author=Christophe Dervieux;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=reprex;">
<meta name="citation_reference" content="citation_title=Rlang: Functions for base types and core r and ’tidyverse’ features;,citation_author=Lionel Henry;,citation_author=Hadley Wickham;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=rlang;">
<meta name="citation_reference" content="citation_title=Rstudioapi: Safely access the RStudio API;,citation_author=Kevin Ushey;,citation_author=JJ Allaire;,citation_author=Hadley Wickham;,citation_author=Gary Ritchie;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=rstudioapi;">
<meta name="citation_reference" content="citation_title=Rvest: Easily harvest (scrape) web pages;,citation_author=Hadley Wickham;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=rvest;">
<meta name="citation_reference" content="citation_title=Simple features for r: Standardized support for spatial vector data;,citation_author=Edzer Pebesma;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=https://doi.org/10.32614/RJ-2018-009;,citation_issue=1;,citation_volume=10;,citation_journal_title=The R Journal;">
<meta name="citation_reference" content="citation_title=Stringr: Simple, consistent wrappers for common string operations;,citation_author=Hadley Wickham;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=stringr;">
<meta name="citation_reference" content="citation_title=Terra: Spatial data analysis;,citation_author=Robert J. Hijmans;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=terra;">
<meta name="citation_reference" content="citation_title=Tibble: Simple data frames;,citation_author=Kirill Müller;,citation_author=Hadley Wickham;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=tibble;">
<meta name="citation_reference" content="citation_title=Tidyr: Tidy messy data;,citation_author=Hadley Wickham;,citation_author=Davis Vaughan;,citation_author=Maximilian Girlich;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=tidyr;">
<meta name="citation_reference" content="citation_title=Welcome to the tidyverse;,citation_author=Hadley Wickham;,citation_author=Mara Averick;,citation_author=Jennifer Bryan;,citation_author=Winston Chang;,citation_author=Lucy D’Agostino McGowan;,citation_author=Romain François;,citation_author=Garrett Grolemund;,citation_author=Alex Hayes;,citation_author=Lionel Henry;,citation_author=Jim Hester;,citation_author=Max Kuhn;,citation_author=Thomas Lin Pedersen;,citation_author=Evan Miller;,citation_author=Stephan Milton Bache;,citation_author=Kirill Müller;,citation_author=Jeroen Ooms;,citation_author=David Robinson;,citation_author=Dana Paige Seidel;,citation_author=Vitalie Spinu;,citation_author=Kohske Takahashi;,citation_author=Davis Vaughan;,citation_author=Claus Wilke;,citation_author=Kara Woo;,citation_author=Hiroaki Yutani;,citation_publication_date=2019;,citation_cover_date=2019;,citation_year=2019;,citation_issue=43;,citation_doi=10.21105/joss.01686;,citation_volume=4;,citation_journal_title=Journal of Open Source Software;">
<meta name="citation_reference" content="citation_title=Measurement units in R;,citation_author=Edzer Pebesma;,citation_author=Thomas Mailund;,citation_author=James Hiebert;,citation_publication_date=2016;,citation_cover_date=2016;,citation_year=2016;,citation_issue=2;,citation_doi=10.32614/RJ-2016-061;,citation_volume=8;,citation_journal_title=R Journal;">
<meta name="citation_reference" content="citation_title=xml2: Parse XML;,citation_author=Hadley Wickham;,citation_author=Jim Hester;,citation_author=Jeroen Ooms;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://CRAN.R-project.org/package=xml2;">
<meta name="citation_reference" content="citation_title=BARRA2: Development of the next-generation australian regional atmospheric reanalysis;,citation_author=Chun-Hsu Su;,citation_author=Susan Rennie;,citation_author=Imtiaz Dharssi;,citation_author=J. Torrance;,citation_author=A. Smith;,citation_author=T. Le;,citation_author=Peter Steinle;,citation_author=C. Stassen;,citation_author=R. A. Warren;,citation_author=C. Wang;,citation_author=J. Le Marshall;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=http://www.bom.gov.au/research/publications/researchreports/BRR-067.pdf;,citation_journal_title=Geoscientific Model Development (Bureau Research Report 067);">
<meta name="citation_reference" content="citation_title=Bureau of meteorology atmospheric high-resolution regional reanalysis for australia - version 2 (BARRA2);,citation_author=Bureau of Meteorology (BOM);,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://doi.org/10.25914/1X6G-2V48;">
<meta name="citation_reference" content="citation_title=AusBathyTopo (australia) 250m 2023 - a high-resolution depth model (20230004C);,citation_author=Robin J. Beaman;,citation_publication_date=2023;,citation_cover_date=2023;,citation_year=2023;,citation_fulltext_html_url=https://doi.org/10.26186/148758;,citation_inbook_title=undefined;">
<meta name="citation_reference" content="citation_title=Towards a dynamically balanced eddy-resolving ocean reanalysis: BRAN3;,citation_author=Peter R. Oke;,citation_author=Pavel Sakov;,citation_author=Madeleine L. Cahill;,citation_author=Jeff R. Dunn;,citation_author=Russell Fiedler;,citation_author=David A. Griffin;,citation_author=Jim V. Mansbridge;,citation_author=Ken R. Ridgway;,citation_author=Andreas Schiller;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_fulltext_html_url=https://doi.org/10.1016/j.ocemod.2013.03.008;,citation_volume=67;,citation_journal_title=Ocean Modelling;">
<meta name="citation_reference" content="citation_title=The bluelink ocean data assimilation system (BODAS);,citation_author=Peter R. Oke;,citation_author=Gary B. Brassington;,citation_author=David A. Griffin;,citation_author=Andreas Schiller;,citation_publication_date=2013;,citation_cover_date=2013;,citation_year=2013;,citation_fulltext_html_url=https://doi.org/10.1016/j.ocemod.2007.11.002;,citation_volume=21;,citation_journal_title=Ocean Modelling;">
<meta name="citation_reference" content="citation_title=Next generation of bluelink ocean reanalysis with multiscale data assimilation: BRAN2020;,citation_author=Matthew A. Chamberlain;,citation_author=Peter R. Oke;,citation_author=Russell A. S. Fiedler;,citation_author=Helen M. Beggs;,citation_author=Gary B. Brassington;,citation_author=Prasanth Divakaran;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://doi.org/10.5194/essd-13-5663-2021;,citation_volume=13;,citation_journal_title=Earth System Science Data;">
<meta name="citation_reference" content="citation_title=IMOS - combined biogeochemical parameters (reference stations);,citation_author=Integrated Marine Observing System (IMOS);,citation_author=CSIRO Oceans and Atmosphere;,citation_author=Australian Institute of Marine Science (AIMS);,citation_author=Department of Planning and Environment (DPE), New South Wales Government;,citation_author=South Australian Research and Development Institute (SARDI);,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://catalogue-imos.aodn.org.au/geonetwork/srv/eng/catalog.search#/metadata/b442c3e8-3d30-48ad-b144-680afd848167;">
<meta name="citation_reference" content="citation_title=Xarray: v0.8.0;,citation_author=Stephan Hoyer;,citation_author=Clark Fitzgerald;,citation_author=Joe Hamman;,citation_author=others;,citation_publication_date=2016-08;,citation_cover_date=2016-08;,citation_year=2016;,citation_fulltext_html_url=https://doi.org/10.5281/zenodo.59499;,citation_doi=10.5281/zenodo.59499;">
<meta name="citation_reference" content="citation_title=Xarray: N-D labeled arrays and datasets in Python;,citation_author=S. Hoyer;,citation_author=J. Hamman;,citation_publication_date=2017;,citation_cover_date=2017;,citation_year=2017;,citation_fulltext_html_url=https://doi.org/10.5334/jors.148;,citation_issue=1;,citation_doi=10.5334/jors.148;,citation_volume=5;,citation_journal_title=Journal of Open Research Software;,citation_publisher=Ubiquity Press;">
<meta name="citation_reference" content="citation_title=Pandas-dev/pandas: pandas;,citation_author=The team;,citation_publication_date=2020-02;,citation_cover_date=2020-02;,citation_year=2020;,citation_fulltext_html_url=https://doi.org/10.5281/zenodo.3509134;,citation_doi=10.5281/zenodo.3509134;,citation_publisher=Zenodo;">
<meta name="citation_reference" content="citation_title=Data Structures for Statistical Computing in Python;,citation_author=undefined McKinney;,citation_editor=undefined Walt;,citation_editor=undefined Millman;,citation_publication_date=2010;,citation_cover_date=2010;,citation_year=2010;,citation_doi= 10.25080/Majora-92bf1922-00a ;,citation_conference_title=Proceedings of the 9th Python in Science Conference;">
<meta name="citation_reference" content="citation_title=Pandas-dev/pandas: pandas;,citation_author=The team;,citation_publication_date=2020-02;,citation_cover_date=2020-02;,citation_year=2020;,citation_fulltext_html_url=https://doi.org/10.5281/zenodo.3509134;,citation_doi=10.5281/zenodo.3509134;,citation_publisher=Zenodo;">
<meta name="citation_reference" content="citation_title=SciPy 1.0: Fundamental Algorithms for Scientific Computing in Python;,citation_author=Pauli Virtanen;,citation_author=Ralf Gommers;,citation_author=Travis E. Oliphant;,citation_author=Matt Haberland;,citation_author=Tyler Reddy;,citation_author=David Cournapeau;,citation_author=Evgeni Burovski;,citation_author=Pearu Peterson;,citation_author=Warren Weckesser;,citation_author=Jonathan Bright;,citation_author=Stéfan J. Walt;,citation_author=Matthew Brett;,citation_author=Joshua Wilson;,citation_author=K. Jarrod Millman;,citation_author=Nikolay Mayorov;,citation_author=Andrew R. J. Nelson;,citation_author=Eric Jones;,citation_author=Robert Kern;,citation_author=Eric Larson;,citation_author=C J Carey;,citation_author=İlhan Polat;,citation_author=Yu Feng;,citation_author=Eric W. Moore;,citation_author=Jake VanderPlas;,citation_author=Denis Laxalde;,citation_author=Josef Perktold;,citation_author=Robert Cimrman;,citation_author=Ian Henriksen;,citation_author=E. A. Quintero;,citation_author=Charles R. Harris;,citation_author=Anne M. Archibald;,citation_author=Antônio H. Ribeiro;,citation_author=Fabian Pedregosa;,citation_author=Paul Mulbregt;,citation_author=SciPy 1.0 Contributors;,citation_publication_date=2020;,citation_cover_date=2020;,citation_year=2020;,citation_fulltext_html_url=https://doi.org/10.1038/s41592-019-0686-2;,citation_volume=17;,citation_journal_title=Nature Methods;">
<meta name="citation_reference" content="citation_title=Arrow: Integration to ’apache’ ’arrow’;,citation_author=Neal Richardson;,citation_author=Ian Cook;,citation_author=Nic Crane;,citation_author=Dewey Dunnington;,citation_author=Romain François;,citation_author=Jonathan Keane;,citation_author=Dragoș Moldovan-Grünfeld;,citation_author=Jeroen Ooms;,citation_author=Jacob Wujciak-Jens;,citation_author=Apache Arrow;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=arrow;">
<meta name="citation_reference" content="citation_title=Devtools: Tools to make developing r packages easier;,citation_author=Hadley Wickham;,citation_author=Jim Hester;,citation_author=Winston Chang;,citation_author=Jennifer Bryan;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://CRAN.R-project.org/package=devtools;">
<meta name="citation_reference" content="citation_title=Geosphere: Spherical trigonometry;,citation_author=Robert J. Hijmans;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://CRAN.R-project.org/package=geosphere;">
<meta name="citation_reference" content="citation_title=ncdf4: Interface to unidata netCDF (version 4 or earlier) format data files;,citation_author=David Pierce;,citation_publication_date=2024;,citation_cover_date=2024;,citation_year=2024;,citation_fulltext_html_url=https://CRAN.R-project.org/package=ncdf4;">
<meta name="citation_reference" content="citation_title=pacman: Package management for R;,citation_author=Tyler W. Rinker;,citation_author=Dason Kurkiewicz;,citation_publication_date=2018;,citation_cover_date=2018;,citation_year=2018;,citation_fulltext_html_url=http://github.com/trinker/pacman;">
<meta name="citation_reference" content="citation_title=A unifying framework for parallel and distributed processing in r using futures;,citation_author=Henrik Bengtsson;,citation_publication_date=2021;,citation_cover_date=2021;,citation_year=2021;,citation_fulltext_html_url=https://doi.org/10.32614/RJ-2021-048;,citation_issue=2;,citation_doi=10.32614/RJ-2021-048;,citation_volume=13;,citation_journal_title=The R Journal;">
<meta name="citation_reference" content="citation_title=Furrr: Apply mapping functions in parallel using futures;,citation_author=Davis Vaughan;,citation_author=Matt Dancho;,citation_publication_date=2022;,citation_cover_date=2022;,citation_year=2022;,citation_fulltext_html_url=https://CRAN.R-project.org/package=furrr;">
</head>

  <body class="quarto-notebook">
    <div id="quarto-embed-header" class="headroom fixed-top bg-primary">
      
      <a onclick="window.quartoBackToArticle(); return false;" class="btn btn-primary quarto-back-link" href=""><i class="bi bi-caret-left"></i> Back to Article</a>
      <h6><i class="bi bi-journal-code"></i> Article Notebook</h6>

            <a href="./manuscript.qmd" class="btn btn-primary quarto-download-embed" download="manuscript.qmd">Download Source</a>
          </div>

     <header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Local to global N input modelling</h1>
          </div>

    
    <div class="quarto-title-meta-container">
      <div class="quarto-title-meta-column-start">
        
        <div class="quarto-title-meta">

                
          
                
              </div>
      </div>
      <div class="quarto-title-meta-column-end quarto-other-formats-target">
      </div>
    </div>



    <div class="quarto-other-links-text-target">
    </div>  </div>
</header><div id="quarto-content" class="page-columns page-rows-contents page-layout-article toc-left">
<div id="quarto-sidebar-toc-left" class="sidebar toc-left">
  <nav id="TOC" role="doc-toc" class="toc-active" data-toc-expanded="99">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#methods" id="toc-methods" class="nav-link active" data-scroll-target="#methods">Methods</a>
  <ul class="collapse">
  <li><a href="#feeding-rate" id="toc-feeding-rate" class="nav-link" data-scroll-target="#feeding-rate">Feeding rate</a></li>
  <li><a href="#farm-temperature-forcings" id="toc-farm-temperature-forcings" class="nav-link" data-scroll-target="#farm-temperature-forcings">Farm temperature forcings</a></li>
  <li><a href="#production-parameters" id="toc-production-parameters" class="nav-link" data-scroll-target="#production-parameters">Production parameters</a></li>
  <li><a href="#model-parameterisation" id="toc-model-parameterisation" class="nav-link" data-scroll-target="#model-parameterisation">Model parameterisation</a>
  <ul class="collapse">
  <li><a href="#parameter-sensitivity" id="toc-parameter-sensitivity" class="nav-link" data-scroll-target="#parameter-sensitivity">Parameter sensitivity</a></li>
  </ul></li>
  <li><a href="#sec-feeds" id="toc-sec-feeds" class="nav-link" data-scroll-target="#sec-feeds">Modelled experimental feeds</a></li>
  <li><a href="#model-population-and-biomass" id="toc-model-population-and-biomass" class="nav-link" data-scroll-target="#model-population-and-biomass">Model population and biomass</a></li>
  <li><a href="#impacts-of-nutrient-input" id="toc-impacts-of-nutrient-input" class="nav-link" data-scroll-target="#impacts-of-nutrient-input">Impacts of nutrient input</a></li>
  </ul></li>
  <li><a href="#results" id="toc-results" class="nav-link" data-scroll-target="#results">Results</a>
  <ul class="collapse">
  <li><a href="#uneaten-feed" id="toc-uneaten-feed" class="nav-link" data-scroll-target="#uneaten-feed">Uneaten feed</a>
  <ul class="collapse">
  <li><a href="#amount" id="toc-amount" class="nav-link" data-scroll-target="#amount">Amount</a></li>
  </ul></li>
  <li><a href="#excreted-faeces" id="toc-excreted-faeces" class="nav-link" data-scroll-target="#excreted-faeces">Excreted faeces</a>
  <ul class="collapse">
  <li><a href="#amount-1" id="toc-amount-1" class="nav-link" data-scroll-target="#amount-1">Amount</a></li>
  <li><a href="#composition" id="toc-composition" class="nav-link" data-scroll-target="#composition">Composition</a></li>
  </ul></li>
  <li><a href="#total-nutrients" id="toc-total-nutrients" class="nav-link" data-scroll-target="#total-nutrients">Total nutrients</a></li>
  </ul></li>
  <li><a href="#statistical-analysis-code-availability" id="toc-statistical-analysis-code-availability" class="nav-link" data-scroll-target="#statistical-analysis-code-availability">Statistical analysis &amp; code availability</a></li>
  <li><a href="#supplementary-stuff" id="toc-supplementary-stuff" class="nav-link" data-scroll-target="#supplementary-stuff">Supplementary stuff</a>
  <ul class="collapse">
  <li><a href="#sec-parameter-sensitivity" id="toc-sec-parameter-sensitivity" class="nav-link" data-scroll-target="#sec-parameter-sensitivity">Parameter sensitivity</a></li>
  </ul></li>
  <li><a href="#references" id="toc-references" class="nav-link" data-scroll-target="#references">References</a></li>
  </ul>
</nav>
</div>
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar zindex-bottom">
</div>
<main class="content quarto-banner-title-block" id="quarto-document-content">      

       <div class="cell-container code-fold"><div class="cell-decorator"><pre>In [1]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magrittr)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'dplyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:stats':

    filter, lag</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:base':

    intersect, setdiff, setequal, union</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'tidyr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:magrittr':

    extract</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(stringr)</span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggplot2' was built under R version 4.4.3</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggh4x)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'ggh4x' was built under R version 4.4.3</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>here() starts at C:/Users/treimer/Documents/R-temp-files/local_to_global_mariculture_modelling</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggpubr)</span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(cowplot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'cowplot'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggpubr':

    get_legend</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(patchwork)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'patchwork'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:cowplot':

    align_plots</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(qs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'qs' was built under R version 4.4.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>qs 0.27.3. Announcement: https://github.com/qsbase/qs/issues/103</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(arrow)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'arrow'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:magrittr':

    is_in</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:utils':

    timestamp</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(targets)</span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Linking to GEOS 3.13.0, GDAL 3.10.1, PROJ 9.5.1; sf_use_s2() is TRUE</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(terra)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>terra 1.8.50</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'terra'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:arrow':

    buffer</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:patchwork':

    area</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:ggpubr':

    rotate</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:tidyr':

    extract</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following objects are masked from 'package:magrittr':

    extract, inset</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rnaturalearth)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rnaturalearthdata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'rnaturalearthdata'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:rnaturalearth':

    countries110</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(rnaturalearthhires)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dtplyr)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(furrr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Loading required package: future</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb42-2"><a href="#cb42-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(purrr)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>
Attaching package: 'purrr'</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>The following object is masked from 'package:magrittr':

    set_names</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb45"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(future)</span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'readxl' was built under R version 4.4.3</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb47"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb47-1"><a href="#cb47-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(units)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: package 'units' was built under R version 4.4.3</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>udunits database from C:/Users/treimer/AppData/Local/R/cache/R/renv/cache/v5/windows/R-4.4/x86_64-w64-mingw32/units/0.8-7/5d0b024902d5da97a2b64f002e92a869/units/share/udunits/udunits2.xml</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(conflicted)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lubridate)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="fu">conflicts_prefer</span>(dplyr<span class="sc">::</span><span class="fu">select</span>(), dplyr<span class="sc">::</span><span class="fu">filter</span>(), dplyr<span class="sc">::</span><span class="fu">intersect</span>(), <span class="at">.quiet =</span> T)</span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co"># This markdown uses TinyTex - install with tinytex::install_tinytex()</span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"g_fish"</span>)</span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"kg_fish"</span>)</span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a><span class="fu">remove_unit</span>(<span class="st">"t_fish"</span>)</span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"g_fish"</span>)</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"kg_fish"</span>, <span class="st">"1000 g_fish"</span>)</span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a><span class="fu">install_unit</span>(<span class="st">"t_fish"</span>, <span class="st">"1000 kg_fish"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [2]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb51"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb51-1"><a href="#cb51-1" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">here</span>(<span class="st">"00_model_functions.R"</span>))</span>
<span id="cb51-2"><a href="#cb51-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-3"><a href="#cb51-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Input paths</span></span>
<span id="cb51-4"><a href="#cb51-4" aria-hidden="true" tabindex="-1"></a>input_farm_coords_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"_general_data"</span>, <span class="st">"farm_locations"</span>)</span>
<span id="cb51-5"><a href="#cb51-5" aria-hidden="true" tabindex="-1"></a>input_farm_sst_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"_general_data"</span>, <span class="st">"SST"</span>)</span>
<span id="cb51-6"><a href="#cb51-6" aria-hidden="true" tabindex="-1"></a>input_species_param_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"atlantic_salmon"</span>, <span class="st">"params"</span>)</span>
<span id="cb51-7"><a href="#cb51-7" aria-hidden="true" tabindex="-1"></a>input_feed_profile_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="st">"_general_data"</span>, <span class="st">"diets"</span>)</span>
<span id="cb51-8"><a href="#cb51-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-9"><a href="#cb51-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Output paths</span></span>
<span id="cb51-10"><a href="#cb51-10" aria-hidden="true" tabindex="-1"></a>output_farm_data_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"outputs"</span>, <span class="st">"farm_data"</span>)</span>
<span id="cb51-11"><a href="#cb51-11" aria-hidden="true" tabindex="-1"></a>output_species_data_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"outputs"</span>, <span class="st">"species_data"</span>)</span>
<span id="cb51-12"><a href="#cb51-12" aria-hidden="true" tabindex="-1"></a>output_sens_data_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"outputs"</span>, <span class="st">"sensitivity_data"</span>)</span>
<span id="cb51-13"><a href="#cb51-13" aria-hidden="true" tabindex="-1"></a>output_farm_growth_data_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"outputs"</span>, <span class="st">"farm_growth_data"</span>)</span>
<span id="cb51-14"><a href="#cb51-14" aria-hidden="true" tabindex="-1"></a>output_cohort_growth_data_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"outputs"</span>, <span class="st">"cohort_growth_data"</span>)</span>
<span id="cb51-15"><a href="#cb51-15" aria-hidden="true" tabindex="-1"></a>data_analysis_path <span class="ot">&lt;-</span> <span class="fu">here</span>() <span class="sc">%&gt;%</span> <span class="fu">file.path</span>(<span class="st">"outputs"</span>, <span class="st">"data_analysis"</span>)</span>
<span id="cb51-16"><a href="#cb51-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-17"><a href="#cb51-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Filenames</span></span>
<span id="cb51-18"><a href="#cb51-18" aria-hidden="true" tabindex="-1"></a>farm_coords_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_farm_data_path, <span class="st">"farm_coords.qs"</span>)</span>
<span id="cb51-19"><a href="#cb51-19" aria-hidden="true" tabindex="-1"></a>species_params_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_species_data_path, <span class="st">"species_params.qs"</span>)</span>
<span id="cb51-20"><a href="#cb51-20" aria-hidden="true" tabindex="-1"></a>pop_params_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_species_data_path, <span class="st">"pop_params.qs"</span>)</span>
<span id="cb51-21"><a href="#cb51-21" aria-hidden="true" tabindex="-1"></a>feed_params_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_species_data_path, <span class="st">"feed_params.qs"</span>)</span>
<span id="cb51-22"><a href="#cb51-22" aria-hidden="true" tabindex="-1"></a>sens_params_file <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_species_data_path, <span class="st">"sens_params.qs"</span>)</span>
<span id="cb51-23"><a href="#cb51-23" aria-hidden="true" tabindex="-1"></a>sens_results_files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_sens_data_path) <span class="sc">%&gt;%</span> <span class="fu">str_subset</span>(<span class="st">"results"</span>)</span>
<span id="cb51-24"><a href="#cb51-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-25"><a href="#cb51-25" aria-hidden="true" tabindex="-1"></a>species_params <span class="ot">&lt;-</span> <span class="fu">qread</span>(species_params_file)</span>
<span id="cb51-26"><a href="#cb51-26" aria-hidden="true" tabindex="-1"></a>pop_params <span class="ot">&lt;-</span> <span class="fu">qread</span>(pop_params_file)</span>
<span id="cb51-27"><a href="#cb51-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb51-28"><a href="#cb51-28" aria-hidden="true" tabindex="-1"></a>prettyplot <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb51-29"><a href="#cb51-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb51-30"><a href="#cb51-30" aria-hidden="true" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>,</span>
<span id="cb51-31"><a href="#cb51-31" aria-hidden="true" tabindex="-1"></a>          <span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">family =</span> <span class="st">"serif"</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">colour =</span> <span class="st">"black"</span>),</span>
<span id="cb51-32"><a href="#cb51-32" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">vjust =</span> <span class="fl">0.5</span>),</span>
<span id="cb51-33"><a href="#cb51-33" aria-hidden="true" tabindex="-1"></a>          <span class="at">axis.title.y =</span> <span class="fu">element_text</span>(<span class="at">hjust =</span> <span class="fl">0.5</span>))</span>
<span id="cb51-34"><a href="#cb51-34" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb51-35"><a href="#cb51-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb51-36"><a href="#cb51-36" aria-hidden="true" tabindex="-1"></a>feed_pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"past"</span> <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">1</span>], <span class="st">"reference"</span> <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">2</span>], <span class="st">"future"</span> <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">3</span>])</span>
<span id="cb51-37"><a href="#cb51-37" aria-hidden="true" tabindex="-1"></a>cohort_pal <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="st">"1"</span> <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">1</span>], <span class="st">"2"</span> <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">2</span>], <span class="st">"3"</span> <span class="ot">=</span> <span class="fu">brewer.pal</span>(<span class="dv">9</span>, <span class="st">"Set1"</span>)[<span class="dv">3</span>], <span class="st">"all"</span> <span class="ot">=</span> <span class="st">"black"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [3]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Function to create sf polygons from bounding boxes</summary>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a>create_boxes <span class="ot">&lt;-</span> <span class="cf">function</span>(box_list, label_positions, offset_deg) {</span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a>  boxes <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>  labels <span class="ot">&lt;-</span> <span class="fu">list</span>()</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a>  <span class="cf">for</span>(i <span class="cf">in</span> <span class="fu">seq_along</span>(box_list)) {</span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a>    name <span class="ot">&lt;-</span> <span class="fu">names</span>(box_list)[i]</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a>    coords <span class="ot">&lt;-</span> box_list[[i]]</span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a>    pos <span class="ot">&lt;-</span> label_positions[i]</span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb52-10"><a href="#cb52-10" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create polygon from bounding box coordinates</span></span>
<span id="cb52-11"><a href="#cb52-11" aria-hidden="true" tabindex="-1"></a>    box_coords <span class="ot">&lt;-</span> <span class="fu">matrix</span>(<span class="fu">c</span>(</span>
<span id="cb52-12"><a href="#cb52-12" aria-hidden="true" tabindex="-1"></a>      coords[<span class="dv">1</span>], coords[<span class="dv">3</span>],  <span class="co"># lonmin, latmin</span></span>
<span id="cb52-13"><a href="#cb52-13" aria-hidden="true" tabindex="-1"></a>      coords[<span class="dv">2</span>], coords[<span class="dv">3</span>],  <span class="co"># lonmax, latmin  </span></span>
<span id="cb52-14"><a href="#cb52-14" aria-hidden="true" tabindex="-1"></a>      coords[<span class="dv">2</span>], coords[<span class="dv">4</span>],  <span class="co"># lonmax, latmax</span></span>
<span id="cb52-15"><a href="#cb52-15" aria-hidden="true" tabindex="-1"></a>      coords[<span class="dv">1</span>], coords[<span class="dv">4</span>],  <span class="co"># lonmin, latmax</span></span>
<span id="cb52-16"><a href="#cb52-16" aria-hidden="true" tabindex="-1"></a>      coords[<span class="dv">1</span>], coords[<span class="dv">3</span>]   <span class="co"># close polygon</span></span>
<span id="cb52-17"><a href="#cb52-17" aria-hidden="true" tabindex="-1"></a>    ), <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">byrow =</span> <span class="cn">TRUE</span>)</span>
<span id="cb52-18"><a href="#cb52-18" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb52-19"><a href="#cb52-19" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Create sf polygon</span></span>
<span id="cb52-20"><a href="#cb52-20" aria-hidden="true" tabindex="-1"></a>    poly <span class="ot">&lt;-</span> <span class="fu">st_polygon</span>(<span class="fu">list</span>(box_coords))</span>
<span id="cb52-21"><a href="#cb52-21" aria-hidden="true" tabindex="-1"></a>    boxes[[i]] <span class="ot">&lt;-</span> <span class="fu">st_sfc</span>(poly, <span class="at">crs =</span> <span class="dv">4326</span>)</span>
<span id="cb52-22"><a href="#cb52-22" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb52-23"><a href="#cb52-23" aria-hidden="true" tabindex="-1"></a>    <span class="co"># Calculate label position based on specification</span></span>
<span id="cb52-24"><a href="#cb52-24" aria-hidden="true" tabindex="-1"></a>    label_coords <span class="ot">&lt;-</span> <span class="fu">get_label_position</span>(coords, pos, offset_deg[i])</span>
<span id="cb52-25"><a href="#cb52-25" aria-hidden="true" tabindex="-1"></a>    label_point <span class="ot">&lt;-</span> <span class="fu">st_point</span>(<span class="fu">c</span>(label_coords[<span class="dv">1</span>], label_coords[<span class="dv">2</span>]))</span>
<span id="cb52-26"><a href="#cb52-26" aria-hidden="true" tabindex="-1"></a>    labels[[i]] <span class="ot">&lt;-</span> <span class="fu">st_sfc</span>(label_point, <span class="at">crs =</span> <span class="dv">4326</span>)</span>
<span id="cb52-27"><a href="#cb52-27" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb52-28"><a href="#cb52-28" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-29"><a href="#cb52-29" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Combine into sf objects</span></span>
<span id="cb52-30"><a href="#cb52-30" aria-hidden="true" tabindex="-1"></a>  boxes_sf <span class="ot">&lt;-</span> <span class="fu">st_sf</span>(</span>
<span id="cb52-31"><a href="#cb52-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">names</span>(box_list),</span>
<span id="cb52-32"><a href="#cb52-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">letter =</span> LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(box_list)],</span>
<span id="cb52-33"><a href="#cb52-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">geometry =</span> <span class="fu">do.call</span>(c, boxes)</span>
<span id="cb52-34"><a href="#cb52-34" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb52-35"><a href="#cb52-35" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-36"><a href="#cb52-36" aria-hidden="true" tabindex="-1"></a>  labels_sf <span class="ot">&lt;-</span> <span class="fu">st_sf</span>(</span>
<span id="cb52-37"><a href="#cb52-37" aria-hidden="true" tabindex="-1"></a>    <span class="at">name =</span> <span class="fu">names</span>(box_list), </span>
<span id="cb52-38"><a href="#cb52-38" aria-hidden="true" tabindex="-1"></a>    <span class="at">letter =</span> LETTERS[<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(box_list)],</span>
<span id="cb52-39"><a href="#cb52-39" aria-hidden="true" tabindex="-1"></a>    <span class="at">geometry =</span> <span class="fu">do.call</span>(c, labels)</span>
<span id="cb52-40"><a href="#cb52-40" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb52-41"><a href="#cb52-41" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-42"><a href="#cb52-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">list</span>(<span class="at">boxes =</span> boxes_sf, <span class="at">labels =</span> labels_sf))</span>
<span id="cb52-43"><a href="#cb52-43" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb52-44"><a href="#cb52-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-45"><a href="#cb52-45" aria-hidden="true" tabindex="-1"></a><span class="co"># Helper function to calculate label positions</span></span>
<span id="cb52-46"><a href="#cb52-46" aria-hidden="true" tabindex="-1"></a>get_label_position <span class="ot">&lt;-</span> <span class="cf">function</span>(coords, position, offset) {</span>
<span id="cb52-47"><a href="#cb52-47" aria-hidden="true" tabindex="-1"></a>  lonmin <span class="ot">&lt;-</span> coords[<span class="dv">1</span>]</span>
<span id="cb52-48"><a href="#cb52-48" aria-hidden="true" tabindex="-1"></a>  lonmax <span class="ot">&lt;-</span> coords[<span class="dv">2</span>]</span>
<span id="cb52-49"><a href="#cb52-49" aria-hidden="true" tabindex="-1"></a>  latmin <span class="ot">&lt;-</span> coords[<span class="dv">3</span>]</span>
<span id="cb52-50"><a href="#cb52-50" aria-hidden="true" tabindex="-1"></a>  latmax <span class="ot">&lt;-</span> coords[<span class="dv">4</span>]</span>
<span id="cb52-51"><a href="#cb52-51" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-52"><a href="#cb52-52" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Parse position string</span></span>
<span id="cb52-53"><a href="#cb52-53" aria-hidden="true" tabindex="-1"></a>  parts <span class="ot">&lt;-</span> <span class="fu">strsplit</span>(position, <span class="st">"_"</span>)[[<span class="dv">1</span>]]</span>
<span id="cb52-54"><a href="#cb52-54" aria-hidden="true" tabindex="-1"></a>  vertical <span class="ot">&lt;-</span> parts[<span class="dv">1</span>]    <span class="co"># "top" or "bottom"</span></span>
<span id="cb52-55"><a href="#cb52-55" aria-hidden="true" tabindex="-1"></a>  horizontal <span class="ot">&lt;-</span> parts[<span class="dv">2</span>]  <span class="co"># "left" or "right"</span></span>
<span id="cb52-56"><a href="#cb52-56" aria-hidden="true" tabindex="-1"></a>  location <span class="ot">&lt;-</span> parts[<span class="dv">3</span>]    <span class="co"># "inside" or "outside"</span></span>
<span id="cb52-57"><a href="#cb52-57" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-58"><a href="#cb52-58" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Base coordinates for corners</span></span>
<span id="cb52-59"><a href="#cb52-59" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(vertical <span class="sc">==</span> <span class="st">"bottom"</span>) {base_lat <span class="ot">&lt;-</span> latmin} <span class="cf">else</span> {base_lat <span class="ot">&lt;-</span> latmax}</span>
<span id="cb52-60"><a href="#cb52-60" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(horizontal <span class="sc">==</span> <span class="st">"right"</span>) {base_lon <span class="ot">&lt;-</span> lonmax} <span class="cf">else</span> {base_lon <span class="ot">&lt;-</span> lonmin}</span>
<span id="cb52-61"><a href="#cb52-61" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-62"><a href="#cb52-62" aria-hidden="true" tabindex="-1"></a>  <span class="co"># Apply offset for outside positioning</span></span>
<span id="cb52-63"><a href="#cb52-63" aria-hidden="true" tabindex="-1"></a>  <span class="cf">if</span>(location <span class="sc">==</span> <span class="st">"outside"</span>) {</span>
<span id="cb52-64"><a href="#cb52-64" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(horizontal <span class="sc">==</span> <span class="st">"left"</span>) {base_lon <span class="ot">&lt;-</span> base_lon<span class="sc">-</span>offset} <span class="cf">else</span> {base_lon <span class="ot">&lt;-</span> base_lon<span class="sc">+</span>offset}</span>
<span id="cb52-65"><a href="#cb52-65" aria-hidden="true" tabindex="-1"></a>    <span class="cf">if</span>(vertical <span class="sc">==</span> <span class="st">"top"</span>) {base_lat <span class="ot">&lt;-</span> base_lat <span class="sc">+</span> offset} <span class="cf">else</span> {base_lat <span class="ot">&lt;-</span> base_lat <span class="sc">-</span> offset}</span>
<span id="cb52-66"><a href="#cb52-66" aria-hidden="true" tabindex="-1"></a>  }</span>
<span id="cb52-67"><a href="#cb52-67" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb52-68"><a href="#cb52-68" aria-hidden="true" tabindex="-1"></a>  <span class="fu">return</span>(<span class="fu">c</span>(base_lon, base_lat))</span>
<span id="cb52-69"><a href="#cb52-69" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb52-70"><a href="#cb52-70" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-71"><a href="#cb52-71" aria-hidden="true" tabindex="-1"></a>lim_robin <span class="ot">&lt;-</span> <span class="cf">function</span>(<span class="at">lons =</span> <span class="fu">c</span>(<span class="fl">144.0</span>, <span class="fl">149.5</span>), <span class="at">lats =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">39.75</span>, <span class="sc">-</span><span class="fl">44.00</span>)) {</span>
<span id="cb52-72"><a href="#cb52-72" aria-hidden="true" tabindex="-1"></a>  coords <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">lon =</span> lons, <span class="at">lat =</span> lats) <span class="sc">%&gt;%</span></span>
<span id="cb52-73"><a href="#cb52-73" aria-hidden="true" tabindex="-1"></a>    <span class="fu">st_as_sf</span>(<span class="at">coords =</span> <span class="fu">c</span>(<span class="st">"lon"</span>, <span class="st">"lat"</span>), <span class="at">crs =</span> <span class="dv">4326</span>) <span class="sc">%&gt;%</span></span>
<span id="cb52-74"><a href="#cb52-74" aria-hidden="true" tabindex="-1"></a>    <span class="fu">st_transform</span>(<span class="at">crs =</span> <span class="st">"+proj=robin"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb52-75"><a href="#cb52-75" aria-hidden="true" tabindex="-1"></a>    <span class="fu">st_coordinates</span>()</span>
<span id="cb52-76"><a href="#cb52-76" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list</span>(</span>
<span id="cb52-77"><a href="#cb52-77" aria-hidden="true" tabindex="-1"></a>    <span class="at">xlims =</span> <span class="fu">range</span>(coords[, <span class="st">"X"</span>]),</span>
<span id="cb52-78"><a href="#cb52-78" aria-hidden="true" tabindex="-1"></a>    <span class="at">ylims =</span> <span class="fu">range</span>(coords[, <span class="st">"Y"</span>])</span>
<span id="cb52-79"><a href="#cb52-79" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb52-80"><a href="#cb52-80" aria-hidden="true" tabindex="-1"></a>}</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>This document was last updated at 20:00 on 06 Jun, 2025.</p>
<section id="methods" class="level1">
<h1>Methods</h1>
<section id="feeding-rate" class="level3">
<h3 class="anchored" data-anchor-id="feeding-rate">Feeding rate</h3>
<p>To detail here:</p>
<ul>
<li>Functional response to temperature (species-specific)</li>
<li>Overfeeding parameters and how they come into play</li>
<li>Deminstration of how feeding response and actual food given vary with temperature?</li>
</ul>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [4]:</pre></div><div id="cell-fig-functional-response-to-temperature" class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb53"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb53-1"><a href="#cb53-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">water_temp =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">5</span>, <span class="dv">26</span>, <span class="fl">0.25</span>))</span>
<span id="cb53-2"><a href="#cb53-2" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>feeding <span class="ot">&lt;-</span> <span class="fu">sapply</span>(<span class="at">FUN =</span> feeding_rate, <span class="at">X =</span> df<span class="sc">$</span>water_temp, <span class="at">species_params =</span> species_params)</span>
<span id="cb53-3"><a href="#cb53-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb53-4"><a href="#cb53-4" aria-hidden="true" tabindex="-1"></a><span class="fu">mutate</span>(<span class="at">ing_pot =</span> pop_params[<span class="st">'meanImax'</span>] <span class="sc">*</span> (pop_params[<span class="st">'meanW'</span>]<span class="sc">^</span>species_params[<span class="st">'m'</span>]) <span class="sc">*</span> feeding,</span>
<span id="cb53-5"><a href="#cb53-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">food_prov =</span> <span class="cn">NA</span>)</span>
<span id="cb53-6"><a href="#cb53-6" aria-hidden="true" tabindex="-1"></a><span class="cf">for</span> (i <span class="cf">in</span> <span class="dv">1</span><span class="sc">:</span><span class="fu">nrow</span>(df)) {</span>
<span id="cb53-7"><a href="#cb53-7" aria-hidden="true" tabindex="-1"></a>  df<span class="sc">$</span>food_prov[i] <span class="ot">&lt;-</span> <span class="fu">food_prov_rate</span>(<span class="at">pop_params =</span> pop_params, <span class="at">rel_feeding =</span> df<span class="sc">$</span>feeding[i], <span class="at">ing_pot =</span> df<span class="sc">$</span>ing_pot[i], </span>
<span id="cb53-8"><a href="#cb53-8" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">ing_pot_10 =</span> pop_params[<span class="st">'meanImax'</span>] <span class="sc">*</span> (pop_params[<span class="st">'meanW'</span>]<span class="sc">^</span>species_params[<span class="st">'m'</span>]) <span class="sc">*</span> <span class="fl">0.1</span>)</span>
<span id="cb53-9"><a href="#cb53-9" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb53-10"><a href="#cb53-10" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>ing_pot <span class="ot">&lt;-</span> df<span class="sc">$</span>ing_pot<span class="sc">/</span><span class="fu">maxna</span>(df<span class="sc">$</span>ing_pot)</span>
<span id="cb53-11"><a href="#cb53-11" aria-hidden="true" tabindex="-1"></a>df<span class="sc">$</span>food_prov <span class="ot">&lt;-</span> df<span class="sc">$</span>food_prov<span class="sc">/</span><span class="fu">maxna</span>(df<span class="sc">$</span>ing_pot)</span>
<span id="cb53-12"><a href="#cb53-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb53-13"><a href="#cb53-13" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb53-14"><a href="#cb53-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> water_temp, <span class="at">y =</span> ing_pot)) <span class="sc">+</span></span>
<span id="cb53-15"><a href="#cb53-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="fl">0.75</span>) <span class="sc">+</span></span>
<span id="cb53-16"><a href="#cb53-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">y =</span> food_prov), <span class="at">linewidth =</span> <span class="fl">0.75</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>) <span class="sc">+</span></span>
<span id="cb53-17"><a href="#cb53-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>,<span class="fl">1.5</span>,<span class="fl">0.25</span>)) <span class="sc">+</span></span>
<span id="cb53-18"><a href="#cb53-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">5</span>,<span class="dv">30</span>,<span class="dv">5</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">5</span>,<span class="dv">26</span>)) <span class="sc">+</span></span>
<span id="cb53-19"><a href="#cb53-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Water temperature ("</span><span class="sc">*</span>degree<span class="sc">*</span><span class="st">"C)"</span>), <span class="at">y =</span> <span class="st">"Relative food ingested/provided"</span>) <span class="sc">+</span></span>
<span id="cb53-20"><a href="#cb53-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb53-21"><a href="#cb53-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">text =</span> <span class="fu">element_text</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">family =</span> <span class="st">"serif"</span>, <span class="at">size =</span> <span class="dv">12</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
</section>
<section id="farm-temperature-forcings" class="level2">
<h2 class="anchored" data-anchor-id="farm-temperature-forcings">Farm temperature forcings</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [5]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Get general farm info</summary>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a>farms_to_omit <span class="ot">&lt;-</span> <span class="fu">file.path</span>(input_farm_coords_path, <span class="st">"atlantic_salmon_farms_to_omit.qs"</span>) <span class="sc">%&gt;%</span> qs<span class="sc">::</span><span class="fu">qread</span>()</span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>farm_ts_data <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_farm_data_path, <span class="st">"farm_ts_data.qs"</span>) <span class="sc">%&gt;%</span> qs<span class="sc">::</span><span class="fu">qread</span>()</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a>farm_coords <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_farm_data_path, <span class="st">"farm_coords.qs"</span>) <span class="sc">%&gt;%</span> qs<span class="sc">::</span><span class="fu">qread</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">hemisphere =</span> <span class="fu">case_when</span>(lat <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"S"</span>, T <span class="sc">~</span> <span class="st">"N"</span>) <span class="sc">%&gt;%</span> <span class="fu">as.factor</span>()) </span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-6"><a href="#cb54-6" aria-hidden="true" tabindex="-1"></a>farms_geometry <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_farm_data_path, <span class="st">"farm_geometry.qs"</span>) <span class="sc">%&gt;%</span> qs<span class="sc">::</span><span class="fu">qread</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb54-7"><a href="#cb54-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">farm_ID =</span> farm_id)</span>
<span id="cb54-8"><a href="#cb54-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-9"><a href="#cb54-9" aria-hidden="true" tabindex="-1"></a>sorted_countries <span class="ot">&lt;-</span> farm_coords <span class="sc">%&gt;%</span> </span>
<span id="cb54-10"><a href="#cb54-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(farms_geometry, <span class="at">by =</span> <span class="st">"farm_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb54-11"><a href="#cb54-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(country) <span class="sc">%&gt;%</span> </span>
<span id="cb54-12"><a href="#cb54-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">av_lat =</span> <span class="fu">mean</span>(lat)) <span class="sc">%&gt;%</span> </span>
<span id="cb54-13"><a href="#cb54-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="sc">-</span>av_lat) <span class="sc">%&gt;%</span> </span>
<span id="cb54-14"><a href="#cb54-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(country) <span class="sc">%&gt;%</span> </span>
<span id="cb54-15"><a href="#cb54-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">unique</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [6]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Get the farm temperature forcings</summary>
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a>farm_temp_means <span class="ot">&lt;-</span> farm_ts_data <span class="sc">%&gt;%</span> </span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(farm_ID) <span class="sc">%&gt;%</span> </span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">mean_temp =</span> <span class="fu">mean</span>(temp_c)) <span class="sc">%&gt;%</span> </span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(farms_geometry, <span class="at">by =</span> <span class="st">"farm_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(farm_coords, <span class="at">by =</span> <span class="st">"farm_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">filter</span>(<span class="sc">!</span>farm_ID <span class="sc">%in%</span> farms_to_omit)</span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-8"><a href="#cb55-8" aria-hidden="true" tabindex="-1"></a>overall_mean <span class="ot">&lt;-</span> <span class="fu">mean</span>(farm_temp_means<span class="sc">$</span>mean_temp)</span>
<span id="cb55-9"><a href="#cb55-9" aria-hidden="true" tabindex="-1"></a>overall_sd <span class="ot">&lt;-</span> <span class="fu">sd</span>(farm_temp_means<span class="sc">$</span>mean_temp)</span>
<span id="cb55-10"><a href="#cb55-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-11"><a href="#cb55-11" aria-hidden="true" tabindex="-1"></a><span class="fu">rm</span>(farm_ts_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>Mean daily temperatures at the farms globally ranged from from 6<span class="math inline">\(^{\circ}\)</span>C to 16.7<span class="math inline">\(^{\circ}\)</span>C, with an overall mean of 10.1<span class="math inline">\(^{\circ}\)</span>C (<span class="math inline">\(\pm\)</span> 1.8<span class="math inline">\(^{\circ}\)</span>C SD). Farms with a mean temperature of $<span class="math inline">\(6\)</span>^$C were excluded (4 farms). The total number of farms included in the analysis was therefore 2717.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [7]:</pre></div><div id="cell-fig-farm-mean-temps" class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a>worldmap <span class="ot">&lt;-</span> <span class="fu">ne_countries</span>(<span class="at">scale =</span> <span class="st">"medium"</span>, <span class="at">returnclass =</span> <span class="st">"sf"</span>) <span class="sc">%&gt;%</span> <span class="fu">st_transform</span>(<span class="at">crs =</span> <span class="st">"+proj=robin"</span>)</span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>graticules <span class="ot">&lt;-</span> <span class="fu">st_graticule</span>(worldmap, <span class="at">lon =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">180</span>, <span class="dv">180</span>, <span class="dv">30</span>), <span class="at">lat =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">90</span>, <span class="dv">90</span>, <span class="dv">30</span>)) </span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a>inset_boxes <span class="ot">&lt;-</span> <span class="fu">list</span>(   <span class="co"># lonmin, lonmax, latmin, latmax</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">CAN1 =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">132</span>, <span class="sc">-</span><span class="dv">122</span>, <span class="fl">47.75</span>, <span class="fl">54.25</span>),</span>
<span id="cb56-6"><a href="#cb56-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">CAN2 =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">70</span>, <span class="sc">-</span><span class="dv">54</span>, <span class="dv">43</span>, <span class="fl">48.5</span>),</span>
<span id="cb56-7"><a href="#cb56-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">EUR =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">26</span>, <span class="dv">30</span>, <span class="dv">51</span>, <span class="dv">71</span>),</span>
<span id="cb56-8"><a href="#cb56-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">CHI =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">77.5</span>, <span class="sc">-</span><span class="fl">62.5</span>, <span class="sc">-</span><span class="dv">56</span>, <span class="sc">-</span><span class="dv">25</span>),</span>
<span id="cb56-9"><a href="#cb56-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">AUS =</span> <span class="fu">c</span>(<span class="dv">144</span>, <span class="fl">149.5</span>, <span class="sc">-</span><span class="fl">39.75</span>, <span class="sc">-</span><span class="dv">44</span>)</span>
<span id="cb56-10"><a href="#cb56-10" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb56-11"><a href="#cb56-11" aria-hidden="true" tabindex="-1"></a>labels_spec <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">CAN1 =</span> <span class="st">"bottom_left_outside"</span>, <span class="at">CAN2 =</span> <span class="st">"bottom_right_outside"</span>, <span class="at">EUR =</span> <span class="st">"bottom_left_outside"</span>, <span class="at">CHI =</span> <span class="st">"top_left_outside"</span>, <span class="at">AUS =</span> <span class="st">"bottom_left_outside"</span>)</span>
<span id="cb56-12"><a href="#cb56-12" aria-hidden="true" tabindex="-1"></a>labels_offset <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="at">CAN1 =</span> <span class="dv">1</span>, <span class="at">CAN2 =</span> <span class="dv">2</span>, <span class="at">EUR =</span> <span class="fl">1.75</span>, <span class="at">CHI =</span> <span class="fl">1.75</span>, <span class="at">AUS =</span> <span class="fl">3.5</span>)</span>
<span id="cb56-13"><a href="#cb56-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-14"><a href="#cb56-14" aria-hidden="true" tabindex="-1"></a>box_data <span class="ot">&lt;-</span> <span class="fu">create_boxes</span>(<span class="at">box_list =</span> inset_boxes, <span class="co"># see function create_boxes above</span></span>
<span id="cb56-15"><a href="#cb56-15" aria-hidden="true" tabindex="-1"></a>                         <span class="at">label_positions =</span> <span class="fu">unname</span>(labels_spec),</span>
<span id="cb56-16"><a href="#cb56-16" aria-hidden="true" tabindex="-1"></a>                         <span class="at">offset_deg =</span> <span class="fu">unname</span>(labels_offset))</span>
<span id="cb56-17"><a href="#cb56-17" aria-hidden="true" tabindex="-1"></a>boxes_robinson <span class="ot">&lt;-</span> <span class="fu">st_transform</span>(box_data<span class="sc">$</span>boxes, <span class="at">crs =</span> <span class="st">"+proj=robin"</span>)</span>
<span id="cb56-18"><a href="#cb56-18" aria-hidden="true" tabindex="-1"></a>labels_robinson <span class="ot">&lt;-</span> <span class="fu">st_transform</span>(box_data<span class="sc">$</span>labels, <span class="at">crs =</span> <span class="st">"+proj=robin"</span>)</span>
<span id="cb56-19"><a href="#cb56-19" aria-hidden="true" tabindex="-1"></a>farm_temp_robinson <span class="ot">&lt;-</span> <span class="fu">st_sf</span>(farm_temp_means) <span class="sc">%&gt;%</span> <span class="fu">st_transform</span>(<span class="at">crs =</span> <span class="st">"+proj=robin"</span>)</span>
<span id="cb56-20"><a href="#cb56-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-21"><a href="#cb56-21" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb56-22"><a href="#cb56-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> graticules, <span class="at">color =</span> <span class="st">"gray80"</span>, <span class="at">size =</span> <span class="fl">0.3</span>) <span class="sc">+</span></span>
<span id="cb56-23"><a href="#cb56-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> worldmap, <span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"dimgray"</span>) <span class="sc">+</span></span>
<span id="cb56-24"><a href="#cb56-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> farm_temp_robinson, <span class="fu">aes</span>(<span class="at">color =</span> mean_temp), <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb56-25"><a href="#cb56-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>() <span class="sc">+</span></span>
<span id="cb56-26"><a href="#cb56-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">option =</span> <span class="st">"turbo"</span>) <span class="sc">+</span></span>
<span id="cb56-27"><a href="#cb56-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">col =</span> <span class="fu">guide_colourbar</span>(</span>
<span id="cb56-28"><a href="#cb56-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Mean temperature (°C)"</span>,</span>
<span id="cb56-29"><a href="#cb56-29" aria-hidden="true" tabindex="-1"></a>    <span class="at">direction =</span> <span class="st">"horizontal"</span>, <span class="at">position =</span> <span class="st">"bottom"</span>,</span>
<span id="cb56-30"><a href="#cb56-30" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.position =</span> <span class="st">"bottom"</span>, <span class="at">title.position =</span> <span class="st">"top"</span>, </span>
<span id="cb56-31"><a href="#cb56-31" aria-hidden="true" tabindex="-1"></a>    <span class="at">title.vjust =</span> <span class="dv">1</span>, <span class="at">title.hjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb56-32"><a href="#cb56-32" aria-hidden="true" tabindex="-1"></a>    <span class="at">frame.colour =</span> <span class="st">"black"</span>, <span class="at">ticks.colour =</span> <span class="st">"black"</span>,<span class="at">ticks.linewidth =</span> <span class="dv">2</span>,</span>
<span id="cb56-33"><a href="#cb56-33" aria-hidden="true" tabindex="-1"></a>    <span class="at">barwidth =</span> <span class="dv">25</span>,<span class="at">barheight =</span> <span class="fl">1.5</span></span>
<span id="cb56-34"><a href="#cb56-34" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb56-35"><a href="#cb56-35" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> boxes_robinson, <span class="at">fill =</span> <span class="cn">NA</span>, <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="fl">1.65</span>) <span class="sc">+</span></span>
<span id="cb56-36"><a href="#cb56-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf_text</span>(<span class="at">data =</span> labels_robinson, <span class="fu">aes</span>(<span class="at">label =</span> letter), </span>
<span id="cb56-37"><a href="#cb56-37" aria-hidden="true" tabindex="-1"></a>               <span class="at">color =</span> <span class="st">"darkred"</span>, <span class="at">size =</span> <span class="dv">5</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>, </span>
<span id="cb56-38"><a href="#cb56-38" aria-hidden="true" tabindex="-1"></a>               <span class="at">hjust =</span> <span class="fl">0.5</span>, <span class="at">vjust =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb56-39"><a href="#cb56-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_void</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a>p2 <span class="ot">&lt;-</span> farm_temp_means <span class="sc">%&gt;%</span> </span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y =</span> mean_temp, <span class="at">x =</span> <span class="fu">abs</span>(lat), <span class="at">colour =</span> country)) <span class="sc">+</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Absolute latitude ("</span><span class="sc">*</span>degree<span class="sc">*</span><span class="st">")"</span>), <span class="at">y =</span> <span class="fu">expression</span>(<span class="st">"Mean temperature ("</span><span class="sc">*</span>degree<span class="sc">*</span><span class="st">"C)"</span>)) <span class="sc">+</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"top"</span>, <span class="at">legend.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a>p3 <span class="ot">&lt;-</span> farm_temp_means <span class="sc">%&gt;%</span> </span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">hemisphere =</span> <span class="fu">case_when</span>(lat <span class="sc">&lt;</span> <span class="dv">0</span> <span class="sc">~</span> <span class="st">"S"</span>, T <span class="sc">~</span> <span class="st">"N"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> mean_temp, <span class="at">fill =</span> hemisphere)) <span class="sc">+</span></span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">position =</span> <span class="st">"identity"</span>, <span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.5</span>, <span class="at">binwidth =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="fu">expression</span>(<span class="st">"Mean temperature ("</span><span class="sc">*</span>degree<span class="sc">*</span><span class="st">"C)"</span>), <span class="at">y =</span> <span class="st">"Frequency"</span>) <span class="sc">+</span></span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"top"</span>, <span class="at">legend.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a>(p2 <span class="sc">+</span> p3) <span class="sc">/</span> p1 <span class="sc">+</span> <span class="fu">plot_layout</span>(<span class="at">heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.5</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-farm-mean-temps" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-farm-mean-temps-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-farm-mean-temps-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-1" title="Figure&nbsp;1: Potential options for showing farm site mean temperatures. Top left: Mean temperature by latitude across countries. Top right: Frequency distribution of mean temperatures by hemisphere. Bottom: global distribution of mean temperatures."><img src="manuscript_files/figure-html/fig-farm-mean-temps-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="912"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-farm-mean-temps-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;1: Potential options for showing farm site mean temperatures. Top left: Mean temperature by latitude across countries. Top right: Frequency distribution of mean temperatures by hemisphere. Bottom: global distribution of mean temperatures.
</figcaption>
</figure>
</div>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [8]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>no_margins <span class="ot">&lt;-</span> <span class="cf">function</span>() {</span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"none"</span>, </span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>        <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>),</span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title =</span> <span class="fu">element_blank</span>())</span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>}</span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a>inset_boxes_sm <span class="ot">&lt;-</span> <span class="fu">list</span>(   <span class="co"># lonmin, lonmax, latmin, latmax</span></span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">CAN1 =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">132</span>, <span class="sc">-</span><span class="dv">122</span>, <span class="fl">48.5</span>, <span class="dv">54</span>),</span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>  <span class="at">CAN2 =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">70</span>, <span class="sc">-</span><span class="dv">54</span>, <span class="dv">43</span>, <span class="fl">48.5</span>),</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>  <span class="at">EUR =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">23</span>, <span class="fl">28.5</span>, <span class="dv">52</span>, <span class="dv">71</span>),</span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>  <span class="at">CHI =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">78</span>, <span class="sc">-</span><span class="dv">62</span>, <span class="sc">-</span><span class="fl">55.5</span>, <span class="sc">-</span><span class="dv">27</span>),</span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>  <span class="at">AUS =</span> <span class="fu">c</span>(<span class="dv">144</span>, <span class="dv">149</span>, <span class="sc">-</span><span class="fl">40.75</span>, <span class="sc">-</span><span class="fl">43.75</span>)</span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>worldmap <span class="ot">&lt;-</span> <span class="fu">ne_countries</span>(<span class="at">scale =</span> <span class="st">"large"</span>, <span class="at">returnclass =</span> <span class="st">"sf"</span>)</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a>p1 <span class="ot">&lt;-</span> <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> worldmap, <span class="at">fill =</span> <span class="st">"white"</span>, <span class="at">color =</span> <span class="st">"dimgray"</span>) <span class="sc">+</span></span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_sf</span>(<span class="at">data =</span> farm_temp_means, <span class="fu">aes</span>(<span class="at">color =</span> mean_temp, <span class="at">geometry =</span> geometry), <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb59-19"><a href="#cb59-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>() <span class="sc">+</span></span>
<span id="cb59-20"><a href="#cb59-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_viridis_c</span>(<span class="at">option =</span> <span class="st">"turbo"</span>) <span class="sc">+</span></span>
<span id="cb59-21"><a href="#cb59-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">guides</span>(<span class="at">col =</span> <span class="fu">guide_colourbar</span>(</span>
<span id="cb59-22"><a href="#cb59-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">title =</span> <span class="st">"Mean temperature (°C)"</span>,</span>
<span id="cb59-23"><a href="#cb59-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">direction =</span> <span class="st">"vertical"</span>, <span class="at">position =</span> <span class="st">"right"</span>,</span>
<span id="cb59-24"><a href="#cb59-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">label.position =</span> <span class="st">"right"</span>, <span class="at">title.position =</span> <span class="st">"top"</span>, </span>
<span id="cb59-25"><a href="#cb59-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">title.vjust =</span> <span class="dv">1</span>, <span class="at">title.hjust =</span> <span class="fl">0.5</span>,</span>
<span id="cb59-26"><a href="#cb59-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">frame.colour =</span> <span class="st">"black"</span>, <span class="at">ticks.colour =</span> <span class="st">"black"</span>,<span class="at">ticks.linewidth =</span> <span class="dv">2</span>,</span>
<span id="cb59-27"><a href="#cb59-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">barwidth =</span> <span class="fl">1.5</span>, <span class="at">barheight =</span> <span class="dv">20</span></span>
<span id="cb59-28"><a href="#cb59-28" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">+</span></span>
<span id="cb59-29"><a href="#cb59-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb59-30"><a href="#cb59-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Latitude"</span>, <span class="at">x =</span> <span class="st">"Longitude"</span>)</span>
<span id="cb59-31"><a href="#cb59-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-32"><a href="#cb59-32" aria-hidden="true" tabindex="-1"></a>p_Aus <span class="ot">&lt;-</span> p1 <span class="sc">+</span> </span>
<span id="cb59-33"><a href="#cb59-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">xlim =</span> inset_boxes_sm[[<span class="st">"AUS"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="at">ylim =</span> inset_boxes_sm[[<span class="st">"AUS"</span>]][<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>])  <span class="sc">+</span></span>
<span id="cb59-34"><a href="#cb59-34" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw_label</span>(<span class="st">"E"</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>, </span>
<span id="cb59-35"><a href="#cb59-35" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> inset_boxes_sm[[<span class="st">"AUS"</span>]][<span class="dv">1</span>], <span class="at">y =</span> inset_boxes_sm[[<span class="st">"AUS"</span>]][<span class="dv">3</span>], </span>
<span id="cb59-36"><a href="#cb59-36" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a>p_Eur <span class="ot">&lt;-</span> p1 <span class="sc">+</span> </span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">xlim =</span> inset_boxes_sm[[<span class="st">"EUR"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="at">ylim =</span> inset_boxes_sm[[<span class="st">"EUR"</span>]][<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>]) <span class="sc">+</span></span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw_label</span>(<span class="st">"C"</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>, </span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> inset_boxes_sm[[<span class="st">"EUR"</span>]][<span class="dv">1</span>], <span class="at">y =</span> inset_boxes_sm[[<span class="st">"EUR"</span>]][<span class="dv">4</span>], </span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a>p_Chi <span class="ot">&lt;-</span> p1 <span class="sc">+</span> </span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">xlim =</span> inset_boxes_sm[[<span class="st">"CHI"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="at">ylim =</span> inset_boxes_sm[[<span class="st">"CHI"</span>]][<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>]) <span class="sc">+</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"right"</span>) <span class="sc">+</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw_label</span>(<span class="st">"D"</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>, </span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> inset_boxes_sm[[<span class="st">"CHI"</span>]][<span class="dv">1</span>], <span class="at">y =</span> inset_boxes_sm[[<span class="st">"CHI"</span>]][<span class="dv">4</span>], </span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a>p_Can1 <span class="ot">&lt;-</span> p1 <span class="sc">+</span> </span>
<span id="cb65-2"><a href="#cb65-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">xlim =</span> inset_boxes_sm[[<span class="st">"CAN1"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="at">ylim =</span> inset_boxes_sm[[<span class="st">"CAN1"</span>]][<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>]) <span class="sc">+</span></span>
<span id="cb65-3"><a href="#cb65-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw_label</span>(<span class="st">"A"</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>, </span>
<span id="cb65-4"><a href="#cb65-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> inset_boxes_sm[[<span class="st">"CAN1"</span>]][<span class="dv">1</span>], <span class="at">y =</span> inset_boxes_sm[[<span class="st">"CAN1"</span>]][<span class="dv">4</span>], </span>
<span id="cb65-5"><a href="#cb65-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="sc">-</span><span class="dv">1</span>, <span class="at">vjust =</span> <span class="dv">0</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb67"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb67-1"><a href="#cb67-1" aria-hidden="true" tabindex="-1"></a>p_Can2 <span class="ot">&lt;-</span> p1 <span class="sc">+</span> </span>
<span id="cb67-2"><a href="#cb67-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_sf</span>(<span class="at">xlim =</span> inset_boxes_sm[[<span class="st">"CAN2"</span>]][<span class="dv">1</span><span class="sc">:</span><span class="dv">2</span>], <span class="at">ylim =</span> inset_boxes_sm[[<span class="st">"CAN2"</span>]][<span class="dv">3</span><span class="sc">:</span><span class="dv">4</span>]) <span class="sc">+</span></span>
<span id="cb67-3"><a href="#cb67-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">draw_label</span>(<span class="st">"B"</span>, <span class="at">size =</span> <span class="dv">14</span>, <span class="at">fontface =</span> <span class="st">"bold"</span>, </span>
<span id="cb67-4"><a href="#cb67-4" aria-hidden="true" tabindex="-1"></a>             <span class="at">x =</span> inset_boxes_sm[[<span class="st">"CAN2"</span>]][<span class="dv">1</span>], <span class="at">y =</span> inset_boxes_sm[[<span class="st">"CAN2"</span>]][<span class="dv">4</span>], </span>
<span id="cb67-5"><a href="#cb67-5" aria-hidden="true" tabindex="-1"></a>             <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">vjust =</span> <span class="dv">1</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Coordinate system already present. Adding new coordinate system, which will
replace the existing one.</code></pre>
</div>
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a>grid_1 <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p_Can1, p_Can2, <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">rel_widths =</span>  <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.35</span>))</span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>grid_2 <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(p_Eur,  p_Aus,  <span class="at">nrow =</span> <span class="dv">2</span>, <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>grid_3 <span class="ot">&lt;-</span> cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(grid_2, p_Chi,  <span class="at">ncol =</span> <span class="dv">2</span>, <span class="at">rel_widths =</span>  <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">1</span>))</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a>cowplot<span class="sc">::</span><span class="fu">plot_grid</span>(</span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>  grid_1, grid_3,</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">nrow =</span> <span class="dv">2</span>,</span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">rel_heights =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">1.5</span>)        <span class="co"># relative row heights</span></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p class=""><a href="manuscript_files/figure-html/patchwork-map-mercator-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-2" title="Possibly use a patchwork map to show spatial results?"><img src="manuscript_files/figure-html/patchwork-map-mercator-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="912" alt="Possibly use a patchwork map to show spatial results?"></a></p>
</figure>
</div>
<figcaption>Possibly use a patchwork map to show spatial results?</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
<section id="production-parameters" class="level2">
<h2 class="anchored" data-anchor-id="production-parameters">Production parameters</h2>
<p>Salmon were grown for 548 days and were initialised Spring. This was on day 274 for farms in the southern hemisphere and day 121 for farms in the northern hemisphere. Cohorts were initialised anually, such that there was an overlap of 183 days where one cohort was nearing harvest and another was being initialised.</p>
</section>
<section id="model-parameterisation" class="level2">
<h2 class="anchored" data-anchor-id="model-parameterisation">Model parameterisation</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [9]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Get parameter and sensitivity data</summary>
<div class="sourceCode cell-code" id="cb70"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb70-1"><a href="#cb70-1" aria-hidden="true" tabindex="-1"></a>params <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_species_data_path, <span class="st">"sens_params.qs"</span>) <span class="sc">%&gt;%</span> qs<span class="sc">::</span><span class="fu">qread</span>()</span>
<span id="cb70-2"><a href="#cb70-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb70-3"><a href="#cb70-3" aria-hidden="true" tabindex="-1"></a>param_names <span class="ot">&lt;-</span> tibble<span class="sc">::</span><span class="fu">tribble</span>(</span>
<span id="cb70-4"><a href="#cb70-4" aria-hidden="true" tabindex="-1"></a>  <span class="sc">~</span>name,        <span class="sc">~</span>lab,</span>
<span id="cb70-5"><a href="#cb70-5" aria-hidden="true" tabindex="-1"></a>  <span class="st">"alpha"</span>,      <span class="fu">bquote</span>(alpha),</span>
<span id="cb70-6"><a href="#cb70-6" aria-hidden="true" tabindex="-1"></a>  <span class="st">"epsprot"</span>,    <span class="fu">bquote</span>(epsilon[<span class="st">"P"</span>]),</span>
<span id="cb70-7"><a href="#cb70-7" aria-hidden="true" tabindex="-1"></a>  <span class="st">"epslip"</span>,     <span class="fu">bquote</span>(epsilon[<span class="st">"L"</span>]),</span>
<span id="cb70-8"><a href="#cb70-8" aria-hidden="true" tabindex="-1"></a>  <span class="st">"epscarb"</span>,    <span class="fu">bquote</span>(epsilon[<span class="st">"C"</span>]),</span>
<span id="cb70-9"><a href="#cb70-9" aria-hidden="true" tabindex="-1"></a>  <span class="st">"epsO2"</span>,      <span class="fu">bquote</span>(epsilon[<span class="st">"O"</span>[<span class="dv">2</span>]]),</span>
<span id="cb70-10"><a href="#cb70-10" aria-hidden="true" tabindex="-1"></a>  <span class="st">"pk"</span>,         <span class="fu">bquote</span>(<span class="st">"pk"</span>),</span>
<span id="cb70-11"><a href="#cb70-11" aria-hidden="true" tabindex="-1"></a>  <span class="st">"k0"</span>,         <span class="fu">bquote</span>(<span class="st">"k"</span>[<span class="dv">0</span>]),</span>
<span id="cb70-12"><a href="#cb70-12" aria-hidden="true" tabindex="-1"></a>  <span class="st">"m"</span>,          <span class="fu">bquote</span>(<span class="st">"m"</span>),</span>
<span id="cb70-13"><a href="#cb70-13" aria-hidden="true" tabindex="-1"></a>  <span class="st">"n"</span>,          <span class="fu">bquote</span>(<span class="st">"n"</span>),</span>
<span id="cb70-14"><a href="#cb70-14" aria-hidden="true" tabindex="-1"></a>  <span class="st">"betac"</span>,      <span class="fu">bquote</span>(beta<span class="sc">*</span><span class="st">"C"</span>),</span>
<span id="cb70-15"><a href="#cb70-15" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Tma"</span>,        <span class="fu">bquote</span>(<span class="st">"T"</span>[<span class="st">"max"</span>]),</span>
<span id="cb70-16"><a href="#cb70-16" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Toa"</span>,        <span class="fu">bquote</span>(<span class="st">"T"</span>[<span class="st">"opt"</span>]),</span>
<span id="cb70-17"><a href="#cb70-17" aria-hidden="true" tabindex="-1"></a>  <span class="st">"Taa"</span>,        <span class="fu">bquote</span>(<span class="st">"T"</span>[<span class="st">"min"</span>]),</span>
<span id="cb70-18"><a href="#cb70-18" aria-hidden="true" tabindex="-1"></a>  <span class="st">"omega"</span>,      <span class="fu">bquote</span>(omega),</span>
<span id="cb70-19"><a href="#cb70-19" aria-hidden="true" tabindex="-1"></a>  <span class="st">"a"</span>,          <span class="fu">bquote</span>(<span class="st">"a"</span>),</span>
<span id="cb70-20"><a href="#cb70-20" aria-hidden="true" tabindex="-1"></a>  <span class="st">"k"</span>,          <span class="fu">bquote</span>(<span class="st">"k"</span>),</span>
<span id="cb70-21"><a href="#cb70-21" aria-hidden="true" tabindex="-1"></a>  <span class="st">"eff"</span>,        <span class="fu">bquote</span>(<span class="st">"eff"</span>),</span>
<span id="cb70-22"><a href="#cb70-22" aria-hidden="true" tabindex="-1"></a>  <span class="st">"meanW"</span>,      <span class="fu">bquote</span>(<span class="fu">bar</span>(<span class="st">"W"</span>)),</span>
<span id="cb70-23"><a href="#cb70-23" aria-hidden="true" tabindex="-1"></a>  <span class="st">"deltaW"</span>,     <span class="fu">bquote</span>(Delta<span class="sc">*</span><span class="st">"W"</span>), </span>
<span id="cb70-24"><a href="#cb70-24" aria-hidden="true" tabindex="-1"></a>  <span class="st">"meanImax"</span>,   <span class="fu">bquote</span>(<span class="fu">bar</span>(<span class="st">"I"</span>[<span class="st">"max"</span>])),</span>
<span id="cb70-25"><a href="#cb70-25" aria-hidden="true" tabindex="-1"></a>  <span class="st">"deltaImax"</span>,  <span class="fu">bquote</span>(Delta<span class="sc">*</span><span class="st">"I"</span>[<span class="st">"max"</span>]),</span>
<span id="cb70-26"><a href="#cb70-26" aria-hidden="true" tabindex="-1"></a>  <span class="st">"overFmean"</span>,  <span class="fu">bquote</span>(<span class="fu">bar</span>(<span class="st">"F"</span>)),</span>
<span id="cb70-27"><a href="#cb70-27" aria-hidden="true" tabindex="-1"></a>  <span class="st">"overFdelta"</span>, <span class="fu">bquote</span>(Delta<span class="sc">*</span><span class="st">"F"</span>),</span>
<span id="cb70-28"><a href="#cb70-28" aria-hidden="true" tabindex="-1"></a>  <span class="st">"mortmyt"</span>,    <span class="fu">bquote</span>(<span class="st">"mort"</span>)</span>
<span id="cb70-29"><a href="#cb70-29" aria-hidden="true" tabindex="-1"></a>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p><a href="#tbl-params" class="quarto-xref">Table&nbsp;1</a> shows the parameter values used to model the growth of farmed Atlantic salmon.</p>
<div id="tbl-params" class="quarto-float quarto-figure quarto-figure-center anchored">
<figure class="quarto-float quarto-float-tbl figure">
<figcaption class="quarto-float-caption-top quarto-float-caption quarto-float-tbl" id="tbl-params-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Table&nbsp;1: Model growth parameters for Atlantic salmon
</figcaption>
<div aria-describedby="tbl-params-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<table class="caption-top table">
<thead>
<tr class="header">
<th style="text-align: center;">Parameter</th>
<th style="text-align: center;">Value</th>
<th style="text-align: center;">Units</th>
<th style="text-align: left;">Description</th>
<th style="text-align: left;">Reference</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(\alpha\)</span></td>
<td style="text-align: center;">0.04</td>
<td style="text-align: center;">-</td>
<td style="text-align: left;">Feeding catabolism coefficient</td>
<td style="text-align: left;"><em>Salvelinus alpinus</em>, <em>Oncorhynchus mykiss</em> (temps: 8–13<span class="math inline">\(^{\circ}\)</span>C) <span class="citation" data-cites="broekhuizen_modelling_1994">(<a href="#ref-broekhuizen_modelling_1994" role="doc-biblioref">Broekhuizen et al. 1994</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="math inline">\(\epsilon_{P}\)</span></td>
<td style="text-align: center;">2.36^{4}</td>
<td style="text-align: center;">J g<span class="math inline">\(^{-1}\)</span></td>
<td style="text-align: left;">Energy content of protein</td>
<td style="text-align: left;"><span class="citation" data-cites="baldan_r_2018">Baldan et al. (<a href="#ref-baldan_r_2018" role="doc-biblioref">2018</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(\epsilon_{L}\)</span></td>
<td style="text-align: center;">3.62^{4}</td>
<td style="text-align: center;">J g<span class="math inline">\(^{-1}\)</span></td>
<td style="text-align: left;">Energy content of lipid</td>
<td style="text-align: left;"><span class="citation" data-cites="baldan_r_2018">Baldan et al. (<a href="#ref-baldan_r_2018" role="doc-biblioref">2018</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="math inline">\(\epsilon_{C}\)</span></td>
<td style="text-align: center;">1.72^{4}</td>
<td style="text-align: center;">J g<span class="math inline">\(^{-1}\)</span></td>
<td style="text-align: left;">Energy content of carbohydrate</td>
<td style="text-align: left;"><span class="citation" data-cites="baldan_r_2018">Baldan et al. (<a href="#ref-baldan_r_2018" role="doc-biblioref">2018</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(\epsilon_{O_2}\)</span></td>
<td style="text-align: center;">1.34^{4}</td>
<td style="text-align: center;">J g O<span class="math inline">\(_2^{-1}\)</span></td>
<td style="text-align: left;">Energy consumed by the respiration of 1 g of oxygen</td>
<td style="text-align: left;"><span class="citation" data-cites="elliott_energy_1975">Elliott and Davison (<a href="#ref-elliott_energy_1975" role="doc-biblioref">1975</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;">pk</td>
<td style="text-align: center;">0.05</td>
<td style="text-align: center;">d<span class="math inline">\(^{-1}\)</span></td>
<td style="text-align: left;">Temperature coefficient for the fasting catabolism</td>
<td style="text-align: left;"><span class="citation" data-cites="macnaughton_using_2019">Macnaughton et al. (<a href="#ref-macnaughton_using_2019" role="doc-biblioref">2019</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(k_0\)</span></td>
<td style="text-align: center;">0.003</td>
<td style="text-align: center;">-</td>
<td style="text-align: left;">Fasting catabolism at 0 <span class="math inline">\(^{^\circ}\)</span>C</td>
<td style="text-align: left;"><span class="citation" data-cites="macnaughton_using_2019">Macnaughton et al. (<a href="#ref-macnaughton_using_2019" role="doc-biblioref">2019</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;">m</td>
<td style="text-align: center;">0.75</td>
<td style="text-align: center;">-</td>
<td style="text-align: left;">Weight exponent for anabolism</td>
<td style="text-align: left;">Experimentally-derived alue over a range of <em>Oncorhynchus</em>, <em>Perca</em> and <em>Salmo</em> species (temps: 8–13<span class="math inline">\(^{\circ}\)</span>C) <span class="citation" data-cites="broekhuizen_modelling_1994">(<a href="#ref-broekhuizen_modelling_1994" role="doc-biblioref">Broekhuizen et al. 1994</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: center;">n</td>
<td style="text-align: center;">1</td>
<td style="text-align: center;">-</td>
<td style="text-align: left;">Weight exponent for catabolism</td>
<td style="text-align: left;"><span class="citation" data-cites="baldan_r_2018">Baldan et al. (<a href="#ref-baldan_r_2018" role="doc-biblioref">2018</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="math inline">\(\beta C\)</span></td>
<td style="text-align: center;">0.3234</td>
<td style="text-align: center;">-</td>
<td style="text-align: left;">Shape coefficient for the feeding function</td>
<td style="text-align: left;">Fish size range: 2.25 <span class="math inline">\(\pm\)</span> 0.49 – 11.5 <span class="math inline">\(\pm\)</span> 5.29 g, temperature ranges: 5–24<span class="math inline">\(\^{\circ}\)</span>C. Coefficient for mean daily energy intake of the moderately fast growing group, 0.4398 (mean fast group), 0.2214 (mean slow group) <span class="citation" data-cites="jonsson_thermal_2001">(<a href="#ref-jonsson_thermal_2001" role="doc-biblioref">B. Jonsson et al. 2001</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(T_{max}\)</span></td>
<td style="text-align: center;">26</td>
<td style="text-align: center;"><span class="math inline">\(^{\circ}\)</span>C</td>
<td style="text-align: left;">Maximum lethal temperature</td>
<td style="text-align: left;">Juvenile <em>Oncorhynchus tshawytscha</em> (19.8 <span class="math inline">\(\pm\)</span> 0.02 g, acclimation temp: 15<span class="math inline">\(^{^\circ}\)</span>C, 19<span class="math inline">\({\circ}\)</span>C) <span class="citation" data-cites="poletto_unusual_2017">(<a href="#ref-poletto_unusual_2017" role="doc-biblioref">Poletto et al. 2017</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="math inline">\(T_{opt}\)</span></td>
<td style="text-align: center;">16</td>
<td style="text-align: center;"><span class="math inline">\(^{\circ}\)</span>C</td>
<td style="text-align: left;">Optimal temperature</td>
<td style="text-align: left;">Fish size range: 2.25 <span class="math inline">\(\pm\)</span> 0.49 – 11.5 <span class="math inline">\(\pm\)</span> 5.29 g, temperature ranges: 5–24<span class="math inline">\(^{^\circ}\)</span>C <span class="citation" data-cites="jonsson_thermal_2001">(<a href="#ref-jonsson_thermal_2001" role="doc-biblioref">B. Jonsson et al. 2001</a>)</span>. Alt value: 14°C <span class="citation" data-cites="handeland_effect_2008">(<a href="#ref-handeland_effect_2008" role="doc-biblioref">Handeland, Imsland, and Stefansson 2008</a>)</span>, <em>Salmo salar</em> initial size: 77.0 <span class="math inline">\(\pm\)</span> 14.6 g, temps: 6, 10, 14, 18<span class="math inline">\({\circ}\)</span>C</td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(T_{min}\)</span></td>
<td style="text-align: center;">2</td>
<td style="text-align: center;"><span class="math inline">\(^{\circ}\)</span>C</td>
<td style="text-align: left;">Lowest feeding temperature</td>
<td style="text-align: left;">Fish essentially stop feeding when below 2<span class="math inline">\(^{\circ}\)</span>C <span class="citation" data-cites="vadboncoeur_lowering_2023">(<a href="#ref-vadboncoeur_lowering_2023" role="doc-biblioref">Vadboncoeur et al. 2023</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="math inline">\(\omega\)</span></td>
<td style="text-align: center;">0.6542</td>
<td style="text-align: center;">g O<span class="math inline">\(_2\)</span> g<span class="math inline">\(^{-1}\)</span></td>
<td style="text-align: left;">Oxygen consumption:weight loss ratio</td>
<td style="text-align: left;">Calibrated value from <span class="citation" data-cites="baldan_r_2018">Baldan et al. (<a href="#ref-baldan_r_2018" role="doc-biblioref">2018</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: center;">a</td>
<td style="text-align: center;">2669</td>
<td style="text-align: center;">J g tissue<span class="math inline">\(^{-1}\)</span></td>
<td style="text-align: left;">Energy content of fish tissue</td>
<td style="text-align: left;">Alt value: 17.02350 (nls model fitted to female Atlantic salmon data extracted and transformed from <span class="citation" data-cites="jonsson_energy_2003 van_tien_development_2016">N. Jonsson and Jonsson (<a href="#ref-jonsson_energy_2003" role="doc-biblioref">2003</a>; <a href="#ref-van_tien_development_2016" role="doc-biblioref">Van Tien et al. 2016</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;">k</td>
<td style="text-align: center;">0.1715</td>
<td style="text-align: center;">-</td>
<td style="text-align: left;">Weight exponent for energy content</td>
<td style="text-align: left;">Alt value: 1.65175 (nls model fitted to female Atlantic salmon data extracted and transformed from <span class="citation" data-cites="jonsson_energy_2003 van_tien_development_2016">N. Jonsson and Jonsson (<a href="#ref-jonsson_energy_2003" role="doc-biblioref">2003</a>; <a href="#ref-van_tien_development_2016" role="doc-biblioref">Van Tien et al. 2016</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: center;">eff</td>
<td style="text-align: center;">0.97</td>
<td style="text-align: center;">-</td>
<td style="text-align: left;">Food ingestion efficiency</td>
<td style="text-align: left;">Range: 95-97% ingestion efficiency <span class="citation" data-cites="uglem_does_2020">(<a href="#ref-uglem_does_2020" role="doc-biblioref">Uglem et al. 2020</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="math inline">\(\overline{W}\)</span></td>
<td style="text-align: center;">125</td>
<td style="text-align: center;">g</td>
<td style="text-align: left;">Dry weight mean</td>
<td style="text-align: left;">Range of 100-150 given so used a mean of 125 and sd of 10 which over 1000 draws from a normal distribution delivers a minimum and maximum close to these numbers <span class="citation" data-cites="dempsey_estimating_2023">(<a href="#ref-dempsey_estimating_2023" role="doc-biblioref">Dempsey et al. 2023</a>)</span></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(\Delta{W}\)</span></td>
<td style="text-align: center;">10</td>
<td style="text-align: center;">g</td>
<td style="text-align: left;">Dry weight standard deviation</td>
<td style="text-align: left;">Range of 100-150 given so used a mean of 125 and sd of 10 which over 1000 draws from a normal distribution delivers a minimum and maximum close to these numbers <span class="citation" data-cites="dempsey_estimating_2023">(<a href="#ref-dempsey_estimating_2023" role="doc-biblioref">Dempsey et al. 2023</a>)</span></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="math inline">\(\overline{I_{max}}\)</span></td>
<td style="text-align: center;">0.035</td>
<td style="text-align: center;">g g<span class="math inline">\(^{-1}\)</span> fish d<span class="math inline">\(^{-1}\)</span></td>
<td style="text-align: left;">Ingestion rate mean</td>
<td style="text-align: left;"><a href="https://www.fao.org/fileadmin/user_upload/affris/docs/tabl10.pdf">FAO</a></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(\Delta{I_{max}}\)</span></td>
<td style="text-align: center;">0.005</td>
<td style="text-align: center;">g g<span class="math inline">\(^{-1}\)</span> fish d<span class="math inline">\(^{-1}\)</span></td>
<td style="text-align: left;">Ingestion rate standard deviation</td>
<td style="text-align: left;"><a href="https://www.fao.org/fileadmin/user_upload/affris/docs/tabl10.pdf">FAO</a></td>
</tr>
<tr class="even">
<td style="text-align: center;"><span class="math inline">\(\overline{F}\)</span></td>
<td style="text-align: center;">0.015</td>
<td style="text-align: center;">g g<span class="math inline">\(^{-1}\)</span> fish</td>
<td style="text-align: left;">Overfeeding rate mean</td>
<td style="text-align: left;"></td>
</tr>
<tr class="odd">
<td style="text-align: center;"><span class="math inline">\(\Delta{F}\)</span></td>
<td style="text-align: center;">0.0045</td>
<td style="text-align: center;">g g<span class="math inline">\(^{-1}\)</span> fish</td>
<td style="text-align: left;">Overfeeding rate standard deviation</td>
<td style="text-align: left;"></td>
</tr>
<tr class="even">
<td style="text-align: center;">mort</td>
<td style="text-align: center;">5.99798^{-4}</td>
<td style="text-align: center;">d<span class="math inline">\(^{-1}\)</span></td>
<td style="text-align: left;">Natural mortality rate</td>
<td style="text-align: left;"><span class="citation" data-cites="tvete_towards_2023">Tvete, Aldrin, and Jensen (<a href="#ref-tvete_towards_2023" role="doc-biblioref">2023</a>)</span></td>
</tr>
</tbody>
</table>
</div>
</figure>
</div>
<section id="parameter-sensitivity" class="level3">
<h3 class="anchored" data-anchor-id="parameter-sensitivity">Parameter sensitivity</h3>
<p>Parameter sensitivity was tested by varying each parameter in <a href="#tbl-params" class="quarto-xref">Table&nbsp;1</a> by <span class="math inline">\(\pm 10\)</span>% and calculating sensitivity as <span class="math inline">\(S_p = \frac{(P_{1.1} - P_{0.9})}{0.2 \times P_1}\)</span>, where <span class="math inline">\(S\)</span> is the sensitivity of the measured variable to changes in parameter <span class="math inline">\(p\)</span>, and <span class="math inline">\(P_{1.1}\)</span>, <span class="math inline">\(P_{0.9}\)</span>, and <span class="math inline">\(P_{1}\)</span> are the value of the measured variable when the model is run with parameter <span class="math inline">\(p\)</span> at 110%, 90%, and 100% of its value respectively. All parameters in Table <a href="#tbl-params" class="quarto-xref">Table&nbsp;1</a> were tested. The population-level parameters (<span class="math inline">\(\overline{W}\)</span>, <span class="math inline">\(\Delta{W}\)</span>, <span class="math inline">\(\overline{I_{max}}\)</span>, <span class="math inline">\(\Delta{I_{max}}\)</span>, <span class="math inline">\(\overline{F}\)</span>, <span class="math inline">\(\Delta{F}\)</span>) were tested with a population of of 5000 on a random 10% of farms, while all other parameters were tested across all farms with a population of 10.</p>
</section>
</section>
<section id="sec-feeds" class="level2">
<h2 class="anchored" data-anchor-id="sec-feeds">Modelled experimental feeds</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [10]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb71"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb71-1"><a href="#cb71-1" aria-hidden="true" tabindex="-1"></a>feed_params <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_species_data_path, <span class="st">"feed_params.qs"</span>) <span class="sc">%&gt;%</span> qs<span class="sc">::</span><span class="fu">qread</span>()</span>
<span id="cb71-2"><a href="#cb71-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-3"><a href="#cb71-3" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> feed_params[[<span class="st">"reference"</span>]]</span>
<span id="cb71-4"><a href="#cb71-4" aria-hidden="true" tabindex="-1"></a>contribP_1 <span class="ot">&lt;-</span> df[[<span class="st">"Proteins"</span>]]<span class="sc">$</span>proportion<span class="sc">*</span>df[[<span class="st">"Proteins"</span>]]<span class="sc">$</span>macro</span>
<span id="cb71-5"><a href="#cb71-5" aria-hidden="true" tabindex="-1"></a>tP_1 <span class="ot">&lt;-</span> <span class="fu">sum</span>(contribP_1)</span>
<span id="cb71-6"><a href="#cb71-6" aria-hidden="true" tabindex="-1"></a>digestP_1 <span class="ot">&lt;-</span> <span class="fu">sum</span>(df[[<span class="st">"Proteins"</span>]]<span class="sc">$</span>digest <span class="sc">*</span> contribP_1)<span class="sc">/</span>tP_1</span>
<span id="cb71-7"><a href="#cb71-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-8"><a href="#cb71-8" aria-hidden="true" tabindex="-1"></a>contribL_1 <span class="ot">&lt;-</span> df[[<span class="st">"Lipids"</span>]]<span class="sc">$</span>proportion<span class="sc">*</span>df[[<span class="st">"Lipids"</span>]]<span class="sc">$</span>macro</span>
<span id="cb71-9"><a href="#cb71-9" aria-hidden="true" tabindex="-1"></a>tL_1 <span class="ot">&lt;-</span> <span class="fu">sum</span>(contribL_1)</span>
<span id="cb71-10"><a href="#cb71-10" aria-hidden="true" tabindex="-1"></a>digestL_1 <span class="ot">&lt;-</span> <span class="fu">sum</span>(df[[<span class="st">"Lipids"</span>]]<span class="sc">$</span>digest <span class="sc">*</span> contribL_1)<span class="sc">/</span>tL_1</span>
<span id="cb71-11"><a href="#cb71-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-12"><a href="#cb71-12" aria-hidden="true" tabindex="-1"></a>contribC_1 <span class="ot">&lt;-</span> df[[<span class="st">"Carbohydrates"</span>]]<span class="sc">$</span>proportion<span class="sc">*</span>df[[<span class="st">"Carbohydrates"</span>]]<span class="sc">$</span>macro</span>
<span id="cb71-13"><a href="#cb71-13" aria-hidden="true" tabindex="-1"></a>tC_1 <span class="ot">&lt;-</span> <span class="fu">sum</span>(contribC_1)</span>
<span id="cb71-14"><a href="#cb71-14" aria-hidden="true" tabindex="-1"></a>digestC_1 <span class="ot">&lt;-</span> <span class="fu">sum</span>(df[[<span class="st">"Carbohydrates"</span>]]<span class="sc">$</span>digest <span class="sc">*</span> contribC_1)<span class="sc">/</span>tC_1</span>
<span id="cb71-15"><a href="#cb71-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-16"><a href="#cb71-16" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> feed_params[[<span class="st">"past"</span>]]</span>
<span id="cb71-17"><a href="#cb71-17" aria-hidden="true" tabindex="-1"></a>contribP_2 <span class="ot">&lt;-</span> df[[<span class="st">"Proteins"</span>]]<span class="sc">$</span>proportion<span class="sc">*</span>df[[<span class="st">"Proteins"</span>]]<span class="sc">$</span>macro</span>
<span id="cb71-18"><a href="#cb71-18" aria-hidden="true" tabindex="-1"></a>tP_2 <span class="ot">&lt;-</span> <span class="fu">sum</span>(contribP_2)</span>
<span id="cb71-19"><a href="#cb71-19" aria-hidden="true" tabindex="-1"></a>digestP_2 <span class="ot">&lt;-</span> <span class="fu">sum</span>(df[[<span class="st">"Proteins"</span>]]<span class="sc">$</span>digest <span class="sc">*</span> contribP_2)<span class="sc">/</span>tP_2</span>
<span id="cb71-20"><a href="#cb71-20" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-21"><a href="#cb71-21" aria-hidden="true" tabindex="-1"></a>contribL_2 <span class="ot">&lt;-</span> df[[<span class="st">"Lipids"</span>]]<span class="sc">$</span>proportion<span class="sc">*</span>df[[<span class="st">"Lipids"</span>]]<span class="sc">$</span>macro</span>
<span id="cb71-22"><a href="#cb71-22" aria-hidden="true" tabindex="-1"></a>tL_2 <span class="ot">&lt;-</span> <span class="fu">sum</span>(contribL_2)</span>
<span id="cb71-23"><a href="#cb71-23" aria-hidden="true" tabindex="-1"></a>digestL_2 <span class="ot">&lt;-</span> <span class="fu">sum</span>(df[[<span class="st">"Lipids"</span>]]<span class="sc">$</span>digest <span class="sc">*</span> contribL_2)<span class="sc">/</span>tL_2</span>
<span id="cb71-24"><a href="#cb71-24" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-25"><a href="#cb71-25" aria-hidden="true" tabindex="-1"></a>contribC_2 <span class="ot">&lt;-</span> df[[<span class="st">"Carbohydrates"</span>]]<span class="sc">$</span>proportion<span class="sc">*</span>df[[<span class="st">"Carbohydrates"</span>]]<span class="sc">$</span>macro</span>
<span id="cb71-26"><a href="#cb71-26" aria-hidden="true" tabindex="-1"></a>tC_2 <span class="ot">&lt;-</span> <span class="fu">sum</span>(contribC_2)</span>
<span id="cb71-27"><a href="#cb71-27" aria-hidden="true" tabindex="-1"></a>digestC_2 <span class="ot">&lt;-</span> <span class="fu">sum</span>(df[[<span class="st">"Carbohydrates"</span>]]<span class="sc">$</span>digest <span class="sc">*</span> contribC_2)<span class="sc">/</span>tC_2</span>
<span id="cb71-28"><a href="#cb71-28" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-29"><a href="#cb71-29" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> feed_params[[<span class="st">"future"</span>]]</span>
<span id="cb71-30"><a href="#cb71-30" aria-hidden="true" tabindex="-1"></a>contribP_3 <span class="ot">&lt;-</span> df[[<span class="st">"Proteins"</span>]]<span class="sc">$</span>proportion<span class="sc">*</span>df[[<span class="st">"Proteins"</span>]]<span class="sc">$</span>macro</span>
<span id="cb71-31"><a href="#cb71-31" aria-hidden="true" tabindex="-1"></a>tP_3 <span class="ot">&lt;-</span> <span class="fu">sum</span>(contribP_3)</span>
<span id="cb71-32"><a href="#cb71-32" aria-hidden="true" tabindex="-1"></a>digestP_3 <span class="ot">&lt;-</span> <span class="fu">sum</span>(df[[<span class="st">"Proteins"</span>]]<span class="sc">$</span>digest <span class="sc">*</span> contribP_3)<span class="sc">/</span>tP_3</span>
<span id="cb71-33"><a href="#cb71-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-34"><a href="#cb71-34" aria-hidden="true" tabindex="-1"></a>contribL_3 <span class="ot">&lt;-</span> df[[<span class="st">"Lipids"</span>]]<span class="sc">$</span>proportion<span class="sc">*</span>df[[<span class="st">"Lipids"</span>]]<span class="sc">$</span>macro</span>
<span id="cb71-35"><a href="#cb71-35" aria-hidden="true" tabindex="-1"></a>tL_3 <span class="ot">&lt;-</span> <span class="fu">sum</span>(contribL_3)</span>
<span id="cb71-36"><a href="#cb71-36" aria-hidden="true" tabindex="-1"></a>digestL_3 <span class="ot">&lt;-</span> <span class="fu">sum</span>(df[[<span class="st">"Lipids"</span>]]<span class="sc">$</span>digest <span class="sc">*</span> contribL_3)<span class="sc">/</span>tL_3</span>
<span id="cb71-37"><a href="#cb71-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb71-38"><a href="#cb71-38" aria-hidden="true" tabindex="-1"></a>contribC_3 <span class="ot">&lt;-</span> df[[<span class="st">"Carbohydrates"</span>]]<span class="sc">$</span>proportion<span class="sc">*</span>df[[<span class="st">"Carbohydrates"</span>]]<span class="sc">$</span>macro</span>
<span id="cb71-39"><a href="#cb71-39" aria-hidden="true" tabindex="-1"></a>tC_3 <span class="ot">&lt;-</span> <span class="fu">sum</span>(contribC_3)</span>
<span id="cb71-40"><a href="#cb71-40" aria-hidden="true" tabindex="-1"></a>digestC_3 <span class="ot">&lt;-</span> <span class="fu">sum</span>(df[[<span class="st">"Carbohydrates"</span>]]<span class="sc">$</span>digest <span class="sc">*</span> contribC_3)<span class="sc">/</span>tC_3</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class=""><p></p><div class="no-row-height"><span class="margin-aside">Possibly replace this with a better summary of the ingredients that make up each feed, and report these stats in Results under Uneaten feed?</span></div></div>
<ul>
<li>Feeds came from <span class="citation" data-cites="cottrell_origins_nodate cottrell_global_2020">(<a href="#ref-cottrell_origins_nodate" role="doc-biblioref">Cottrell et al., n.d.</a>, <a href="#ref-cottrell_global_2020" role="doc-biblioref">2020</a>)</span>.</li>
<li>The reference feed is 37.8% protein, 36.5% lipid, 20.4% carbohydrate.</li>
<li>The past feed is 49.8% protein, 27.1% lipid, 16.6% carbohydrate.</li>
<li>The future feed is 47.7% protein, 32.9% lipid, 13.9% carbohydrate.</li>
<li>Within the reference feed, protein is overall 89.2% digestible, lipid is overall 96% digestible, and carbohydrate is overall 69.6% digestible.</li>
<li>Within the past feed, protein is overall 90.5% digestible, lipid is overall 96% digestible, and carbohydrate is overall 79.7% digestible.</li>
<li>Within the future feed, protein is overall 86.2% digestible, lipid is overall 96% digestible, and carbohydrate is overall 69.3% digestible.</li>
</ul>
</section>
<section id="model-population-and-biomass" class="level2">
<h2 class="anchored" data-anchor-id="model-population-and-biomass">Model population and biomass</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [11]:</pre></div><div id="cell-fig-cohorts" class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb72"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb72-1"><a href="#cb72-1" aria-hidden="true" tabindex="-1"></a>fnms <span class="ot">&lt;-</span> output_farm_growth_data_path <span class="sc">%&gt;%</span> </span>
<span id="cb72-2"><a href="#cb72-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list.files</span>(<span class="at">full.names =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb72-3"><a href="#cb72-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_subset</span>(<span class="st">"reference"</span>)</span>
<span id="cb72-4"><a href="#cb72-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-5"><a href="#cb72-5" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> qs<span class="sc">::</span><span class="fu">qread</span>(fnms[<span class="dv">1</span>])[[<span class="st">"biomass_stat"</span>]] <span class="sc">%&gt;%</span> </span>
<span id="cb72-6"><a href="#cb72-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb72-7"><a href="#cb72-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mean =</span> <span class="fu">set_units</span>(mean, <span class="st">"g"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"t"</span>),</span>
<span id="cb72-8"><a href="#cb72-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">sd =</span> <span class="fu">set_units</span>(sd, <span class="st">"g"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"t"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb72-9"><a href="#cb72-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>farm_ID)</span>
<span id="cb72-10"><a href="#cb72-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-11"><a href="#cb72-11" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(</span>
<span id="cb72-12"><a href="#cb72-12" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="dv">1</span>),</span>
<span id="cb72-13"><a href="#cb72-13" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">t =</span> t<span class="sc">+</span><span class="dv">365</span>, <span class="at">cohort =</span> <span class="dv">2</span>),</span>
<span id="cb72-14"><a href="#cb72-14" aria-hidden="true" tabindex="-1"></a>  df <span class="sc">%&gt;%</span> <span class="fu">mutate</span>(<span class="at">t =</span> t<span class="sc">+</span><span class="dv">730</span>, <span class="at">cohort =</span> <span class="dv">3</span>)</span>
<span id="cb72-15"><a href="#cb72-15" aria-hidden="true" tabindex="-1"></a>) <span class="sc">%&gt;%</span> </span>
<span id="cb72-16"><a href="#cb72-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">t =</span> (t<span class="sc">-</span><span class="fu">min</span>(t))<span class="sc">+</span><span class="dv">1</span>,</span>
<span id="cb72-17"><a href="#cb72-17" aria-hidden="true" tabindex="-1"></a>         <span class="at">split =</span> <span class="st">"Split by cohort"</span>)</span>
<span id="cb72-18"><a href="#cb72-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-19"><a href="#cb72-19" aria-hidden="true" tabindex="-1"></a>min_t <span class="ot">&lt;-</span> <span class="fu">min</span>(df<span class="sc">$</span>t[df<span class="sc">$</span>cohort <span class="sc">==</span> <span class="dv">2</span>])</span>
<span id="cb72-20"><a href="#cb72-20" aria-hidden="true" tabindex="-1"></a>max_t <span class="ot">&lt;-</span> <span class="fu">max</span>(df<span class="sc">$</span>t[df<span class="sc">$</span>cohort <span class="sc">==</span> <span class="dv">2</span>])</span>
<span id="cb72-21"><a href="#cb72-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-22"><a href="#cb72-22" aria-hidden="true" tabindex="-1"></a>df_2 <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span> </span>
<span id="cb72-23"><a href="#cb72-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(t <span class="sc">%in%</span> min_t<span class="sc">:</span>max_t) <span class="sc">%&gt;%</span> </span>
<span id="cb72-24"><a href="#cb72-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">sd =</span> sd<span class="sc">/</span>mean) <span class="sc">%&gt;%</span> </span>
<span id="cb72-25"><a href="#cb72-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(t) <span class="sc">%&gt;%</span> </span>
<span id="cb72-26"><a href="#cb72-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">sd =</span> <span class="fu">sqrt</span>(<span class="fu">sum</span>(sd<span class="sc">^</span><span class="dv">2</span>)),</span>
<span id="cb72-27"><a href="#cb72-27" aria-hidden="true" tabindex="-1"></a>          <span class="at">mean =</span> <span class="fu">sum</span>(mean),</span>
<span id="cb72-28"><a href="#cb72-28" aria-hidden="true" tabindex="-1"></a>          <span class="at">sd =</span> sd<span class="sc">*</span>mean) <span class="sc">%&gt;%</span> </span>
<span id="cb72-29"><a href="#cb72-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="st">"all"</span>,</span>
<span id="cb72-30"><a href="#cb72-30" aria-hidden="true" tabindex="-1"></a>         <span class="at">split =</span> <span class="st">"Totalled"</span>)</span>
<span id="cb72-31"><a href="#cb72-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-32"><a href="#cb72-32" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> <span class="fu">rbind</span>(df, df_2) <span class="sc">%&gt;%</span> </span>
<span id="cb72-33"><a href="#cb72-33" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">cohort =</span> <span class="fu">as.factor</span>(cohort))</span>
<span id="cb72-34"><a href="#cb72-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb72-35"><a href="#cb72-35" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(df, <span class="fu">aes</span>(<span class="at">x =</span> t, <span class="at">y =</span> mean, <span class="at">ymin =</span> mean<span class="sc">-</span>sd, <span class="at">ymax =</span> mean<span class="sc">+</span>sd, <span class="at">colour =</span> <span class="fu">as.factor</span>(cohort))) <span class="sc">+</span></span>
<span id="cb72-36"><a href="#cb72-36" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="fl">0.75</span>) <span class="sc">+</span></span>
<span id="cb72-37"><a href="#cb72-37" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">alpha =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb72-38"><a href="#cb72-38" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(split)) <span class="sc">+</span></span>
<span id="cb72-39"><a href="#cb72-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">1600</span>, <span class="dv">120</span>)) <span class="sc">+</span></span>
<span id="cb72-40"><a href="#cb72-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_manual</span>(<span class="at">values =</span> cohort_pal) <span class="sc">+</span></span>
<span id="cb72-41"><a href="#cb72-41" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb72-42"><a href="#cb72-42" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Farm biomass"</span>, <span class="at">x =</span> <span class="st">"Day of production"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-cohorts" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-cohorts-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-cohorts-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-3" title="Figure&nbsp;2: Example of how the overlapping cohorts combine into a single farm production curve (in this case biomass). The left plot shows the farm biomass produced by each cohort from initialisation to harvest, while the right plot shows the combined farm biomass over the course of a single production period of 548 days."><img src="manuscript_files/figure-html/fig-cohorts-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="816"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-cohorts-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;2: Example of how the overlapping cohorts combine into a single farm production curve (in this case biomass). The left plot shows the farm biomass produced by each cohort from initialisation to harvest, while the right plot shows the combined farm biomass over the course of a single production period of 548 days.
</figcaption>
</figure>
</div>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [12]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb73"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb73-1"><a href="#cb73-1" aria-hidden="true" tabindex="-1"></a>coho_biom <span class="ot">&lt;-</span> <span class="fu">file.path</span>(data_analysis_path, <span class="st">"biomass_produced_comparison.qs"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb73-2"><a href="#cb73-2" aria-hidden="true" tabindex="-1"></a>  qs<span class="sc">::</span><span class="fu">qread</span>()</span>
<span id="cb73-3"><a href="#cb73-3" aria-hidden="true" tabindex="-1"></a>harv_size <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_farm_data_path, <span class="st">"farm_harvest_size.qs"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb73-4"><a href="#cb73-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">qread</span>()</span>
<span id="cb73-5"><a href="#cb73-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb73-6"><a href="#cb73-6" aria-hidden="true" tabindex="-1"></a>coho_biom_summ <span class="ot">&lt;-</span> coho_biom <span class="sc">%&gt;%</span> </span>
<span id="cb73-7"><a href="#cb73-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">perc_diff =</span> <span class="dv">100</span><span class="sc">*</span>(mean <span class="sc">-</span> tonnes_per_farm)<span class="sc">/</span>tonnes_per_farm)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>Each farm’s final population was calculated by dividing the production tonnage required by the average modelled individual harvest size for each farm (when feed reference feed). A population timeseries was the generated using the standard mortality rate (5.99798^{-4} d<span class="math inline">\(^{-1}\)</span>). Using this method, a farm’s ‘population’ is not a good representation of the actual number of individuals being farmed, but it ensures each farm location is producing the correct tonnage while allowing comparisons between the different feeds.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [13]:</pre></div><div id="cell-fig-harvest-size" class="cell" data-layout-align="center">
<div class="margin-aside">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a>harv_size <span class="sc">%&gt;%</span> </span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">weight =</span> weight <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"kg"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb74-3"><a href="#cb74-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(farm_ID, weight)) <span class="sc">%&gt;%</span> </span>
<span id="cb74-4"><a href="#cb74-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> weight)) <span class="sc">+</span></span>
<span id="cb74-5"><a href="#cb74-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_histogram</span>(<span class="at">binwidth =</span> <span class="fl">0.25</span>, <span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">fill =</span> <span class="st">"salmon"</span>, <span class="at">alpha =</span> <span class="fl">0.75</span>) <span class="sc">+</span></span>
<span id="cb74-6"><a href="#cb74-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">500</span>)) <span class="sc">+</span></span>
<span id="cb74-7"><a href="#cb74-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Frequency"</span>, <span class="at">x =</span> <span class="st">"Harvest weight"</span>) <span class="sc">+</span></span>
<span id="cb74-8"><a href="#cb74-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb74-9"><a href="#cb74-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">aspect.ratio =</span> <span class="fl">0.75</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-harvest-size" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-harvest-size-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-harvest-size-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-4" title="Figure&nbsp;3: Frequency distribution of harvest sizes used to create the population timeseries for each farm, as predicted by the fish-growth model."><img src="manuscript_files/figure-html/fig-harvest-size-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="816"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-harvest-size-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;3: Frequency distribution of harvest sizes used to create the population timeseries for each farm, as predicted by the fish-growth model.
</figcaption>
</figure>
</div>
</div>
</div>
</div></div>
<p>By this method, farm biomass as predicted by the model slightly overestimated farm tonnage. Predicted farm biomass was 2.4-6.5 higher than tonnage produced (<a href="#fig-biomass-produced" class="quarto-xref">Figure&nbsp;6</a>). This variation was not correlated with mean farm temperature or total tonnage produced (<strong>?@fig-biomass-variation</strong>).</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [14]:</pre></div><div id="cell-fig-biomass-variation" class="cell" data-layout-align="center">
<div class="margin-aside">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb75"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb75-1"><a href="#cb75-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(coho_biom_summ<span class="sc">$</span>temp_c, coho_biom_summ<span class="sc">$</span>perc_diff)</span>
<span id="cb75-2"><a href="#cb75-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(coho_biom_summ<span class="sc">$</span>tonnes_per_farm, coho_biom_summ<span class="sc">$</span>perc_diff)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-biomass-variation-1" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center" width="816">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-biomass-variation-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="manuscript_files/figure-html/fig-biomass-variation-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-5" title="Figure&nbsp;4: "><img src="manuscript_files/figure-html/fig-biomass-variation-1.png" id="fig-biomass-variation-1" class="img-fluid quarto-figure quarto-figure-center anchored figure-img" width="816"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-biomass-variation-1-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;4
</figcaption>
</figure>
</div>
</div>
<div class="cell-output-display">
<div id="fig-biomass-variation-2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center" width="816">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-biomass-variation-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
<a href="manuscript_files/figure-html/fig-biomass-variation-2.png" class="lightbox" data-gallery="quarto-lightbox-gallery-6" title="Figure&nbsp;5: "><img src="manuscript_files/figure-html/fig-biomass-variation-2.png" id="fig-biomass-variation-2" class="img-fluid quarto-figure quarto-figure-center anchored figure-img" width="816"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-biomass-variation-2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;5
</figcaption>
</figure>
</div>
</div>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [15]:</pre></div><div id="cell-fig-biomass-produced" class="cell" data-layout-align="center">
<div class="margin-aside">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb76"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb76-1"><a href="#cb76-1" aria-hidden="true" tabindex="-1"></a>coho_biom <span class="sc">%&gt;%</span> </span>
<span id="cb76-2"><a href="#cb76-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> tonnes_per_farm, <span class="at">y =</span> mean, <span class="at">colour =</span> country)) <span class="sc">+</span></span>
<span id="cb76-3"><a href="#cb76-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="fl">2.5</span>) <span class="sc">+</span></span>
<span id="cb76-4"><a href="#cb76-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_abline</span>(<span class="at">slope =</span> <span class="dv">1</span>, <span class="at">intercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb76-5"><a href="#cb76-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1700</span>)) <span class="sc">+</span></span>
<span id="cb76-6"><a href="#cb76-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">limits =</span> <span class="fu">c</span>(<span class="dv">0</span>,<span class="dv">1700</span>)) <span class="sc">+</span></span>
<span id="cb76-7"><a href="#cb76-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_colour_brewer</span>(<span class="at">palette =</span> <span class="st">"Set1"</span>) <span class="sc">+</span></span>
<span id="cb76-8"><a href="#cb76-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Modelled biomass produced (t)"</span>, <span class="at">x =</span> <span class="st">"Observed biomass produced (t)"</span>) <span class="sc">+</span></span>
<span id="cb76-9"><a href="#cb76-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb76-10"><a href="#cb76-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">"top"</span>, <span class="at">legend.title =</span> <span class="fu">element_blank</span>())</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 20 rows containing missing values or values outside the scale range
(`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div id="fig-biomass-produced" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-biomass-produced-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-biomass-produced-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-7" title="Figure&nbsp;6: Modelled vs observed salmon biomass produced"><img src="manuscript_files/figure-html/fig-biomass-produced-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="816"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-biomass-produced-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;6: Modelled vs observed salmon biomass produced
</figcaption>
</figure>
</div>
</div>
</div>
</div></div>
</section>
<section id="impacts-of-nutrient-input" class="level2">
<h2 class="anchored" data-anchor-id="impacts-of-nutrient-input">Impacts of nutrient input</h2>
<p><em>To detail here:</em></p>
<ul>
<li>How to convert feed and uneaten faeces into nutrients - just nitrogen? Carbon too? If just nitrogen, then can probably use the standard N = 6.25 x protein conversion to compare the feeds (lipids and carbohydrates have negligible protein). If doing carbon too, will need to find conversion rates for all of them, possibly splitting carbs/lipids into complex/simple, saturated/unsaturated, etc.</li>
<li>Source of species vulnerability layers</li>
</ul>
</section>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<ul>
<li>Parameter sensitivity results are in supplementary <a href="#sec-parameter-sensitivity" class="quarto-xref">Section&nbsp;4.1</a></li>
</ul>
<p>In general, data outputs comes in the following dimensions:</p>
<ul>
<li>Location -&gt; temperature -&gt; farm_ID</li>
<li>Uneaten feed + excreted faeces = total inputs (g)</li>
<li>Day of production (impacted by location)</li>
<li>Main point is the difference between feeds (kg/t salmon).</li>
<li>Also interesting to see if difference between feeds varies geographically (correlates with latitude, or maybe mean/median/max temperature?)</li>
</ul>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [16]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb78"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb78-1"><a href="#cb78-1" aria-hidden="true" tabindex="-1"></a>farmrun_files <span class="ot">&lt;-</span> <span class="fu">file.path</span>(output_cohort_growth_data_path) <span class="sc">%&gt;%</span> </span>
<span id="cb78-2"><a href="#cb78-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list.files</span>(<span class="at">full.names =</span> T)</span>
<span id="cb78-3"><a href="#cb78-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb78-4"><a href="#cb78-4" aria-hidden="true" tabindex="-1"></a><span class="co"># qs::qread(farmrun_files[1]) %&gt;% names()</span></span>
<span id="cb78-5"><a href="#cb78-5" aria-hidden="true" tabindex="-1"></a>weight <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(farmrun_files, <span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"weight_stat"</span>]]}) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span>
<span id="cb78-6"><a href="#cb78-6" aria-hidden="true" tabindex="-1"></a>biomass <span class="ot">&lt;-</span> purrr<span class="sc">::</span><span class="fu">map</span>(farmrun_files, <span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"biomass_stat"</span>]]}) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<section id="uneaten-feed" class="level2">
<h2 class="anchored" data-anchor-id="uneaten-feed">Uneaten feed</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [17]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Load and prep uneaten feed data</summary>
<div class="sourceCode cell-code" id="cb79"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb79-1"><a href="#cb79-1" aria-hidden="true" tabindex="-1"></a>uneaten_total <span class="ot">&lt;-</span> farmrun_files <span class="sc">%&gt;%</span> </span>
<span id="cb79-2"><a href="#cb79-2" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(<span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"total_uneat_stat"</span>]]}) <span class="sc">%&gt;%</span> </span>
<span id="cb79-3"><a href="#cb79-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb79-4"><a href="#cb79-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(biomass, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"farm_ID"</span>, <span class="st">"feed"</span>, <span class="st">"t"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb79-5"><a href="#cb79-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">uneat_mean =</span> mean.x, <span class="at">biomass_mean =</span> mean.y,</span>
<span id="cb79-6"><a href="#cb79-6" aria-hidden="true" tabindex="-1"></a>         <span class="at">uneat_sd =</span> sd.x, <span class="at">biomass_sd =</span> sd.y) <span class="sc">%&gt;%</span> </span>
<span id="cb79-7"><a href="#cb79-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">uneat_sd =</span> uneat_sd<span class="sc">/</span>uneat_mean,</span>
<span id="cb79-8"><a href="#cb79-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">biomass_sd =</span> biomass_sd<span class="sc">/</span>biomass_mean,</span>
<span id="cb79-9"><a href="#cb79-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">uneat_biom_mean =</span> uneat_mean<span class="sc">/</span>biomass_mean,</span>
<span id="cb79-10"><a href="#cb79-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">uneat_biom_sd =</span> <span class="fu">sqrt</span>(uneat_sd<span class="sc">^</span><span class="dv">2</span><span class="sc">+</span>biomass_sd<span class="sc">^</span><span class="dv">2</span>)<span class="sc">*</span>uneat_biom_mean)</span>
<span id="cb79-11"><a href="#cb79-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb79-12"><a href="#cb79-12" aria-hidden="true" tabindex="-1"></a>uneaten_total_stats <span class="ot">&lt;-</span> uneaten_total <span class="sc">%&gt;%</span> </span>
<span id="cb79-13"><a href="#cb79-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(feed, farm_ID) <span class="sc">%&gt;%</span> </span>
<span id="cb79-14"><a href="#cb79-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">total =</span> <span class="fu">sum</span>(uneat_biom_mean)) <span class="sc">%&gt;%</span> </span>
<span id="cb79-15"><a href="#cb79-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(feed) <span class="sc">%&gt;%</span> </span>
<span id="cb79-16"><a href="#cb79-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">min =</span> <span class="fu">minna</span>(total),</span>
<span id="cb79-17"><a href="#cb79-17" aria-hidden="true" tabindex="-1"></a>          <span class="at">max =</span> <span class="fu">maxna</span>(total),</span>
<span id="cb79-18"><a href="#cb79-18" aria-hidden="true" tabindex="-1"></a>          <span class="at">mean =</span> <span class="fu">meanna</span>(total)) <span class="sc">%&gt;%</span> </span>
<span id="cb79-19"><a href="#cb79-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">min =</span> min <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g g_fish-1"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g kg_fish-1"</span>),</span>
<span id="cb79-20"><a href="#cb79-20" aria-hidden="true" tabindex="-1"></a>         <span class="at">max =</span> max <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g g_fish-1"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g kg_fish-1"</span>),</span>
<span id="cb79-21"><a href="#cb79-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">mean =</span> mean <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g g_fish-1"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g kg_fish-1"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb79-22"><a href="#cb79-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">min =</span> <span class="fu">round</span>(min,<span class="dv">2</span>),</span>
<span id="cb79-23"><a href="#cb79-23" aria-hidden="true" tabindex="-1"></a>         <span class="at">max =</span> <span class="fu">round</span>(max,<span class="dv">2</span>),</span>
<span id="cb79-24"><a href="#cb79-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">mean =</span> <span class="fu">round</span>(mean,<span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<section id="amount" class="level3">
<h3 class="anchored" data-anchor-id="amount">Amount</h3>
<ul>
<li>Total reference uneaten feed across a cohort (548 days) ranged from 48.46 to 48.46 g kg<span class="math inline">\(^{-1}\)</span>, with a mean of 61.71.</li>
<li>There was very little difference between the past and future feeds and the reference feed. Uneaten past feed ranged from 48.72 to 48.72 g kg<span class="math inline">\(^{-1}\)</span> (mean of 62.17) while uneaten future feed ranged from 48.73 to 48.73 g kg<span class="math inline">\(^{-1}\)</span> (mean of 62.05).</li>
<li><a href="#fig-total-uneaten-perday" class="quarto-xref">Figure&nbsp;7</a> shows the range of total uneaten feed per day across countries.</li>
</ul>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [18]:</pre></div><div id="cell-fig-total-uneaten-perday" class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb80"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb80-1"><a href="#cb80-1" aria-hidden="true" tabindex="-1"></a>uneaten_total <span class="sc">%&gt;%</span> </span>
<span id="cb80-2"><a href="#cb80-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(farms_geometry, <span class="at">by =</span> <span class="st">"farm_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb80-3"><a href="#cb80-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">uneat_biom_mean =</span> uneat_biom_mean <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g g_fish-1 d-1"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g kg_fish-1 d-1"</span>),</span>
<span id="cb80-4"><a href="#cb80-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">country =</span> <span class="fu">factor</span>(country, <span class="at">levels =</span> sorted_countries)) <span class="sc">%&gt;%</span> </span>
<span id="cb80-5"><a href="#cb80-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> country, <span class="at">y =</span> uneat_biom_mean, <span class="at">fill =</span> feed)) <span class="sc">+</span></span>
<span id="cb80-6"><a href="#cb80-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb80-7"><a href="#cb80-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb80-8"><a href="#cb80-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> feed_pal) <span class="sc">+</span></span>
<span id="cb80-9"><a href="#cb80-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Uneaten feed"</span>, <span class="at">x =</span> <span class="st">"Country"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-total-uneaten-perday" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center" width="912">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-total-uneaten-perday-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-total-uneaten-perday-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-8" title="Figure&nbsp;7: "><img src="manuscript_files/figure-html/fig-total-uneaten-perday-1.png" id="fig-total-uneaten-perday" class="img-fluid quarto-figure quarto-figure-center anchored figure-img" width="912"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-total-uneaten-perday-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;7
</figcaption>
</figure>
</div>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [19]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb81"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb81-1"><a href="#cb81-1" aria-hidden="true" tabindex="-1"></a>uneaten_P <span class="ot">&lt;-</span> farmrun_files <span class="sc">%&gt;%</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"P_uneat_stat"</span>]]}) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span>
<span id="cb81-2"><a href="#cb81-2" aria-hidden="true" tabindex="-1"></a>uneaten_L <span class="ot">&lt;-</span> farmrun_files <span class="sc">%&gt;%</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"L_uneat_stat"</span>]]}) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span>
<span id="cb81-3"><a href="#cb81-3" aria-hidden="true" tabindex="-1"></a>uneaten_C <span class="ot">&lt;-</span> farmrun_files <span class="sc">%&gt;%</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"C_uneat_stat"</span>]]}) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span>
<span id="cb81-4"><a href="#cb81-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb81-5"><a href="#cb81-5" aria-hidden="true" tabindex="-1"></a>uneaten_all <span class="ot">&lt;-</span> uneaten_total <span class="sc">%&gt;%</span> </span>
<span id="cb81-6"><a href="#cb81-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">contains</span>(<span class="st">"biom"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb81-7"><a href="#cb81-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(uneaten_P, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"farm_ID"</span>, <span class="st">"feed"</span>, <span class="st">"t"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb81-8"><a href="#cb81-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(uneaten_L, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"farm_ID"</span>, <span class="st">"feed"</span>, <span class="st">"t"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb81-9"><a href="#cb81-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(uneaten_C, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"farm_ID"</span>, <span class="st">"feed"</span>, <span class="st">"t"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb81-10"><a href="#cb81-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="fu">contains</span>(<span class="st">"sd"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<ul>
<li>The composition of the uneaten feed is (obviously) the same as the total feed composition (<a href="#sec-feeds" class="quarto-xref">Section&nbsp;1.4</a>).</li>
</ul>
</section>
</section>
<section id="excreted-faeces" class="level2">
<h2 class="anchored" data-anchor-id="excreted-faeces">Excreted faeces</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [20]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Load and prep all excretion data</summary>
<div class="sourceCode cell-code" id="cb82"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb82-1"><a href="#cb82-1" aria-hidden="true" tabindex="-1"></a><span class="co"># The following reads in all the total excretion data and converts the mean and sd values into values/biomass</span></span>
<span id="cb82-2"><a href="#cb82-2" aria-hidden="true" tabindex="-1"></a>excreted_total <span class="ot">&lt;-</span> farmrun_files <span class="sc">%&gt;%</span> </span>
<span id="cb82-3"><a href="#cb82-3" aria-hidden="true" tabindex="-1"></a>  purrr<span class="sc">::</span><span class="fu">map</span>(<span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"total_excr_stat"</span>]]}) <span class="sc">%&gt;%</span> </span>
<span id="cb82-4"><a href="#cb82-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">bind_rows</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb82-5"><a href="#cb82-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(biomass, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"farm_ID"</span>, <span class="st">"feed"</span>, <span class="st">"t"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb82-6"><a href="#cb82-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">excr_mean =</span> mean.x, <span class="at">biomass_mean =</span> mean.y,</span>
<span id="cb82-7"><a href="#cb82-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">excr_sd =</span> sd.x, <span class="at">biomass_sd =</span> sd.y) <span class="sc">%&gt;%</span> </span>
<span id="cb82-8"><a href="#cb82-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">excr_sd =</span> excr_sd<span class="sc">/</span>excr_mean,</span>
<span id="cb82-9"><a href="#cb82-9" aria-hidden="true" tabindex="-1"></a>         <span class="at">biomass_sd =</span> biomass_sd<span class="sc">/</span>biomass_mean,</span>
<span id="cb82-10"><a href="#cb82-10" aria-hidden="true" tabindex="-1"></a>         <span class="at">excr_biom_mean =</span> excr_mean<span class="sc">/</span>biomass_mean,</span>
<span id="cb82-11"><a href="#cb82-11" aria-hidden="true" tabindex="-1"></a>         <span class="at">excr_biom_sd =</span> <span class="fu">sqrt</span>(excr_sd<span class="sc">^</span><span class="dv">2</span><span class="sc">+</span>biomass_sd<span class="sc">^</span><span class="dv">2</span>)<span class="sc">*</span>excr_biom_mean)</span>
<span id="cb82-12"><a href="#cb82-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb82-13"><a href="#cb82-13" aria-hidden="true" tabindex="-1"></a>excreted_total_stats <span class="ot">&lt;-</span> excreted_total <span class="sc">%&gt;%</span> </span>
<span id="cb82-14"><a href="#cb82-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(feed, farm_ID) <span class="sc">%&gt;%</span> </span>
<span id="cb82-15"><a href="#cb82-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">total =</span> <span class="fu">sum</span>(excr_biom_mean)) <span class="sc">%&gt;%</span> </span>
<span id="cb82-16"><a href="#cb82-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(feed) <span class="sc">%&gt;%</span> </span>
<span id="cb82-17"><a href="#cb82-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">min =</span> <span class="fu">minna</span>(total),</span>
<span id="cb82-18"><a href="#cb82-18" aria-hidden="true" tabindex="-1"></a>          <span class="at">max =</span> <span class="fu">maxna</span>(total),</span>
<span id="cb82-19"><a href="#cb82-19" aria-hidden="true" tabindex="-1"></a>          <span class="at">mean =</span> <span class="fu">meanna</span>(total)) <span class="sc">%&gt;%</span> </span>
<span id="cb82-20"><a href="#cb82-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">min =</span> min <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g g_fish-1"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g kg_fish-1"</span>),</span>
<span id="cb82-21"><a href="#cb82-21" aria-hidden="true" tabindex="-1"></a>         <span class="at">max =</span> max <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g g_fish-1"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g kg_fish-1"</span>),</span>
<span id="cb82-22"><a href="#cb82-22" aria-hidden="true" tabindex="-1"></a>         <span class="at">mean =</span> mean <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g g_fish-1"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g kg_fish-1"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb82-23"><a href="#cb82-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">min =</span> <span class="fu">round</span>(min,<span class="dv">2</span>),</span>
<span id="cb82-24"><a href="#cb82-24" aria-hidden="true" tabindex="-1"></a>         <span class="at">max =</span> <span class="fu">round</span>(max,<span class="dv">2</span>),</span>
<span id="cb82-25"><a href="#cb82-25" aria-hidden="true" tabindex="-1"></a>         <span class="at">mean =</span> <span class="fu">round</span>(mean,<span class="dv">2</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<ul>
<li>Total reference excreted faeces across a cohort (548 days) ranged from 194.24 to 295.9 g kg<span class="math inline">\(^{-1}\)</span>, with a mean of 247.38.</li>
<li>There was very little difference between the past and future feeds and the reference feed. Uneaten past feed ranged from 154.33 to 236.05 g kg<span class="math inline">\(^{-1}\)</span> (mean of 196.94 g kg<span class="math inline">\(^{-1}\)</span>) while uneaten future feed ranged from 202.55 to 308.61 g kg<span class="math inline">\(^{-1}\)</span> (mean of 257.91 g kg<span class="math inline">\(^{-1}\)</span>).</li>
<li><a href="#fig-total-excreted-perday" class="quarto-xref">Figure&nbsp;8</a> shows the range of total excreted faeces per day across countries.</li>
</ul>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [21]:</pre></div><div id="cell-fig-total-excreted-perday" class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb83"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb83-1"><a href="#cb83-1" aria-hidden="true" tabindex="-1"></a>excreted_total <span class="sc">%&gt;%</span> </span>
<span id="cb83-2"><a href="#cb83-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(farms_geometry, <span class="at">by =</span> <span class="st">"farm_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb83-3"><a href="#cb83-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">excr_biom_mean =</span> excr_biom_mean <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g g_fish-1 d-1"</span>) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"g kg_fish-1 d-1"</span>),</span>
<span id="cb83-4"><a href="#cb83-4" aria-hidden="true" tabindex="-1"></a>         <span class="at">country =</span> <span class="fu">factor</span>(country, <span class="at">levels =</span> sorted_countries)) <span class="sc">%&gt;%</span> </span>
<span id="cb83-5"><a href="#cb83-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> country, <span class="at">y =</span> excr_biom_mean, <span class="at">fill =</span> feed)) <span class="sc">+</span></span>
<span id="cb83-6"><a href="#cb83-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_boxplot</span>() <span class="sc">+</span></span>
<span id="cb83-7"><a href="#cb83-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb83-8"><a href="#cb83-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> feed_pal) <span class="sc">+</span></span>
<span id="cb83-9"><a href="#cb83-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Excreted faeces"</span>, <span class="at">x =</span> <span class="st">"Country"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-total-excreted-perday" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center" width="864">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-total-excreted-perday-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-total-excreted-perday-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-9" title="Figure&nbsp;8: "><img src="manuscript_files/figure-html/fig-total-excreted-perday-1.png" id="fig-total-excreted-perday" class="img-fluid quarto-figure quarto-figure-center anchored figure-img" width="864"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig quarto-uncaptioned" id="fig-total-excreted-perday-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;8
</figcaption>
</figure>
</div>
</div>
</div></div>
<section id="amount-1" class="level3">
<h3 class="anchored" data-anchor-id="amount-1">Amount</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [22]:</pre></div><div id="cell-fig-total-excr-diff_perc" class="cell" data-layout-align="center">
<div class="margin-aside">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb84"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb84-1"><a href="#cb84-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> excreted_total <span class="sc">%&gt;%</span> </span>
<span id="cb84-2"><a href="#cb84-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">contains</span>(<span class="fu">c</span>(<span class="st">"sd"</span>, <span class="st">"biom"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb84-3"><a href="#cb84-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"feed"</span>, <span class="at">values_from =</span> <span class="st">"excr_mean"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb84-4"><a href="#cb84-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">past =</span> past<span class="sc">-</span>reference,</span>
<span id="cb84-5"><a href="#cb84-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">future =</span> future<span class="sc">-</span>reference) <span class="sc">%&gt;%</span> </span>
<span id="cb84-6"><a href="#cb84-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"feed"</span>, <span class="at">values_to =</span> <span class="st">"diff"</span>, <span class="at">cols =</span> <span class="fu">c</span>(past, future), <span class="at">names_transform =</span> <span class="fu">list</span>(<span class="at">feed =</span> as.factor)) <span class="sc">%&gt;%</span> </span>
<span id="cb84-7"><a href="#cb84-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">reference =</span> <span class="fu">set_units</span>(reference, <span class="st">"g kg_fish-1 d-1"</span>),</span>
<span id="cb84-8"><a href="#cb84-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">diff_perc =</span> <span class="fu">drop_units</span>(diff<span class="sc">/</span>reference)) <span class="sc">%&gt;%</span> </span>
<span id="cb84-9"><a href="#cb84-9" aria-hidden="true" tabindex="-1"></a>    <span class="fu">merge</span>(farm_coords, <span class="at">by =</span> <span class="st">"farm_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb84-10"><a href="#cb84-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">prod_day =</span> t <span class="sc">-</span> t_start <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb84-11"><a href="#cb84-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-12"><a href="#cb84-12" aria-hidden="true" tabindex="-1"></a>summ <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb84-13"><a href="#cb84-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(hemisphere, feed) <span class="sc">%&gt;%</span> </span>
<span id="cb84-14"><a href="#cb84-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">mean =</span> <span class="dv">100</span><span class="sc">*</span><span class="fu">meanna</span>(diff_perc) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>),</span>
<span id="cb84-15"><a href="#cb84-15" aria-hidden="true" tabindex="-1"></a>          <span class="at">min =</span> <span class="dv">100</span><span class="sc">*</span><span class="fu">meanna</span>(diff_perc) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>),</span>
<span id="cb84-16"><a href="#cb84-16" aria-hidden="true" tabindex="-1"></a>          <span class="at">max =</span> <span class="dv">100</span><span class="sc">*</span><span class="fu">maxna</span>(diff_perc) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>))</span>
<span id="cb84-17"><a href="#cb84-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb84-18"><a href="#cb84-18" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb84-19"><a href="#cb84-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> reference, <span class="at">y =</span> diff_perc, <span class="at">colour =</span> hemisphere)) <span class="sc">+</span></span>
<span id="cb84-20"><a href="#cb84-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>() <span class="sc">+</span> </span>
<span id="cb84-21"><a href="#cb84-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb84-22"><a href="#cb84-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>feed) <span class="sc">+</span></span>
<span id="cb84-23"><a href="#cb84-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-total-excr-diff_perc" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-total-excr-diff_perc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-total-excr-diff_perc-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-10" title="Figure&nbsp;9: Mean % difference in excreted faeces of the two experimental feeds (compared to the reference feed) with increasing overall excretion."><img src="manuscript_files/figure-html/fig-total-excr-diff_perc-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="816"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-total-excr-diff_perc-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;9: Mean % difference in excreted faeces of the two experimental feeds (compared to the reference feed) with increasing overall excretion.
</figcaption>
</figure>
</div>
</div>
</div>
</div></div>
<p>The difference in total excretion for the past feed ranged from -26, -26.1–23.9, -23.9% across all farms globally, and for the future feed ranged from 1.3, 1.3-4.1, 3.7%.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [23]:</pre></div><div id="cell-fig-total-excr-diff_perc_2" class="cell" data-layout-align="center">
<div class="margin-aside">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb85"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb85-1"><a href="#cb85-1" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb85-2"><a href="#cb85-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(prod_day, feed, hemisphere) <span class="sc">%&gt;%</span> </span>
<span id="cb85-3"><a href="#cb85-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">diff_perc_mean =</span> <span class="fu">meanna</span>(diff_perc),</span>
<span id="cb85-4"><a href="#cb85-4" aria-hidden="true" tabindex="-1"></a>          <span class="at">diff_perc_sd =</span> <span class="fu">sdna</span>(diff_perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb85-5"><a href="#cb85-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> prod_day, <span class="at">y =</span> diff_perc_mean, <span class="at">ymin =</span> diff_perc_mean<span class="sc">-</span>diff_perc_sd, <span class="at">ymax =</span> diff_perc_mean<span class="sc">+</span>diff_perc_sd, </span>
<span id="cb85-6"><a href="#cb85-6" aria-hidden="true" tabindex="-1"></a>             <span class="at">colour =</span> <span class="fu">as.factor</span>(hemisphere), <span class="at">fill =</span> <span class="fu">as.factor</span>(hemisphere))) <span class="sc">+</span></span>
<span id="cb85-7"><a href="#cb85-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>() <span class="sc">+</span> </span>
<span id="cb85-8"><a href="#cb85-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="at">alpha =</span> <span class="fl">0.25</span>) <span class="sc">+</span></span>
<span id="cb85-9"><a href="#cb85-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="dv">0</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb85-10"><a href="#cb85-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="sc">~</span>feed) <span class="sc">+</span></span>
<span id="cb85-11"><a href="#cb85-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-total-excr-diff_perc_2" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-total-excr-diff_perc_2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-total-excr-diff_perc_2-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-11" title="Figure&nbsp;10: Mean % difference in excreted faeces of the two experimental feeds (compared to the reference feed) over time in one production cycle."><img src="manuscript_files/figure-html/fig-total-excr-diff_perc_2-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="816"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-total-excr-diff_perc_2-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;10: Mean % difference in excreted faeces of the two experimental feeds (compared to the reference feed) over time in one production cycle.
</figcaption>
</figure>
</div>
</div>
</div>
</div></div>
<p>However, the actual difference in excreted faeces varied by feed, time in the production cycle, and location of the farm <a href="#fig-total-excr-act_diff" class="quarto-xref">Figure&nbsp;11</a>.</p>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [24]:</pre></div><div id="cell-fig-total-excr-act_diff" class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb86"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb86-1"><a href="#cb86-1" aria-hidden="true" tabindex="-1"></a>df <span class="ot">&lt;-</span> excreted_total <span class="sc">%&gt;%</span> </span>
<span id="cb86-2"><a href="#cb86-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">contains</span>(<span class="fu">c</span>(<span class="st">"sd"</span>, <span class="st">"excr_mean"</span>, <span class="st">"biomass"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb86-3"><a href="#cb86-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(<span class="at">names_from =</span> <span class="st">"feed"</span>, <span class="at">values_from =</span> <span class="st">"excr_biom_mean"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb86-4"><a href="#cb86-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">past =</span> past<span class="sc">-</span>reference,</span>
<span id="cb86-5"><a href="#cb86-5" aria-hidden="true" tabindex="-1"></a>         <span class="at">future =</span> future<span class="sc">-</span>reference) <span class="sc">%&gt;%</span> </span>
<span id="cb86-6"><a href="#cb86-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"feed"</span>, <span class="at">values_to =</span> <span class="st">"diff"</span>, <span class="at">cols =</span> <span class="fu">c</span>(past, future), <span class="at">names_transform =</span> <span class="fu">list</span>(<span class="at">feed =</span> as.factor)) <span class="sc">%&gt;%</span> </span>
<span id="cb86-7"><a href="#cb86-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">reference =</span> <span class="fu">set_units</span>(reference, <span class="st">"g kg_fish-1 d-1"</span>),</span>
<span id="cb86-8"><a href="#cb86-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">diff =</span> <span class="fu">set_units</span>(diff, <span class="st">"kg t_fish-1 d-1"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb86-9"><a href="#cb86-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(farm_coords, <span class="at">by =</span> <span class="st">"farm_ID"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb86-10"><a href="#cb86-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">prod_day =</span> t<span class="sc">-</span>t_start<span class="sc">+</span><span class="dv">1</span>)</span>
<span id="cb86-11"><a href="#cb86-11" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-12"><a href="#cb86-12" aria-hidden="true" tabindex="-1"></a>summ <span class="ot">&lt;-</span> df <span class="sc">%&gt;%</span></span>
<span id="cb86-13"><a href="#cb86-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(hemisphere, feed) <span class="sc">%&gt;%</span> </span>
<span id="cb86-14"><a href="#cb86-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">mean =</span> <span class="fu">meanna</span>(diff) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>),</span>
<span id="cb86-15"><a href="#cb86-15" aria-hidden="true" tabindex="-1"></a>          <span class="at">min =</span> <span class="fu">meanna</span>(diff) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>),</span>
<span id="cb86-16"><a href="#cb86-16" aria-hidden="true" tabindex="-1"></a>          <span class="at">max =</span> <span class="fu">maxna</span>(diff) <span class="sc">%&gt;%</span> <span class="fu">round</span>(<span class="dv">3</span>))</span>
<span id="cb86-17"><a href="#cb86-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb86-18"><a href="#cb86-18" aria-hidden="true" tabindex="-1"></a>df <span class="sc">%&gt;%</span> </span>
<span id="cb86-19"><a href="#cb86-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(hemisphere, feed, prod_day) <span class="sc">%&gt;%</span> </span>
<span id="cb86-20"><a href="#cb86-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">mean_diff =</span> <span class="fu">meanna</span>(diff),</span>
<span id="cb86-21"><a href="#cb86-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">sd_diff =</span> <span class="fu">sdna</span>(diff) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"kg t_fish-1 d-1"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb86-22"><a href="#cb86-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> prod_day, <span class="at">y =</span> mean_diff, <span class="at">ymin =</span> mean_diff<span class="sc">-</span>sd_diff, <span class="at">ymax =</span> mean_diff<span class="sc">+</span>sd_diff, <span class="at">colour =</span> hemisphere, <span class="at">fill =</span> hemisphere)) <span class="sc">+</span></span>
<span id="cb86-23"><a href="#cb86-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="at">linewidth =</span> <span class="fl">0.75</span>) <span class="sc">+</span></span>
<span id="cb86-24"><a href="#cb86-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_ribbon</span>(<span class="at">alpha =</span> <span class="fl">0.25</span>) <span class="sc">+</span></span>
<span id="cb86-25"><a href="#cb86-25" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_wrap</span>(<span class="sc">~</span>feed) <span class="sc">+</span></span>
<span id="cb86-26"><a href="#cb86-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-total-excr-act_diff" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-total-excr-act_diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-total-excr-act_diff-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-12" title="Figure&nbsp;11: Mean actual difference in excreted faeces of the two experimental feeds (compared to the reference feed) over time in one production cycle."><img src="manuscript_files/figure-html/fig-total-excr-act_diff-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="600"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-total-excr-act_diff-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;11: Mean actual difference in excreted faeces of the two experimental feeds (compared to the reference feed) over time in one production cycle.
</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
<section id="composition" class="level3">
<h3 class="anchored" data-anchor-id="composition">Composition</h3>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [25]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb87"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb87-1"><a href="#cb87-1" aria-hidden="true" tabindex="-1"></a>excreted_P <span class="ot">&lt;-</span> farmrun_files <span class="sc">%&gt;%</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"P_excr_stat"</span>]]}) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span>
<span id="cb87-2"><a href="#cb87-2" aria-hidden="true" tabindex="-1"></a>excreted_L <span class="ot">&lt;-</span> farmrun_files <span class="sc">%&gt;%</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"L_excr_stat"</span>]]}) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span>
<span id="cb87-3"><a href="#cb87-3" aria-hidden="true" tabindex="-1"></a>excreted_C <span class="ot">&lt;-</span> farmrun_files <span class="sc">%&gt;%</span> purrr<span class="sc">::</span><span class="fu">map</span>(<span class="cf">function</span>(f) {qs<span class="sc">::</span><span class="fu">qread</span>(f)[[<span class="st">"C_excr_stat"</span>]]}) <span class="sc">%&gt;%</span> <span class="fu">bind_rows</span>()</span>
<span id="cb87-4"><a href="#cb87-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-5"><a href="#cb87-5" aria-hidden="true" tabindex="-1"></a>excreted_all <span class="ot">&lt;-</span> excreted_total <span class="sc">%&gt;%</span> </span>
<span id="cb87-6"><a href="#cb87-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">contains</span>(<span class="st">"biom"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-7"><a href="#cb87-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(excreted_P, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"farm_ID"</span>, <span class="st">"feed"</span>, <span class="st">"t"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-8"><a href="#cb87-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(excreted_L, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"farm_ID"</span>, <span class="st">"feed"</span>, <span class="st">"t"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-9"><a href="#cb87-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">merge</span>(excreted_C, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"farm_ID"</span>, <span class="st">"feed"</span>, <span class="st">"t"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-10"><a href="#cb87-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span><span class="fu">c</span>(<span class="fu">contains</span>(<span class="st">"sd"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb87-11"><a href="#cb87-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">protein =</span> mean.x, <span class="at">lipids =</span> mean.y, <span class="at">carbohydrates =</span> mean) <span class="sc">%&gt;%</span> </span>
<span id="cb87-12"><a href="#cb87-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">protein =</span> protein<span class="sc">/</span>excr_mean <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"d-1"</span>),</span>
<span id="cb87-13"><a href="#cb87-13" aria-hidden="true" tabindex="-1"></a>         <span class="at">lipids =</span> lipids<span class="sc">/</span>excr_mean <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"d-1"</span>),</span>
<span id="cb87-14"><a href="#cb87-14" aria-hidden="true" tabindex="-1"></a>         <span class="at">carbohydrates =</span> carbohydrates<span class="sc">/</span>excr_mean <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"d-1"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-15"><a href="#cb87-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="sc">-</span>excr_mean)</span>
<span id="cb87-16"><a href="#cb87-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb87-17"><a href="#cb87-17" aria-hidden="true" tabindex="-1"></a>ec <span class="ot">&lt;-</span> excreted_all <span class="sc">%&gt;%</span> </span>
<span id="cb87-18"><a href="#cb87-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(feed) <span class="sc">%&gt;%</span> </span>
<span id="cb87-19"><a href="#cb87-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">P =</span> <span class="fu">mean</span>(protein),</span>
<span id="cb87-20"><a href="#cb87-20" aria-hidden="true" tabindex="-1"></a>          <span class="at">C =</span> <span class="fu">mean</span>(carbohydrates),</span>
<span id="cb87-21"><a href="#cb87-21" aria-hidden="true" tabindex="-1"></a>          <span class="at">L =</span> <span class="fu">mean</span>(lipids)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-22"><a href="#cb87-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">source =</span> <span class="st">"excreted"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb87-23"><a href="#cb87-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rbind</span>(<span class="fu">data.frame</span>(</span>
<span id="cb87-24"><a href="#cb87-24" aria-hidden="true" tabindex="-1"></a>    <span class="at">feed =</span> <span class="fu">c</span>(<span class="st">"reference"</span>, <span class="st">"past"</span>, <span class="st">"future"</span>),</span>
<span id="cb87-25"><a href="#cb87-25" aria-hidden="true" tabindex="-1"></a>    <span class="at">P =</span> <span class="fu">c</span>(tP_1, tP_2, tP_3),</span>
<span id="cb87-26"><a href="#cb87-26" aria-hidden="true" tabindex="-1"></a>    <span class="at">C =</span> <span class="fu">c</span>(tC_1, tC_2, tC_3),</span>
<span id="cb87-27"><a href="#cb87-27" aria-hidden="true" tabindex="-1"></a>    <span class="at">L =</span> <span class="fu">c</span>(tL_1, tL_2, tL_3),</span>
<span id="cb87-28"><a href="#cb87-28" aria-hidden="true" tabindex="-1"></a>    <span class="at">source =</span> <span class="fu">rep</span>(<span class="st">"feed"</span>, <span class="dv">3</span>)</span>
<span id="cb87-29"><a href="#cb87-29" aria-hidden="true" tabindex="-1"></a>  )) <span class="sc">%&gt;%</span> </span>
<span id="cb87-30"><a href="#cb87-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">cols =</span> <span class="fu">c</span>(P,L,C), <span class="at">names_to =</span> <span class="st">"macro"</span>, <span class="at">values_to =</span> <span class="st">"value"</span>, </span>
<span id="cb87-31"><a href="#cb87-31" aria-hidden="true" tabindex="-1"></a>               <span class="at">names_transform =</span> <span class="fu">list</span>(<span class="at">macro =</span> as.factor), <span class="at">values_transform =</span> <span class="fu">list</span>(<span class="at">value =</span> as.numeric)) <span class="sc">%&gt;%</span> </span>
<span id="cb87-32"><a href="#cb87-32" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">source =</span> <span class="fu">factor</span>(source, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"feed"</span>, <span class="st">"excreted"</span>)),</span>
<span id="cb87-33"><a href="#cb87-33" aria-hidden="true" tabindex="-1"></a>         <span class="at">macro =</span> <span class="fu">factor</span>(macro, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"P"</span>, <span class="st">"L"</span>, <span class="st">"C"</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Protein"</span>, <span class="st">"Lipids"</span>, <span class="st">"Carbohydrates"</span>)),</span>
<span id="cb87-34"><a href="#cb87-34" aria-hidden="true" tabindex="-1"></a>         <span class="at">feed =</span> <span class="fu">factor</span>(feed, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"reference"</span>, <span class="st">"past"</span>, <span class="st">"future"</span>)),</span>
<span id="cb87-35"><a href="#cb87-35" aria-hidden="true" tabindex="-1"></a>         <span class="at">source =</span> <span class="fu">factor</span>(source, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"feed"</span>, <span class="st">"excreted"</span>), <span class="at">labels =</span> <span class="fu">c</span>(<span class="st">"Feed"</span>, <span class="st">"Faeces"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<div class=""><p>Unlike the uneaten feed, the composition of excreted faeces differs from that of the incoming feed. </p><div class="no-row-height"><span class="margin-aside">Note that composition does not change through time or with increasing biomass (because digestibility does not change with fish size or temeprature)</span></div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [26]:</pre></div><div class="cell" data-layout-align="center">
<div class="margin-aside">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb88"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb88-1"><a href="#cb88-1" aria-hidden="true" tabindex="-1"></a>excreted_all <span class="sc">%&gt;%</span> </span>
<span id="cb88-2"><a href="#cb88-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_longer</span>(<span class="at">names_to =</span> <span class="st">"macro"</span>, <span class="at">values_to =</span> <span class="st">"perc"</span>, <span class="at">cols =</span> <span class="fu">c</span>(<span class="st">"protein"</span>, <span class="st">"lipids"</span>, <span class="st">"carbohydrates"</span>)) <span class="sc">%&gt;%</span> </span>
<span id="cb88-3"><a href="#cb88-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">macro =</span> <span class="fu">factor</span>(macro, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">"protein"</span>, <span class="st">"lipids"</span>, <span class="st">"carbohydrates"</span>))) <span class="sc">%&gt;%</span> </span>
<span id="cb88-4"><a href="#cb88-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(farm_ID, feed, macro) <span class="sc">%&gt;%</span> </span>
<span id="cb88-5"><a href="#cb88-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">reframe</span>(<span class="at">sd =</span> <span class="fu">sd</span>(perc) <span class="sc">%&gt;%</span> <span class="fu">set_units</span>(<span class="st">"d-1"</span>),</span>
<span id="cb88-6"><a href="#cb88-6" aria-hidden="true" tabindex="-1"></a>          <span class="at">mean =</span> <span class="fu">mean</span>(perc)) <span class="sc">%&gt;%</span> </span>
<span id="cb88-7"><a href="#cb88-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> farm_ID, <span class="at">colour =</span> feed, <span class="at">fill =</span> feed, <span class="at">y =</span> <span class="dv">100</span><span class="sc">*</span>mean, <span class="at">ymin =</span> <span class="dv">100</span><span class="sc">*</span>(mean<span class="sc">-</span>sd), <span class="at">ymax =</span> <span class="dv">100</span><span class="sc">*</span>(mean<span class="sc">+</span>sd))) <span class="sc">+</span></span>
<span id="cb88-8"><a href="#cb88-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">size =</span> <span class="dv">2</span>) <span class="sc">+</span></span>
<span id="cb88-9"><a href="#cb88-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_errorbar</span>(<span class="at">width =</span> <span class="fl">0.5</span>) <span class="sc">+</span></span>
<span id="cb88-10"><a href="#cb88-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> feed_pal) <span class="sc">+</span></span>
<span id="cb88-11"><a href="#cb88-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> feed_pal) <span class="sc">+</span></span>
<span id="cb88-12"><a href="#cb88-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">rows =</span> <span class="fu">vars</span>(macro)) <span class="sc">+</span></span>
<span id="cb88-13"><a href="#cb88-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb88-14"><a href="#cb88-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Macronutrient"</span>, <span class="at">y =</span> <span class="st">"Mean composition of faeces"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div>
</div></div>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [27]:</pre></div><div id="cell-fig-composition-comparison" class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb89"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb89-1"><a href="#cb89-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(ec, <span class="fu">aes</span>(<span class="at">x =</span> source, <span class="at">y =</span> <span class="dv">100</span><span class="sc">*</span>value, <span class="at">fill =</span> feed)) <span class="sc">+</span></span>
<span id="cb89-2"><a href="#cb89-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">position =</span> <span class="fu">position_dodge</span>(<span class="at">width =</span> <span class="fl">0.9</span>), <span class="at">alpha =</span> <span class="fl">0.75</span>) <span class="sc">+</span></span>
<span id="cb89-3"><a href="#cb89-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">facet_grid</span>(<span class="at">cols =</span> <span class="fu">vars</span>(macro)) <span class="sc">+</span></span>
<span id="cb89-4"><a href="#cb89-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb89-5"><a href="#cb89-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> feed_pal) <span class="sc">+</span></span>
<span id="cb89-6"><a href="#cb89-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="dv">0</span>, <span class="dv">60</span>, <span class="dv">10</span>)) <span class="sc">+</span></span>
<span id="cb89-7"><a href="#cb89-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">y =</span> <span class="st">"Composition (%)"</span>, <span class="at">x =</span> <span class="st">"Source"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-composition-comparison" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-composition-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-composition-comparison-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-13" title="Figure&nbsp;12: Comparison of the composition of the reference (blue), past (red) and future (green) feed and faeces of salmon fed each feed."><img src="manuscript_files/figure-html/fig-composition-comparison-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="816"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-composition-comparison-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;12: Comparison of the composition of the reference (blue), past (red) and future (green) feed and faeces of salmon fed each feed.
</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
</section>
<section id="total-nutrients" class="level2">
<h2 class="anchored" data-anchor-id="total-nutrients">Total nutrients</h2>
</section>
</section>
<section id="statistical-analysis-code-availability" class="level1">
<h1>Statistical analysis &amp; code availability</h1>
<p>All analysis was conducted in R version 4.4.2 “Pile of Leaves” <span class="citation" data-cites="R_base">(<a href="#ref-R_base" role="doc-biblioref">R Core Team 2019</a>)</span>. Major packages used include <code>terra</code> <span class="citation" data-cites="terra">(<a href="#ref-terra" role="doc-biblioref">Hijmans 2024</a>)</span>, <code>targets</code> <span class="citation" data-cites="targets">(<a href="#ref-targets" role="doc-biblioref">Landau 2021</a>)</span>, <code>future</code> <span class="citation" data-cites="future">(<a href="#ref-future" role="doc-biblioref">Bengtsson 2021</a>)</span>, <code>furrr</code> <span class="citation" data-cites="furrr">(<a href="#ref-furrr" role="doc-biblioref">Vaughan and Dancho 2022</a>)</span>, <code>renv</code> <span class="citation" data-cites="renv">(<a href="#ref-renv" role="doc-biblioref">Ushey and Wickham 2025</a>)</span>, and the <code>tidyverse</code> <span class="citation" data-cites="tidyverse">(<a href="#ref-tidyverse" role="doc-biblioref">Wickham et al. 2019</a>)</span>. For a full list of R packages used see the lockfile on <a href="https://www.github.com">Github</a>. This manuscript was written in Quarto <span class="citation" data-cites="quarto">(<a href="#ref-quarto" role="doc-biblioref">Allaire et al. 2024</a>)</span> using TinyTex <span class="citation" data-cites="tinytex">(<a href="#ref-tinytex" role="doc-biblioref">Xie 2017</a>)</span> and the acronyms extension <span class="citation" data-cites="acronyms">(<a href="#ref-acronyms" role="doc-biblioref">Chaput 2024</a>)</span>.</p>
</section>
<section id="supplementary-stuff" class="level1">
<h1>Supplementary stuff</h1>
<section id="sec-parameter-sensitivity" class="level2">
<h2 class="anchored" data-anchor-id="sec-parameter-sensitivity">Parameter sensitivity</h2>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [28]:</pre></div><div class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb90"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb90-1"><a href="#cb90-1" aria-hidden="true" tabindex="-1"></a>sens_weight <span class="ot">&lt;-</span> output_sens_data_path <span class="sc">%&gt;%</span> </span>
<span id="cb90-2"><a href="#cb90-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list.files</span>(<span class="at">full.names =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb90-3"><a href="#cb90-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_subset</span>(<span class="st">"results"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb90-4"><a href="#cb90-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_subset</span>(<span class="st">"weight"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb90-5"><a href="#cb90-5" aria-hidden="true" tabindex="-1"></a>  qs<span class="sc">::</span><span class="fu">qread</span>()</span>
<span id="cb90-6"><a href="#cb90-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-7"><a href="#cb90-7" aria-hidden="true" tabindex="-1"></a>sens_excr <span class="ot">&lt;-</span> output_sens_data_path <span class="sc">%&gt;%</span> </span>
<span id="cb90-8"><a href="#cb90-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">list.files</span>(<span class="at">full.names =</span> T) <span class="sc">%&gt;%</span> </span>
<span id="cb90-9"><a href="#cb90-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_subset</span>(<span class="st">"results"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb90-10"><a href="#cb90-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_subset</span>(<span class="st">"total_excr"</span>) <span class="sc">%&gt;%</span> </span>
<span id="cb90-11"><a href="#cb90-11" aria-hidden="true" tabindex="-1"></a>  qs<span class="sc">::</span><span class="fu">qread</span>() </span>
<span id="cb90-12"><a href="#cb90-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb90-13"><a href="#cb90-13" aria-hidden="true" tabindex="-1"></a>sens_data <span class="ot">&lt;-</span> <span class="fu">rbind</span>(sens_weight, sens_excr) <span class="sc">%&gt;%</span> </span>
<span id="cb90-14"><a href="#cb90-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">adj_param =</span> <span class="fu">factor</span>(adj_param, <span class="at">levels =</span> param_names<span class="sc">$</span>name))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
</div></div>
<p>Looking at <a href="#fig-param-sens-weight" class="quarto-xref">Figure&nbsp;13</a>:</p>
<ul>
<li>The most sensitive parameter by far is <span class="math inline">\(m\)</span> (the weight exponent for anabolism), followed closely by <span class="math inline">\(n\)</span> the (weight exponent for catabolism)</li>
<li>These two are also relatively uncertain, as <span class="math inline">\(m\)</span> was taken from <em>Salmo trutta</em> measurements in a fairly old paper <span class="citation" data-cites="broekhuizen_modelling_1994">(<a href="#ref-broekhuizen_modelling_1994" role="doc-biblioref">Broekhuizen et al. 1994</a>)</span> and <span class="math inline">\(n\)</span> was taken from <span class="citation" data-cites="baldan_r_2018">Baldan et al. (<a href="#ref-baldan_r_2018" role="doc-biblioref">2018</a>)</span> which did not specifically look at Atlantic salmon. There must be more up-to-date measurements for Atlantic salmon in particular.</li>
</ul>
<div class="cell-container code-fold"><div class="cell-decorator"><pre>In [29]:</pre></div><div id="cell-fig-param-sens-weight" class="cell" data-layout-align="center">
<details class="code-fold">
<summary>Code</summary>
<div class="sourceCode cell-code" id="cb91"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb91-1"><a href="#cb91-1" aria-hidden="true" tabindex="-1"></a>sens_data <span class="sc">%&gt;%</span> </span>
<span id="cb91-2"><a href="#cb91-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> adj_param, <span class="at">y =</span> mean_sens, <span class="at">ymin =</span> mean_sens <span class="sc">-</span> sd_sens, <span class="at">ymax =</span> mean_sens <span class="sc">+</span> sd_sens, <span class="at">fill =</span> measure)) <span class="sc">+</span></span>
<span id="cb91-3"><a href="#cb91-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">position =</span> <span class="fu">position_dodge</span>(), <span class="at">width =</span> <span class="fl">0.95</span>, <span class="at">colour =</span> <span class="st">"black"</span>, <span class="at">alpha =</span> <span class="fl">0.75</span>) <span class="sc">+</span></span>
<span id="cb91-4"><a href="#cb91-4" aria-hidden="true" tabindex="-1"></a>  <span class="co"># geom_errorbar(position = position_dodge(width = 0.95), width = 0.3) +</span></span>
<span id="cb91-5"><a href="#cb91-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="fu">aes</span>(<span class="at">yintercept =</span> <span class="dv">0</span>), <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb91-6"><a href="#cb91-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">breaks =</span> <span class="fu">seq</span>(<span class="sc">-</span><span class="dv">30</span>,<span class="dv">30</span>,<span class="dv">5</span>), <span class="at">limits =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">10</span>,<span class="dv">30</span>)) <span class="sc">+</span></span>
<span id="cb91-7"><a href="#cb91-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_x_discrete</span>(<span class="at">limits =</span> <span class="fu">rev</span>(param_names<span class="sc">$</span>name), <span class="at">labels =</span> <span class="fu">rev</span>(param_names<span class="sc">$</span>lab)) <span class="sc">+</span></span>
<span id="cb91-8"><a href="#cb91-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="at">weight =</span> <span class="st">"salmon"</span>, <span class="at">total_excr =</span> <span class="st">"steelblue"</span>)) <span class="sc">+</span></span>
<span id="cb91-9"><a href="#cb91-9" aria-hidden="true" tabindex="-1"></a>  <span class="co"># scale_fill_manual(values = c("weight" = "salmon", "uneat" = "steelblue", "excr" = "darkgreen")) +</span></span>
<span id="cb91-10"><a href="#cb91-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">x =</span> <span class="st">"Adjusted parameter"</span>, <span class="at">y =</span> <span class="st">"Sensitivity"</span>) <span class="sc">+</span></span>
<span id="cb91-11"><a href="#cb91-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">prettyplot</span>() <span class="sc">+</span></span>
<span id="cb91-12"><a href="#cb91-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">strip.text =</span> <span class="fu">element_blank</span>(), </span>
<span id="cb91-13"><a href="#cb91-13" aria-hidden="true" tabindex="-1"></a>        <span class="at">aspect.ratio =</span> <span class="fl">0.8</span>,</span>
<span id="cb91-14"><a href="#cb91-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>()) <span class="sc">+</span></span>
<span id="cb91-15"><a href="#cb91-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">coord_flip</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</details>
<div class="cell-output-display">
<div id="fig-param-sens-weight" class="quarto-float quarto-figure quarto-figure-center anchored" data-fig-align="center">
<figure class="quarto-float quarto-float-fig figure">
<div aria-describedby="fig-param-sens-weight-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca" class="">
<a href="manuscript_files/figure-html/fig-param-sens-weight-1.png" class="lightbox" data-gallery="quarto-lightbox-gallery-14" title="Figure&nbsp;13: Impact of increasing each model parameter by 10% on final farm biomass (red) and total excreted nutrients (blue) over the whole production period."><img src="manuscript_files/figure-html/fig-param-sens-weight-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="816"></a>
</div>
<figcaption class="quarto-float-caption-bottom quarto-float-caption quarto-float-fig" id="fig-param-sens-weight-caption-0ceaefa1-69ba-4598-a22c-09a6ac19f8ca">
Figure&nbsp;13: Impact of increasing each model parameter by 10% on final farm biomass (red) and total excreted nutrients (blue) over the whole production period.
</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
</section>
<section id="references" class="level1 unnumbered">
<h1 class="unnumbered">References</h1>
<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-quarto" class="csl-entry" role="listitem">
Allaire, J. J., Charles Teague, Carlos Scheidegger, Yihui Xie, and Christophe Dervieux. 2024. <span>‘<span>Quarto</span>’</span>. <a href="https://doi.org/10.5281/zenodo.5960048">https://doi.org/10.5281/zenodo.5960048</a>.
</div>
<div id="ref-baldan_r_2018" class="csl-entry" role="listitem">
Baldan, Damiano, Erika M. D. Porporato, Roberto Pastres, and Daniele Brigolin. 2018. <span>‘An <span>R</span> Package for Simulating Growth and Organic Wastage in Aquaculture Farms in Response to Environmental Conditions and Husbandry Practices’</span>. <em>PLOS ONE</em> 13 (5): e0195732. https://doi.org/<a href="https://doi.org/10.1371/journal.pone.0195732">https://doi.org/10.1371/journal.pone.0195732</a>.
</div>
<div id="ref-future" class="csl-entry" role="listitem">
Bengtsson, Henrik. 2021. <span>‘A Unifying Framework for Parallel and Distributed Processing in r Using Futures’</span>. <em>The R Journal</em> 13 (2): 208–27. <a href="https://doi.org/10.32614/RJ-2021-048">https://doi.org/10.32614/RJ-2021-048</a>.
</div>
<div id="ref-broekhuizen_modelling_1994" class="csl-entry" role="listitem">
Broekhuizen, Niall., William. S. C. Gurney, A. Jones, and A. D. Bryant. 1994. <span>‘Modelling <span>Compensatory</span> <span>Growth</span>’</span>. <em>Functional Ecology</em> 8 (6): 770–82. https://doi.org/<a href="https://doi.org/10.2307/2390237">https://doi.org/10.2307/2390237</a>.
</div>
<div id="ref-acronyms" class="csl-entry" role="listitem">
Chaput, Remy. 2024. <span>‘<span class="nocase">acronyms</span>’</span>. <a href="https://github.com/rchaput/acronyms">https://github.com/rchaput/acronyms</a>.
</div>
<div id="ref-cottrell_global_2020" class="csl-entry" role="listitem">
Cottrell, Richard S., Julia L. Blanchard, Benjamin S. Halpern, Marc Metian, and Halley E. Froehlich. 2020. <span>‘Global Adoption of Novel Aquaculture Feeds Could Substantially Reduce Forage Fish Demand by 2030’</span>. <em>Nature Food</em> 1 (5): 301–8. <a href="https://doi.org/10.1038/s43016-020-0078-x">https://doi.org/10.1038/s43016-020-0078-x</a>.
</div>
<div id="ref-cottrell_origins_nodate" class="csl-entry" role="listitem">
Cottrell, Richard S., Benjamin S. Halpern, Marceau Cormery, Helen A. Hamilton, Gage Clawson, Beth Penrose, Louise Adams, et al. n.d. <span>‘The Origins of Ingredients Matter More Than Ingredient Types for the Environmental Footprint of Aquaculture Feeds’</span>.
</div>
<div id="ref-dempsey_estimating_2023" class="csl-entry" role="listitem">
Dempsey, Danielle P., Gregor K. Reid, Leah Lewis-McCrea, Toby Balch, Roland Cusack, André Dumas, and Jack Rensel. 2023. <span>‘Estimating Stocking Weights for <span>Atlantic</span> Salmon to Grow to Market Size at Novel Aquaculture Sites with Extreme Temperatures’</span>. <em>Frontiers in Marine Science</em> 10 (May). https://doi.org/<a href="https://doi.org/ 10.3389/fmars.2023.1094247">https://doi.org/ 10.3389/fmars.2023.1094247</a>.
</div>
<div id="ref-elliott_energy_1975" class="csl-entry" role="listitem">
Elliott, J. M., and W. Davison. 1975. <span>‘Energy Equivalents of Oxygen Consumption in Animal Energetics’</span>. <em>Oecologia</em> 19 (3): 195–201. https://doi.org/<a href="https://doi.org/10.1007/BF00345305">https://doi.org/10.1007/BF00345305</a>.
</div>
<div id="ref-handeland_effect_2008" class="csl-entry" role="listitem">
Handeland, Sigurd O., Albert K. Imsland, and Sigurd O. Stefansson. 2008. <span>‘The Effect of Temperature and Fish Size on Growth, Feed Intake, Food Conversion Efficiency and Stomach Evacuation Rate of <span>Atlantic</span> Salmon Post-Smolts’</span>. <em>Aquaculture</em> 283 (1): 36–42. https://doi.org/<a href="https://doi.org/ 10.1016/j.aquaculture.2008.06.042">https://doi.org/ 10.1016/j.aquaculture.2008.06.042</a>.
</div>
<div id="ref-terra" class="csl-entry" role="listitem">
Hijmans, Robert J. 2024. <em>Terra: Spatial Data Analysis</em>. <a href="https://CRAN.R-project.org/package=terra">https://CRAN.R-project.org/package=terra</a>.
</div>
<div id="ref-jonsson_thermal_2001" class="csl-entry" role="listitem">
Jonsson, Bror, Torbjørn Forseth, Arne. J. Jensen, and Tor. F. Næsje. 2001. <span>‘Thermal Performance of Juvenile <span>Atlantic</span> <span>Salmon</span>, <em><span>Salmo</span> Salar</em> <span>L</span>.’</span> <em>Functional Ecology</em> 15 (6): 701–11. https://doi.org/<a href="https://doi.org/10.1046/j.0269-8463.2001.00572.x">https://doi.org/10.1046/j.0269-8463.2001.00572.x</a>.
</div>
<div id="ref-jonsson_energy_2003" class="csl-entry" role="listitem">
Jonsson, Nina, and Bror Jonsson. 2003. <span>‘Energy Allocation Among Developmental Stages, Age Groups, and Types of <span>Atlantic</span> Salmon (<em><span>Salmo</span> Salar</em>) Spawners’</span>. <em>Canadian Journal of Fisheries and Aquatic Sciences</em> 60 (5): 506–16. https://doi.org/<a href="https://doi.org/ 10.1139/f03-042">https://doi.org/ 10.1139/f03-042</a>.
</div>
<div id="ref-targets" class="csl-entry" role="listitem">
Landau, William Michael. 2021. <span>‘The Targets r Package: A Dynamic Make-Like Function-Oriented Pipeline Toolkit for Reproducibility and High-Performance Computing’</span>. <em>Journal of Open Source Software</em> 6 (57): 2959. <a href="https://doi.org/10.21105/joss.02959">https://doi.org/10.21105/joss.02959</a>.
</div>
<div id="ref-macnaughton_using_2019" class="csl-entry" role="listitem">
Macnaughton, Camille J., David Deslauriers, Erinn L. Ipsen, Emily Corey, and Eva C. Enders. 2019. <span>‘Using Meta-Analysis to Derive a Respiration Model for <span>Atlantic</span> Salmon (<em><span>Salmo</span> Salar</em>) to Assess Bioenergetics Requirements of Juveniles in Two <span>Canadian</span> Rivers’</span>. <em>Canadian Journal of Fisheries and Aquatic Sciences</em> 76 (12): 2225–34. https://doi.org/<a href="https://doi.org/ 10.1139/cjfas-2018-0436">https://doi.org/ 10.1139/cjfas-2018-0436</a>.
</div>
<div id="ref-poletto_unusual_2017" class="csl-entry" role="listitem">
Poletto, Jamilynn B., Dennis E. Cocherell, Sarah E. Baird, Trinh X. Nguyen, Valentina Cabrera-Stagno, Anthony P. Farrell, and Nann A. Fangue. 2017. <span>‘Unusual Aerobic Performance at High Temperatures in Juvenile <span>Chinook</span> Salmon, <em><span>Oncorhynchus</span> Tshawytscha</em>’</span>. <em>Conservation Physiology</em> 5 (1). https://doi.org/<a href="https://doi.org/ 10.1093/conphys/cow067">https://doi.org/ 10.1093/conphys/cow067</a>.
</div>
<div id="ref-R_base" class="csl-entry" role="listitem">
R Core Team. 2019. <em>R: A Language and Environment for Statistical Computing</em>. Vienna, Austria: R Foundation for Statistical Computing. <a href="https://www.R-project.org">https://www.R-project.org</a>.
</div>
<div id="ref-tvete_towards_2023" class="csl-entry" role="listitem">
Tvete, Ingunn Fride, Magne Aldrin, and Britt Bang Jensen. 2023. <span>‘Towards Better Survival: <span>Modeling</span> Drivers for Daily Mortality in <span>Norwegian</span> <span>Atlantic</span> Salmon Farming’</span>. <em>Preventive Veterinary Medicine</em> 210 (January): 105798. https://doi.org/<a href="https://doi.org/ 10.1016/j.prevetmed.2022.105798">https://doi.org/ 10.1016/j.prevetmed.2022.105798</a>.
</div>
<div id="ref-uglem_does_2020" class="csl-entry" role="listitem">
Uglem, Ingebrigt, Kilian Toledo-Guedes, Pablo Sanchez-Jerez, Eva M. Ulvan, Tor Evensen, and Bjørn S. Sæther. 2020. <span>‘Does Waste Feed from Salmon Farming Affect the Quality of Saithe (<em><span>Pollachius</span> Virens</em> <span>L</span>.) Attracted to Fish Farms?’</span> <em>Aquaculture Research</em> 51 (4): 1720–30. https://doi.org/<a href="https://doi.org/ 10.1111/are.14519">https://doi.org/ 10.1111/are.14519</a>.
</div>
<div id="ref-renv" class="csl-entry" role="listitem">
Ushey, Kevin, and Hadley Wickham. 2025. <em>Renv: Project Environments</em>. <a href="https://rstudio.github.io/renv/">https://rstudio.github.io/renv/</a>.
</div>
<div id="ref-vadboncoeur_lowering_2023" class="csl-entry" role="listitem">
Vadboncoeur, Émile, Charlotte Nelson, Jennifer R. Hall, Kathy A. Clow, Rebeccah M. Sandrelli, Colin J. Brauner, Andrew K. Swanson, and A. Kurt Gamperl. 2023. <span>‘Lowering Temperature to 1&nbsp;°<span>C</span> Results in Physiological Changes, Stress and Mortality in Cultured <span>Atlantic</span> <span>Salmon</span> (<em><span>Salmo</span> Salar</em>)’</span>. <em>Aquaculture</em> 568 (April): 739313. https://doi.org/<a href="https://doi.org/ 10.1016/j.aquaculture.2023.739313">https://doi.org/ 10.1016/j.aquaculture.2023.739313</a>.
</div>
<div id="ref-van_tien_development_2016" class="csl-entry" role="listitem">
Van Tien, Nguyen, Dam T. M. Chinh, Tran T. M. Huong, Truong H. Phuong, Simon Irvin, and Brett Glencross. 2016. <span>‘Development of a Nutritional Model to Define the Energy and Protein Requirements of Cobia, <em><span>Rachycentron</span> Canadum</em>’</span>. <em>Aquaculture</em> 463 (October): 193–200. https://doi.org/<a href="https://doi.org/ 10.1016/j.aquaculture.2016.05.037">https://doi.org/ 10.1016/j.aquaculture.2016.05.037</a>.
</div>
<div id="ref-furrr" class="csl-entry" role="listitem">
Vaughan, Davis, and Matt Dancho. 2022. <em>Furrr: Apply Mapping Functions in Parallel Using Futures</em>. <a href="https://CRAN.R-project.org/package=furrr">https://CRAN.R-project.org/package=furrr</a>.
</div>
<div id="ref-tidyverse" class="csl-entry" role="listitem">
Wickham, Hadley, Mara Averick, Jennifer Bryan, Winston Chang, Lucy D’Agostino McGowan, Romain François, Garrett Grolemund, et al. 2019. <span>‘Welcome to the <span class="nocase">tidyverse</span>’</span>. <em>Journal of Open Source Software</em> 4 (43): 1686. <a href="https://doi.org/10.21105/joss.01686">https://doi.org/10.21105/joss.01686</a>.
</div>
<div id="ref-tinytex" class="csl-entry" role="listitem">
Xie, Yihui. 2017. <span>‘Tinytex: Helper Functions to Install and Maintain TeX Live, and Compile LaTeX Documents’</span>. <a href="https://cran.r-project.org/web/packages/tinytex">https://cran.r-project.org/web/packages/tinytex</a>.
</div>
</div>
</section>
     </main>
<!-- /main column -->  <script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>  </div> <!-- /content -->  <script>var lightboxQuarto = GLightbox({"closeEffect":"zoom","selector":".lightbox","openEffect":"zoom","loop":false,"descPosition":"bottom"});
(function() {
  let previousOnload = window.onload;
  window.onload = () => {
    if (previousOnload) {
      previousOnload();
    }
    lightboxQuarto.on('slide_before_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      const href = trigger.getAttribute('href');
      if (href !== null) {
        const imgEl = window.document.querySelector(`a[href="${href}"] img`);
        if (imgEl !== null) {
          const srcAttr = imgEl.getAttribute("src");
          if (srcAttr && srcAttr.startsWith("data:")) {
            slideConfig.href = srcAttr;
          }
        }
      } 
    });
  
    lightboxQuarto.on('slide_after_load', (data) => {
      const { slideIndex, slideNode, slideConfig, player, trigger } = data;
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(slideNode);
      }
    });
  
  };
  
})();
          </script> 
  
</body></html>